// @alchemy/aa-accounts/dist/cjs/src/light-account/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLightAccountVersion = exports.getLightAccountVersionDef = exports.LightAccountUnsupported1271Factories = exports.LightAccountUnsupported1271Impls = exports.getDefaultLightAccountFactoryAddress = exports.defaultLightAccountVersion = exports.AccountVersionRegistry = exports.supportedChains = void 0;
const aa_core_1 = require("@alchemy/aa-core");
const viem_1 = require("viem");
exports.supportedChains = [
    aa_core_1.mainnet,
    aa_core_1.sepolia,
    aa_core_1.goerli,
    aa_core_1.polygon,
    aa_core_1.polygonAmoy,
    aa_core_1.polygonMumbai,
    aa_core_1.optimism,
    aa_core_1.optimismGoerli,
    aa_core_1.optimismSepolia,
    aa_core_1.arbitrum,
    aa_core_1.arbitrumGoerli,
    aa_core_1.arbitrumSepolia,
    aa_core_1.base,
    aa_core_1.baseGoerli,
    aa_core_1.baseSepolia,
    aa_core_1.fraxtal,
    aa_core_1.fraxtalSepolia,
    aa_core_1.zora,
    aa_core_1.zoraSepolia,
];
exports.AccountVersionRegistry = {
    LightAccount: {
        "v1.0.1": {
            type: "LightAccount",
            version: "v1.0.1",
            address: (0, aa_core_1.toRecord)(exports.supportedChains, "id", () => ({
                factory: "0x000000893A26168158fbeaDD9335Be5bC96592E2".toLowerCase(),
                impl: "0xc1b2fc4197c9187853243e6e4eb5a4af8879a1c0".toLowerCase(),
            })),
            entryPointVersion: "0.6.0",
        },
        "v1.0.2": {
            type: "LightAccount",
            version: "v1.0.2",
            address: (0, aa_core_1.toRecord)(exports.supportedChains, "id", () => ({
                factory: "0x00000055C0b4fA41dde26A74435ff03692292FBD".toLowerCase(),
                impl: "0x5467b1947F47d0646704EB801E075e72aeAe8113".toLowerCase(),
            })),
            entryPointVersion: "0.6.0",
        },
        "v1.1.0": {
            type: "LightAccount",
            version: "v1.1.0",
            address: (0, aa_core_1.toRecord)(exports.supportedChains, "id", () => ({
                factory: "0x00004EC70002a32400f8ae005A26081065620D20".toLowerCase(),
                impl: "0xae8c656ad28F2B59a196AB61815C16A0AE1c3cba".toLowerCase(),
            })),
            entryPointVersion: "0.6.0",
        },
        "v2.0.0": {
            type: "LightAccount",
            version: "v2.0.0",
            address: (0, aa_core_1.toRecord)(exports.supportedChains, "id", () => ({
                factory: "0x0000000000400CdFef5E2714E63d8040b700BC24".toLowerCase(),
                impl: "0x8E8e658E22B12ada97B402fF0b044D6A325013C7".toLowerCase(),
            })),
            entryPointVersion: "0.7.0",
        },
    },
    MultiOwnerLightAccount: {
        "v2.0.0": {
            type: "MultiOwnerLightAccount",
            version: "v2.0.0",
            address: (0, aa_core_1.toRecord)(exports.supportedChains, "id", () => ({
                factory: "0x000000000019d2Ee9F2729A65AfE20bb0020AefC".toLowerCase(),
                impl: "0xd2c27F9eE8E4355f71915ffD5568cB3433b6823D".toLowerCase(),
            })),
            entryPointVersion: "0.7.0",
        },
    },
};
const defaultLightAccountVersion = (type) => (type === "LightAccount"
    ? "v1.1.0"
    : "v2.0.0");
exports.defaultLightAccountVersion = defaultLightAccountVersion;
const getDefaultLightAccountFactoryAddress = (chain, version = "v1.1.0") => {
    const address = exports.AccountVersionRegistry.LightAccount[version].address[chain.id];
    if (!address)
        throw new aa_core_1.DefaultFactoryNotDefinedError("LightAccount", chain, "0.6.0");
    return address.factory;
};
exports.getDefaultLightAccountFactoryAddress = getDefaultLightAccountFactoryAddress;
exports.LightAccountUnsupported1271Impls = [
    exports.AccountVersionRegistry.LightAccount["v1.0.1"],
    exports.AccountVersionRegistry.LightAccount["v1.0.2"],
];
exports.LightAccountUnsupported1271Factories = new Set(exports.LightAccountUnsupported1271Impls.map((x) => Object.values(x.address).map((addr) => addr.factory)).flat());
async function getLightAccountVersionDef(account, chain) {
    const accountType = account.source;
    const factoryAddress = await account.getFactoryAddress();
    const implAddress = await account.getImplementationAddress();
    const implToVersion = new Map(Object.entries(exports.AccountVersionRegistry[accountType])
        .map((pair) => {
        const [version, def] = pair;
        return chain.id in def.address
            ? [def.address[chain.id].impl, version]
            : [null, version];
    })
        .filter(([impl]) => impl !== null));
    const factoryToVersion = new Map(Object.entries(exports.AccountVersionRegistry[accountType])
        .map((pair) => {
        const [version, def] = pair;
        return chain.id in def.address
            ? [def.address[chain.id].factory, version]
            : [null, version];
    })
        .filter(([impl]) => impl !== null));
    const version = (0, viem_1.fromHex)(implAddress, "bigint") === 0n
        ? factoryToVersion.get(factoryAddress.toLowerCase())
        : implToVersion.get(implAddress.toLowerCase());
    if (!version) {
        throw new Error(`Could not determine ${account.source} version for chain ${chain.id}`);
    }
    return exports.AccountVersionRegistry[accountType][version];
}
exports.getLightAccountVersionDef = getLightAccountVersionDef;
async function getLightAccountVersion(account, chain = exports.supportedChains[0]) {
    return (await getLightAccountVersionDef(account, chain)).version;
}
exports.getLightAccountVersion = getLightAccountVersion;
