// @alchemy/aa-accounts/dist/cjs/plugindefs/session-key/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionKeyPluginGenConfig = void 0;
const aa_core_1 = require("@alchemy/aa-core");
const viem_1 = require("viem");
const config_js_1 = require("../multi-owner/config.js");
const abi_js_1 = require("./abi.js");
exports.SessionKeyPluginGenConfig = {
    name: "SessionKeyPlugin",
    abi: abi_js_1.SessionKeyPluginAbi,
    addresses: {
        [aa_core_1.sepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.baseSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.polygon.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.polygonAmoy.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.polygonMumbai.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.optimism.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.optimismSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.arbitrum.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.arbitrumSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.base.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.baseSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.fraxtal.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.fraxtalSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.zora.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.zoraSepolia.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
        [aa_core_1.mainnet.id]: "0x0000003E0000a96de4058e1E02a62FaaeCf23d8d",
    },
    chain: aa_core_1.sepolia,
    installConfig: {
        initAbiParams: (0, viem_1.parseAbiParameters)("address[] initialKeys, bytes32[] tags, bytes[][] initialPermissions"),
        dependencies: [
            {
                plugin: config_js_1.MultiOwnerPluginGenConfig,
                functionId: "0x0",
            },
            {
                plugin: config_js_1.MultiOwnerPluginGenConfig,
                functionId: "0x1",
            },
        ],
    },
};
