// @alchemy/aa-accounts/dist/esm/src/msca/plugins/multisig/actions/signMultisigUserOperation.js
import { AccountNotFoundError, IncompatibleClientError, SmartAccountWithSignerRequiredError, isSmartAccountClient, isSmartAccountWithSigner, } from "@alchemy/aa-core";
import {} from "viem";
import { MultisigMissingSignatureError } from "../../../errors.js";
import { combineSignatures, splitAggregatedSignature } from "../index.js";
import {} from "../types.js";
export async function signMultisigUserOperation(client, params) {
    const { account = client.account, signatures, userOperationRequest } = params;
    if (!account) {
        throw new AccountNotFoundError();
    }
    if (!isSmartAccountClient(client)) {
        throw new IncompatibleClientError("SmartAccountClient", "signMultisigUserOperation", client);
    }
    if (!isSmartAccountWithSigner(account)) {
        throw new SmartAccountWithSignerRequiredError();
    }
    if (!signatures.length) {
        throw new MultisigMissingSignatureError();
    }
    const signerAddress = await account.getSigner().getAddress();
    const signedRequest = await client.signUserOperation({
        account,
        uoStruct: userOperationRequest,
        context: {
            aggregatedSignature: combineSignatures({
                signatures,
                upperLimitMaxFeePerGas: userOperationRequest.maxFeePerGas,
                upperLimitMaxPriorityFeePerGas: userOperationRequest.maxPriorityFeePerGas,
                upperLimitPvg: userOperationRequest.preVerificationGas,
                usingMaxValues: false,
            }),
            signatures,
            userOpSignatureType: "UPPERLIMIT",
        },
    });
    const splitSignatures = await splitAggregatedSignature({
        account,
        request: signedRequest,
        aggregatedSignature: signedRequest.signature,
        threshold: signatures.length + 2,
    });
    const signatureObj = splitSignatures.signatures.find((x) => x.signer === signerAddress);
    if (!signatureObj) {
        throw new Error("INTERNAL ERROR: signature not found in split signatures, this is an internal bug please report");
    }
    return {
        signatureObj,
        signature: signatureObj.signature,
        aggregatedSignature: signedRequest.signature,
    };
}
