// @alchemy/aa-accounts/dist/cjs/src/msca/account/multiOwnerAccount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMultiOwnerModularAccount = void 0;
const aa_core_1 = require("@alchemy/aa-core");
const viem_1 = require("viem");
const MultiOwnerModularAccountFactory_js_1 = require("../abis/MultiOwnerModularAccountFactory.js");
const signer_js_1 = require("../plugins/multi-owner/signer.js");
const utils_js_1 = require("../utils.js");
const standardExecutor_js_1 = require("./standardExecutor.js");
async function createMultiOwnerModularAccount({ transport, chain, signer, accountAddress, initCode, entryPoint = (0, aa_core_1.getEntryPoint)(chain, { version: "0.6.0" }), factoryAddress = (0, utils_js_1.getDefaultMultiOwnerModularAccountFactoryAddress)(chain), owners = [], salt = 0n, }) {
    const client = (0, aa_core_1.createBundlerClient)({
        transport,
        chain,
    });
    const getAccountInitCode = async () => {
        if (initCode) {
            return initCode;
        }
        const ownerAddress = await signer.getAddress();
        const owners_ = Array.from(new Set([...owners, ownerAddress]))
            .filter((x) => (0, viem_1.hexToBigInt)(x) !== 0n)
            .sort((a, b) => {
            const bigintA = (0, viem_1.hexToBigInt)(a);
            const bigintB = (0, viem_1.hexToBigInt)(b);
            return bigintA < bigintB ? -1 : bigintA > bigintB ? 1 : 0;
        });
        return (0, viem_1.concatHex)([
            factoryAddress,
            (0, viem_1.encodeFunctionData)({
                abi: MultiOwnerModularAccountFactory_js_1.MultiOwnerModularAccountFactoryAbi,
                functionName: "createAccount",
                args: [salt, owners_],
            }),
        ]);
    };
    const _accountAddress = await (0, aa_core_1.getAccountAddress)({
        client,
        entryPoint,
        accountAddress,
        getAccountInitCode,
    });
    const baseAccount = await (0, aa_core_1.toSmartContractAccount)({
        transport,
        chain,
        entryPoint,
        accountAddress: _accountAddress,
        source: `MultiOwnerModularAccount`,
        getAccountInitCode,
        ...standardExecutor_js_1.standardExecutor,
        ...(0, signer_js_1.multiOwnerMessageSigner)(client, _accountAddress, () => signer),
    });
    return {
        ...baseAccount,
        publicKey: await signer.getAddress(),
        getSigner: () => signer,
    };
}
exports.createMultiOwnerModularAccount = createMultiOwnerModularAccount;
