// @alchemy/aa-accounts/dist/cjs/plugindefs/multisig/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultisigPluginGenConfig = void 0;
const aa_core_1 = require("@alchemy/aa-core");
const viem_1 = require("viem");
const abi_js_1 = require("./abi.js");
exports.MultisigPluginGenConfig = {
    name: "MultisigPlugin",
    abi: abi_js_1.MultisigPluginAbi,
    addresses: {
        [aa_core_1.sepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.baseSepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.polygon.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.mainnet.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.polygonAmoy.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.optimism.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.optimismSepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.arbitrum.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.arbitrumSepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.base.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.fraxtal.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.fraxtalSepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.zora.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
        [aa_core_1.zoraSepolia.id]: "0x000000000000A53f64b7bcf4Cd59624943C43Fc7",
    },
    chain: aa_core_1.sepolia,
    installConfig: {
        initAbiParams: (0, viem_1.parseAbiParameters)("address[], uint"),
    },
};
