// @bgd-labs/aave-address-book/dist/AaveV3Gnosis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/AaveV3Gnosis.ts
var AaveV3Gnosis_exports = {};
__export(AaveV3Gnosis_exports, {
  AAVE_PROTOCOL_DATA_PROVIDER: () => AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN: () => ACL_ADMIN,
  ACL_MANAGER: () => ACL_MANAGER,
  ASSETS: () => ASSETS,
  CHAIN_ID: () => CHAIN_ID,
  CLINIC_STEWARD: () => CLINIC_STEWARD,
  COLLECTOR: () => COLLECTOR,
  CONFIG_ENGINE: () => CONFIG_ENGINE,
  DEBT_SWAP_ADAPTER: () => DEBT_SWAP_ADAPTER,
  DEFAULT_A_TOKEN_IMPL: () => DEFAULT_A_TOKEN_IMPL,
  DEFAULT_INCENTIVES_CONTROLLER: () => DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL: () => DEFAULT_VARIABLE_DEBT_TOKEN_IMPL,
  DUST_BIN: () => DUST_BIN,
  EDGE_INJECTOR_CAPS: () => EDGE_INJECTOR_CAPS,
  EDGE_RISK_STEWARD_CAPS: () => EDGE_RISK_STEWARD_CAPS,
  EMISSION_MANAGER: () => EMISSION_MANAGER,
  EXTERNAL_LIBRARIES: () => EXTERNAL_LIBRARIES,
  E_MODES: () => E_MODES,
  LEGACY_STATIC_A_TOKEN_FACTORY: () => LEGACY_STATIC_A_TOKEN_FACTORY,
  ORACLE: () => ORACLE,
  POOL: () => POOL,
  POOL_ADDRESSES_PROVIDER: () => POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY: () => POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR: () => POOL_CONFIGURATOR,
  POOL_CONFIGURATOR_IMPL: () => POOL_CONFIGURATOR_IMPL,
  POOL_EXPOSURE_STEWARD: () => POOL_EXPOSURE_STEWARD,
  POOL_IMPL: () => POOL_IMPL,
  REPAY_WITH_COLLATERAL_ADAPTER: () => REPAY_WITH_COLLATERAL_ADAPTER,
  RISK_STEWARD: () => RISK_STEWARD,
  STATA_FACTORY: () => STATA_FACTORY,
  SWAP_COLLATERAL_ADAPTER: () => SWAP_COLLATERAL_ADAPTER,
  UI_INCENTIVE_DATA_PROVIDER: () => UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER: () => UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER: () => WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY: () => WETH_GATEWAY,
  WITHDRAW_SWAP_ADAPTER: () => WITHDRAW_SWAP_ADAPTER
});
module.exports = __toCommonJS(AaveV3Gnosis_exports);
var POOL_ADDRESSES_PROVIDER = "0x36616cf17557639614c1cdDb356b1B83fc0B2132";
var POOL = "0xb50201558B00496A145fE76f7424749556E326D8";
var POOL_CONFIGURATOR = "0x7304979ec9E4EaA0273b6A037a31c4e9e5A75D16";
var ORACLE = "0xeb0a051be10228213BAEb449db63719d6742F7c4";
var ACL_ADMIN = "0x1dF462e2712496373A347f8ad10802a5E95f053D";
var ACL_MANAGER = "0xEc710f59005f48703908bC519D552Df5B8472614";
var AAVE_PROTOCOL_DATA_PROVIDER = "0xF1F5acB596568895393cB5E4D0452D6592A2fA70";
var POOL_IMPL = "0xf30E36C6bf04067B721636525Cf8Cb44ef73E12a";
var POOL_CONFIGURATOR_IMPL = "0x5793FE4de34532F162B4e207aF872729880ec2b6";
var DEFAULT_INCENTIVES_CONTROLLER = "0xaD4F91D26254B6B0C6346b390dDA2991FDE2F20d";
var EMISSION_MANAGER = "0x41585C50524fb8c3899B43D7D797d9486AAc94DB";
var COLLECTOR = "0x3e652E97ff339B73421f824F5b03d75b62F1Fb51";
var DEFAULT_A_TOKEN_IMPL = "0xe04BFE47101bBe9Aa35f09210589517F04136d23";
var DEFAULT_VARIABLE_DEBT_TOKEN_IMPL = "0x71Fe38F6c9DFb5B15Fa69B26B2D4208F3B17Eb02";
var CONFIG_ENGINE = "0x0FC9eB644dF453B53C7d9A6892c878f14382ddc3";
var POOL_ADDRESSES_PROVIDER_REGISTRY = "0x1236010CECea55998384e795B59815D871f5f94d";
var UI_INCENTIVE_DATA_PROVIDER = "0x5d4D4007A4c6336550DdAa2a7c0d5e7972eebd16";
var UI_POOL_DATA_PROVIDER = "0x5598BbFA2f4fE8151f45bBA0a3edE1b54B51a0a9";
var WALLET_BALANCE_PROVIDER = "0x4172E6aAEC070ACB31aaCE343A58c93E4C70f44D";
var WETH_GATEWAY = "0x721B9abAb6511b46b9ee83A1aba23BDAcB004149";
var RISK_STEWARD = "0x1AA25FdD7d55FA8a401D6EFba8e48874Ef730E55";
var LEGACY_STATIC_A_TOKEN_FACTORY = "0x02e9b27599C4Bf8f789d34b6E65C51092c3d9FA6";
var STATA_FACTORY = "0x33992721c565dA3248bd3af80524e054F5F05b42";
var REPAY_WITH_COLLATERAL_ADAPTER = "0x86b0521f92a554057e54B93098BA2A6Aaa2F4ACB";
var SWAP_COLLATERAL_ADAPTER = "0x63dfa7c09Dc2Ff4030d6B8Dc2ce6262BF898C8A4";
var DEBT_SWAP_ADAPTER = "0xE28E2c8d240dd5eBd0adcab86fbD79df7a052034";
var WITHDRAW_SWAP_ADAPTER = "0xc0179321f0825c3e0F59Fe7Ca4E40557b97797a3";
var DUST_BIN = "0xFf3a13703236A77Da603c7D413cf6F6AfB866D34";
var CLINIC_STEWARD = "0x0447Be151c126D2d704b1Ce5F5b3ed6b7D50FABA";
var POOL_EXPOSURE_STEWARD = "0xdA12A8F009258592A9B5618129259992858636dc";
var EDGE_RISK_STEWARD_CAPS = "0x655252250f4A453854040A49E8280951A76f3033";
var EDGE_INJECTOR_CAPS = "0x394aDC8772DDD076BD3c5C545c4Edd3617C7d5e6";
var CHAIN_ID = 100;
var ASSETS = {
  WETH: {
    decimals: 18,
    id: 0,
    UNDERLYING: "0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1",
    A_TOKEN: "0xa818F1B57c201E092C4A2017A91815034326Efd1",
    V_TOKEN: "0x0c0fce05F2314540EcB095bF4D069e5E0ED90fF8",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0xa767f745331D267c7751297D982b050c93985627",
    STATIC_A_TOKEN: "0xD843FB478c5aA9759FeA3f3c98D467e2F136190a",
    STATA_TOKEN: "0x57f664882F762FA37903FC864e2B633D384B411A"
  },
  wstETH: {
    decimals: 18,
    id: 1,
    UNDERLYING: "0x6C76971f98945AE98dD7d4DFcA8711ebea946eA6",
    A_TOKEN: "0x23e4E76D01B2002BE436CE8d6044b0aA2f68B68a",
    V_TOKEN: "0x9D881f67F20B49243c98f53d2B9E91E39d02Ae09",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x8Ee42Ba520cA106294163fb8b1ffE9C6Fba35507",
    STATIC_A_TOKEN: "0xECfD0638175e291BA3F784A58FB9D38a25418904",
    STATA_TOKEN: "0x773CDA0CADe2A3d86E6D4e30699d40bB95174ff2"
  },
  GNO: {
    decimals: 18,
    id: 2,
    UNDERLYING: "0x9C58BAcC331c9aa871AFD802DB6379a98e80CEdb",
    A_TOKEN: "0xA1Fa064A85266E2Ca82DEe5C5CcEC84DF445760e",
    V_TOKEN: "0xBc59E99198DbA71985A66E1713cC89FFEC53f7FC",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x22441d81416430A54336aB28765abd31a792Ad37",
    STATIC_A_TOKEN: "0x2D737e2B0e175f05D0904C208d6C4e40da570f65",
    STATA_TOKEN: "0x7c16F0185A26Db0AE7a9377f23BC18ea7ce5d644"
  },
  USDC: {
    decimals: 6,
    id: 3,
    UNDERLYING: "0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83",
    A_TOKEN: "0xc6B7AcA6DE8a6044E0e32d0c841a89244A10D284",
    V_TOKEN: "0x5F6f7B0a87CA3CF3d0b431Ae03EF3305180BFf4d",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x1450C4BE9dd66889CddAB77e5947B6166ACbfAE2",
    STATIC_A_TOKEN: "0x270bA1f35D8b87510D24F693fcCc0da02e6E4EeB"
  },
  WXDAI: {
    decimals: 18,
    id: 4,
    UNDERLYING: "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d",
    A_TOKEN: "0xd0Dd6cEF72143E22cCED4867eb0d5F2328715533",
    V_TOKEN: "0x281963D7471eCdC3A2Bd4503e24e89691cfe420D",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x7443afE82986d7475Cea0c5b04C6F1581fdAce87",
    STATIC_A_TOKEN: "0x7f0EAE87Df30C468E0680c83549D0b3DE7664D4B"
  },
  EURe: {
    decimals: 18,
    id: 5,
    UNDERLYING: "0xcB444e90D8198415266c6a2724b7900fb12FC56E",
    A_TOKEN: "0xEdBC7449a9b594CA4E053D9737EC5Dc4CbCcBfb2",
    V_TOKEN: "0xb96404e475f337A7E98e4a541C9b71309BB66c5A",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0xab70BCB260073d036d1660201e9d5405F5829b7a",
    STATIC_A_TOKEN: "0x8418D17640a74F1614AC3E1826F29e78714488a1",
    STATA_TOKEN: "0x417bc5b940475203A18C2f320a5ba470D6c5E463"
  },
  sDAI: {
    decimals: 18,
    id: 6,
    UNDERLYING: "0xaf204776c7245bF4147c2612BF6e5972Ee483701",
    A_TOKEN: "0x7a5c3860a77a8DC1b225BD46d0fb2ac1C6D191BC",
    V_TOKEN: "0x8Fe06E1D8Aff42Bf6812CacF7854A2249a00bED7",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x620424f393dD413c2F8Dc2980905c4daa3619e61",
    STATIC_A_TOKEN: "0xf3f45960f8dE00D8ED614D445a5a268c6F6Dec4f"
  },
  USDCe: {
    decimals: 6,
    id: 7,
    UNDERLYING: "0x2a22f9c3b484c3629090FeED35F17Ff8F88f76F0",
    A_TOKEN: "0xC0333cb85B59a788d8C7CAe5e1Fd6E229A3E5a65",
    V_TOKEN: "0x37B9Ad6b5DC8Ad977AD716e92F49e9D200e58431",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x1450C4BE9dd66889CddAB77e5947B6166ACbfAE2",
    STATIC_A_TOKEN: "0xf0E7eC247b918311afa054E0AEdb99d74c31b809",
    STATA_TOKEN: "0x51350d88c1bd32Cc6A79368c9Fb70373Fb71F375"
  },
  GHO: {
    decimals: 18,
    id: 8,
    UNDERLYING: "0xfc421aD3C883Bf9E7C4f42dE845C4e4405799e73",
    A_TOKEN: "0x3FdCeC11B4f15C79d483Aedc56F37D302837Cf4d",
    V_TOKEN: "0x2766EEFE0311Bf7421cC30155b03d210BCE30dF8",
    INTEREST_RATE_STRATEGY: "0x4cE496f0a390745102540faF041EF92FfD588b44",
    ORACLE: "0x360d8aa8F6b09B7BC57aF34db2Eb84dD87bf4d12",
    STATA_TOKEN: "0x58d9AcAC48A4077E4909181c48DEcD00E5ba5DE4"
  }
};
var E_MODES = {
  "1": {
    label: "ETH correlated",
    collateralBitmap: "3",
    collateralAssets: [
      "0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1",
      "0x6C76971f98945AE98dD7d4DFcA8711ebea946eA6"
    ],
    borrowableBitmap: "1",
    borrowableAssets: ["0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1"],
    ltv: 9e3,
    liquidationThreshold: 9300,
    liquidationBonus: 10100
  },
  "2": {
    label: "sDAI / EURe",
    collateralBitmap: "64",
    collateralAssets: ["0xaf204776c7245bF4147c2612BF6e5972Ee483701"],
    borrowableBitmap: "32",
    borrowableAssets: ["0xcB444e90D8198415266c6a2724b7900fb12FC56E"],
    ltv: 8500,
    liquidationThreshold: 8750,
    liquidationBonus: 10500
  },
  "3": {
    label: "sDAI/USDCe",
    collateralBitmap: "64",
    collateralAssets: ["0xaf204776c7245bF4147c2612BF6e5972Ee483701"],
    borrowableBitmap: "128",
    borrowableAssets: ["0x2a22f9c3b484c3629090FeED35F17Ff8F88f76F0"],
    ltv: 9e3,
    liquidationThreshold: 9200,
    liquidationBonus: 10400
  }
};
var EXTERNAL_LIBRARIES = {
  FLASHLOAN_LOGIC: "0x5e84CEe2afb7B37d2AB14722C39A7c1C26F5B0BB",
  BORROW_LOGIC: "0xFeD9871528E713B5038c4c44BbE7a315f56cAdc6",
  E_MODE_LOGIC: "0xD1bddC05A3BB5A7907d82A1b4F1E21dBCE69c3d5",
  LIQUIDATION_LOGIC: "0x36Ae486289bB807C3C79A1427b9c3D934294ef43",
  POOL_LOGIC: "0xE51B69e5722Bf547866A4d7Bc190c6e81b626806",
  SUPPLY_LOGIC: "0x034Fd14b9Ae6bB066a1F9f85A55e990b0b25c168"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN,
  ACL_MANAGER,
  ASSETS,
  CHAIN_ID,
  CLINIC_STEWARD,
  COLLECTOR,
  CONFIG_ENGINE,
  DEBT_SWAP_ADAPTER,
  DEFAULT_A_TOKEN_IMPL,
  DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL,
  DUST_BIN,
  EDGE_INJECTOR_CAPS,
  EDGE_RISK_STEWARD_CAPS,
  EMISSION_MANAGER,
  EXTERNAL_LIBRARIES,
  E_MODES,
  LEGACY_STATIC_A_TOKEN_FACTORY,
  ORACLE,
  POOL,
  POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR,
  POOL_CONFIGURATOR_IMPL,
  POOL_EXPOSURE_STEWARD,
  POOL_IMPL,
  REPAY_WITH_COLLATERAL_ADAPTER,
  RISK_STEWARD,
  STATA_FACTORY,
  SWAP_COLLATERAL_ADAPTER,
  UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY,
  WITHDRAW_SWAP_ADAPTER
});
