// @bgd-labs/aave-address-book/dist/AaveV3Scroll.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/AaveV3Scroll.ts
var AaveV3Scroll_exports = {};
__export(AaveV3Scroll_exports, {
  AAVE_PROTOCOL_DATA_PROVIDER: () => AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN: () => ACL_ADMIN,
  ACL_MANAGER: () => ACL_MANAGER,
  ASSETS: () => ASSETS,
  CHAIN_ID: () => CHAIN_ID,
  CLINIC_STEWARD: () => CLINIC_STEWARD,
  COLLECTOR: () => COLLECTOR,
  CONFIG_ENGINE: () => CONFIG_ENGINE,
  DEFAULT_A_TOKEN_IMPL: () => DEFAULT_A_TOKEN_IMPL,
  DEFAULT_INCENTIVES_CONTROLLER: () => DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL: () => DEFAULT_VARIABLE_DEBT_TOKEN_IMPL,
  DUST_BIN: () => DUST_BIN,
  EMISSION_MANAGER: () => EMISSION_MANAGER,
  EXTERNAL_LIBRARIES: () => EXTERNAL_LIBRARIES,
  E_MODES: () => E_MODES,
  L2_ENCODER: () => L2_ENCODER,
  ORACLE: () => ORACLE,
  POOL: () => POOL,
  POOL_ADDRESSES_PROVIDER: () => POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY: () => POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR: () => POOL_CONFIGURATOR,
  POOL_CONFIGURATOR_IMPL: () => POOL_CONFIGURATOR_IMPL,
  POOL_EXPOSURE_STEWARD: () => POOL_EXPOSURE_STEWARD,
  POOL_IMPL: () => POOL_IMPL,
  PRICE_ORACLE_SENTINEL: () => PRICE_ORACLE_SENTINEL,
  RISK_STEWARD: () => RISK_STEWARD,
  STATA_FACTORY: () => STATA_FACTORY,
  UI_INCENTIVE_DATA_PROVIDER: () => UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER: () => UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER: () => WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY: () => WETH_GATEWAY
});
module.exports = __toCommonJS(AaveV3Scroll_exports);
var POOL_ADDRESSES_PROVIDER = "0x69850D0B276776781C063771b161bd8894BCdD04";
var POOL = "0x11fCfe756c05AD438e312a7fd934381537D3cFfe";
var POOL_CONFIGURATOR = "0x32BCab42a2bb5AC577D24b425D46d8b8e0Df9b7f";
var ORACLE = "0x04421D8C506E2fA2371a08EfAaBf791F624054F3";
var PRICE_ORACLE_SENTINEL = "0xfD0Ba55775C1e53f50736FA5528d8aa45FBcA391";
var ACL_ADMIN = "0xc1ABF87FfAdf4908f4eC8dc54A25DCFEabAE4A24";
var ACL_MANAGER = "0x7633F981D87dC6307227de9383D2ce7243158081";
var AAVE_PROTOCOL_DATA_PROVIDER = "0xBEa2B648f05887eCbF1d115da8b7E4A317975A51";
var POOL_IMPL = "0xA401e87AB4AB647165D78C2dbBD8D066b4b76D5b";
var POOL_CONFIGURATOR_IMPL = "0x5793FE4de34532F162B4e207aF872729880ec2b6";
var DEFAULT_INCENTIVES_CONTROLLER = "0xa3f3100C4f1D0624DB9DB97b40C13885Ce297799";
var EMISSION_MANAGER = "0x6091546836DAe0487A50E300da3F237727441D90";
var COLLECTOR = "0x90eB541e1a431D8a30ED85A77675D1F001128cb5";
var DEFAULT_A_TOKEN_IMPL = "0x2016ced44D605CE7eF4ade24d74F3a281d8B6C70";
var DEFAULT_VARIABLE_DEBT_TOKEN_IMPL = "0x87CeE33AFE2585de36B622bc7bfC8539864E5B13";
var L2_ENCODER = "0x8714E5ED2d8edD4E88eFf66637C3FE8eCf2B8C40";
var RISK_STEWARD = "0x0451f67bA61966C346daBAbB50a30Cc6A9A67C69";
var CONFIG_ENGINE = "0xB076204f0E181A224e6eC1eA61Cb5976CBCfFF05";
var POOL_ADDRESSES_PROVIDER_REGISTRY = "0xFBedc64AeE24921cb43004312B9eF367a4162b57";
var UI_INCENTIVE_DATA_PROVIDER = "0x5d4D4007A4c6336550DdAa2a7c0d5e7972eebd16";
var UI_POOL_DATA_PROVIDER = "0x5598BbFA2f4fE8151f45bBA0a3edE1b54B51a0a9";
var WALLET_BALANCE_PROVIDER = "0xE51642875Af07Fea1B03d821E41a9DB24bC4447E";
var WETH_GATEWAY = "0xE79Ca44408Dae5a57eA2a9594532f1E84d2edAa4";
var DUST_BIN = "0xD1aE9Da215B06910F246b13fbCbd74D9657AF365";
var CLINIC_STEWARD = "0x7616355B00c9d1D2Faf3937Dacb5B9aA1e6755de";
var POOL_EXPOSURE_STEWARD = "0xf4c8A85fee25f055795E5DcDF39b447aD5761aC7";
var STATA_FACTORY = "0x01cfB64B99f717d791260ECb502e675d6E8Cf522";
var CHAIN_ID = 534352;
var ASSETS = {
  WETH: {
    decimals: 18,
    id: 0,
    UNDERLYING: "0x5300000000000000000000000000000000000004",
    A_TOKEN: "0xf301805bE1Df81102C957f6d4Ce29d2B8c056B2a",
    V_TOKEN: "0xfD7344CeB1Df9Cf238EcD667f4A6F99c6Ef44a56",
    INTEREST_RATE_STRATEGY: "0xC37353E5766164D8654D3CB395acfDcA4c2E7Ddc",
    ORACLE: "0x6bF14CB0A831078629D993FDeBcB182b21A8774C"
  },
  USDC: {
    decimals: 6,
    id: 1,
    UNDERLYING: "0x06eFdBFf2a14a7c8E15944D1F4A48F9F95F663A4",
    A_TOKEN: "0x1D738a3436A8C49CefFbaB7fbF04B660fb528CbD",
    V_TOKEN: "0x3d2E209af5BFa79297C88D6b57F89d792F6E28EE",
    INTEREST_RATE_STRATEGY: "0xC37353E5766164D8654D3CB395acfDcA4c2E7Ddc",
    ORACLE: "0x1685D81212580DD4cDA287616C2f6F4794927e18"
  },
  wstETH: {
    decimals: 18,
    id: 2,
    UNDERLYING: "0xf610A9dfB7C89644979b4A0f27063E9e7d7Cda32",
    A_TOKEN: "0x5B1322eeb46240b02e20062b8F0F9908d525B09c",
    V_TOKEN: "0x8a035644322129800C3f747f54Db0F4d3c0A2877",
    INTEREST_RATE_STRATEGY: "0xC37353E5766164D8654D3CB395acfDcA4c2E7Ddc",
    ORACLE: "0x4EdAbf45e78363b8Dcd763bBbd05665c6e24975C"
  },
  weETH: {
    decimals: 18,
    id: 3,
    UNDERLYING: "0x01f0a31698C4d065659b9bdC21B3610292a1c506",
    A_TOKEN: "0xd80A5e16DBDC52Bd1C947CEDfA22c562Be9129C8",
    V_TOKEN: "0x009D88C6a6B4CaA240b71C98BA93732e26F2A55A",
    INTEREST_RATE_STRATEGY: "0xC37353E5766164D8654D3CB395acfDcA4c2E7Ddc",
    ORACLE: "0x32f924C0e0F1Abf5D1ff35B05eBc5E844dEdD2A9"
  },
  SCR: {
    decimals: 18,
    id: 4,
    UNDERLYING: "0xd29687c813D741E2F938F4aC377128810E217b1b",
    A_TOKEN: "0x25718130C2a8eb94e2e1FAFB5f1cDd4b459aCf64",
    V_TOKEN: "0xFFbA405BBF25B2e6C454d18165F2fd8786858c6B",
    INTEREST_RATE_STRATEGY: "0xC37353E5766164D8654D3CB395acfDcA4c2E7Ddc",
    ORACLE: "0x26f6F7C468EE309115d19Aa2055db5A74F8cE7A5"
  }
};
var E_MODES = {
  "1": {
    label: "ETH correlated",
    collateralBitmap: "13",
    collateralAssets: [
      "0x5300000000000000000000000000000000000004",
      "0xf610A9dfB7C89644979b4A0f27063E9e7d7Cda32",
      "0x01f0a31698C4d065659b9bdC21B3610292a1c506"
    ],
    borrowableBitmap: "1",
    borrowableAssets: ["0x5300000000000000000000000000000000000004"],
    ltv: 9e3,
    liquidationThreshold: 9300,
    liquidationBonus: 10100
  }
};
var EXTERNAL_LIBRARIES = {
  FLASHLOAN_LOGIC: "0x5e84CEe2afb7B37d2AB14722C39A7c1C26F5B0BB",
  BORROW_LOGIC: "0xFeD9871528E713B5038c4c44BbE7a315f56cAdc6",
  E_MODE_LOGIC: "0xD1bddC05A3BB5A7907d82A1b4F1E21dBCE69c3d5",
  LIQUIDATION_LOGIC: "0x36Ae486289bB807C3C79A1427b9c3D934294ef43",
  POOL_LOGIC: "0xE51B69e5722Bf547866A4d7Bc190c6e81b626806",
  SUPPLY_LOGIC: "0x034Fd14b9Ae6bB066a1F9f85A55e990b0b25c168"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN,
  ACL_MANAGER,
  ASSETS,
  CHAIN_ID,
  CLINIC_STEWARD,
  COLLECTOR,
  CONFIG_ENGINE,
  DEFAULT_A_TOKEN_IMPL,
  DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL,
  DUST_BIN,
  EMISSION_MANAGER,
  EXTERNAL_LIBRARIES,
  E_MODES,
  L2_ENCODER,
  ORACLE,
  POOL,
  POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR,
  POOL_CONFIGURATOR_IMPL,
  POOL_EXPOSURE_STEWARD,
  POOL_IMPL,
  PRICE_ORACLE_SENTINEL,
  RISK_STEWARD,
  STATA_FACTORY,
  UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY
});
