// @bgd-labs/aave-address-book/dist/AaveAddressBook.mjs
// src/ts/AaveAddressBook.ts
import * as GovernanceV3Ethereum from "./GovernanceV3Ethereum.mjs";
import * as GovernanceV3Arbitrum from "./GovernanceV3Arbitrum.mjs";
import * as GovernanceV3Avalanche from "./GovernanceV3Avalanche.mjs";
import * as GovernanceV3Fuji from "./GovernanceV3Fuji.mjs";
import * as GovernanceV3Optimism from "./GovernanceV3Optimism.mjs";
import * as GovernanceV3Polygon from "./GovernanceV3Polygon.mjs";
import * as GovernanceV3Metis from "./GovernanceV3Metis.mjs";
import * as GovernanceV3Base from "./GovernanceV3Base.mjs";
import * as GovernanceV3BNB from "./GovernanceV3BNB.mjs";
import * as GovernanceV3Gnosis from "./GovernanceV3Gnosis.mjs";
import * as GovernanceV3Scroll from "./GovernanceV3Scroll.mjs";
import * as GovernanceV3ZkSync from "./GovernanceV3ZkSync.mjs";
import * as GovernanceV3Celo from "./GovernanceV3Celo.mjs";
import * as GovernanceV3Linea from "./GovernanceV3Linea.mjs";
import * as GovernanceV3Mantle from "./GovernanceV3Mantle.mjs";
import * as GovernanceV3Sonic from "./GovernanceV3Sonic.mjs";
import * as GovernanceV3Soneium from "./GovernanceV3Soneium.mjs";
import * as GovernanceV3Bob from "./GovernanceV3Bob.mjs";
import * as GovernanceV3Ink from "./GovernanceV3Ink.mjs";
import * as GovernanceV3InkWhitelabel from "./GovernanceV3InkWhitelabel.mjs";
import * as GovernanceV3Plasma from "./GovernanceV3Plasma.mjs";
import * as AaveV1 from "./AaveV1.mjs";
import * as AaveV2EthereumAMM from "./AaveV2EthereumAMM.mjs";
import * as AaveV2EthereumArc from "./AaveV2EthereumArc.mjs";
import * as AaveV2Ethereum from "./AaveV2Ethereum.mjs";
import * as AaveV2Polygon from "./AaveV2Polygon.mjs";
import * as AaveV2Fuji from "./AaveV2Fuji.mjs";
import * as AaveV2Avalanche from "./AaveV2Avalanche.mjs";
import * as AaveV3Ethereum from "./AaveV3Ethereum.mjs";
import * as AaveV3Sepolia from "./AaveV3Sepolia.mjs";
import * as AaveV3Polygon from "./AaveV3Polygon.mjs";
import * as AaveV3Avalanche from "./AaveV3Avalanche.mjs";
import * as AaveV3Fuji from "./AaveV3Fuji.mjs";
import * as AaveV3Base from "./AaveV3Base.mjs";
import * as AaveV3BaseSepolia from "./AaveV3BaseSepolia.mjs";
import * as AaveV3BaseSepoliaLido from "./AaveV3BaseSepoliaLido.mjs";
import * as AaveV3Metis from "./AaveV3Metis.mjs";
import * as AaveV3Gnosis from "./AaveV3Gnosis.mjs";
import * as AaveV3BNB from "./AaveV3BNB.mjs";
import * as AaveV3Arbitrum from "./AaveV3Arbitrum.mjs";
import * as AaveV3ArbitrumSepolia from "./AaveV3ArbitrumSepolia.mjs";
import * as AaveV3Optimism from "./AaveV3Optimism.mjs";
import * as AaveV3OptimismSepolia from "./AaveV3OptimismSepolia.mjs";
import * as AaveV3Scroll from "./AaveV3Scroll.mjs";
import * as AaveV3ScrollSepolia from "./AaveV3ScrollSepolia.mjs";
import * as AaveV3ZkSync from "./AaveV3ZkSync.mjs";
import * as AaveV3EthereumLido from "./AaveV3EthereumLido.mjs";
import * as AaveV3EthereumEtherFi from "./AaveV3EthereumEtherFi.mjs";
import * as AaveV3Linea from "./AaveV3Linea.mjs";
import * as AaveV3Celo from "./AaveV3Celo.mjs";
import * as AaveV3Sonic from "./AaveV3Sonic.mjs";
import * as AaveV3Soneium from "./AaveV3Soneium.mjs";
import * as AaveV3InkWhitelabel from "./AaveV3InkWhitelabel.mjs";
import * as AaveV3Plasma from "./AaveV3Plasma.mjs";
import * as MiscArbitrum from "./MiscArbitrum.mjs";
import * as MiscArbitrumSepolia from "./MiscArbitrumSepolia.mjs";
import * as MiscAvalanche from "./MiscAvalanche.mjs";
import * as MiscFuji from "./MiscFuji.mjs";
import * as MiscBase from "./MiscBase.mjs";
import * as MiscBaseSepolia from "./MiscBaseSepolia.mjs";
import * as MiscEthereum from "./MiscEthereum.mjs";
import * as MiscFantom from "./MiscFantom.mjs";
import * as MiscOptimism from "./MiscOptimism.mjs";
import * as MiscOptimismSepolia from "./MiscOptimismSepolia.mjs";
import * as MiscPolygon from "./MiscPolygon.mjs";
import * as MiscMetis from "./MiscMetis.mjs";
import * as MiscGnosis from "./MiscGnosis.mjs";
import * as MiscBNB from "./MiscBNB.mjs";
import * as MiscCelo from "./MiscCelo.mjs";
import * as MiscScroll from "./MiscScroll.mjs";
import * as MiscSepolia from "./MiscSepolia.mjs";
import * as MiscZkSync from "./MiscZkSync.mjs";
import * as MiscLinea from "./MiscLinea.mjs";
import * as MiscMantle from "./MiscMantle.mjs";
import * as MiscSonic from "./MiscSonic.mjs";
import * as MiscSoneium from "./MiscSoneium.mjs";
import * as MiscBob from "./MiscBob.mjs";
import * as MiscInk from "./MiscInk.mjs";
import * as MiscInkWhitelabel from "./MiscInkWhitelabel.mjs";
import * as MiscPlasma from "./MiscPlasma.mjs";
import * as AaveGovernanceV2 from "./AaveGovernanceV2.mjs";
import * as AaveSafetyModule from "./AaveSafetyModule.mjs";
import * as GhoEthereum from "./GhoEthereum.mjs";
import * as GhoArbitrum from "./GhoArbitrum.mjs";
import * as GhoBase from "./GhoBase.mjs";
import * as GhoAvalanche from "./GhoAvalanche.mjs";
import * as GhoGnosis from "./GhoGnosis.mjs";
import * as GhoInk from "./GhoInk.mjs";
import * as UmbrellaEthereum from "./UmbrellaEthereum.mjs";
import * as UmbrellaBaseSepolia from "./UmbrellaBaseSepolia.mjs";
import * as ChainlinkEthereum from "./ChainlinkEthereum.mjs";
import * as ChainlinkOptimism from "./ChainlinkOptimism.mjs";
import * as ChainlinkBNB from "./ChainlinkBNB.mjs";
import * as ChainlinkGnosis from "./ChainlinkGnosis.mjs";
import * as ChainlinkPolygon from "./ChainlinkPolygon.mjs";
import * as ChainlinkSonic from "./ChainlinkSonic.mjs";
import * as ChainlinkZkSync from "./ChainlinkZkSync.mjs";
import * as ChainlinkMetis from "./ChainlinkMetis.mjs";
import * as ChainlinkSoneium from "./ChainlinkSoneium.mjs";
import * as ChainlinkMantle from "./ChainlinkMantle.mjs";
import * as ChainlinkBase from "./ChainlinkBase.mjs";
import * as ChainlinkArbitrum from "./ChainlinkArbitrum.mjs";
import * as ChainlinkCelo from "./ChainlinkCelo.mjs";
import * as ChainlinkAvalanche from "./ChainlinkAvalanche.mjs";
import * as ChainlinkLinea from "./ChainlinkLinea.mjs";
import * as ChainlinkScroll from "./ChainlinkScroll.mjs";
export {
  AaveGovernanceV2,
  AaveSafetyModule,
  AaveV1,
  AaveV2Avalanche,
  AaveV2Ethereum,
  AaveV2EthereumAMM,
  AaveV2EthereumArc,
  AaveV2Fuji,
  AaveV2Polygon,
  AaveV3Arbitrum,
  AaveV3ArbitrumSepolia,
  AaveV3Avalanche,
  AaveV3BNB,
  AaveV3Base,
  AaveV3BaseSepolia,
  AaveV3BaseSepoliaLido,
  AaveV3Celo,
  AaveV3Ethereum,
  AaveV3EthereumEtherFi,
  AaveV3EthereumLido,
  AaveV3Fuji,
  AaveV3Gnosis,
  AaveV3InkWhitelabel,
  AaveV3Linea,
  AaveV3Metis,
  AaveV3Optimism,
  AaveV3OptimismSepolia,
  AaveV3Plasma,
  AaveV3Polygon,
  AaveV3Scroll,
  AaveV3ScrollSepolia,
  AaveV3Sepolia,
  AaveV3Soneium,
  AaveV3Sonic,
  AaveV3ZkSync,
  ChainlinkArbitrum,
  ChainlinkAvalanche,
  ChainlinkBNB,
  ChainlinkBase,
  ChainlinkCelo,
  ChainlinkEthereum,
  ChainlinkGnosis,
  ChainlinkLinea,
  ChainlinkMantle,
  ChainlinkMetis,
  ChainlinkOptimism,
  ChainlinkPolygon,
  ChainlinkScroll,
  ChainlinkSoneium,
  ChainlinkSonic,
  ChainlinkZkSync,
  GhoArbitrum,
  GhoAvalanche,
  GhoBase,
  GhoEthereum,
  GhoGnosis,
  GhoInk,
  GovernanceV3Arbitrum,
  GovernanceV3Avalanche,
  GovernanceV3BNB,
  GovernanceV3Base,
  GovernanceV3Bob,
  GovernanceV3Celo,
  GovernanceV3Ethereum,
  GovernanceV3Fuji,
  GovernanceV3Gnosis,
  GovernanceV3Ink,
  GovernanceV3InkWhitelabel,
  GovernanceV3Linea,
  GovernanceV3Mantle,
  GovernanceV3Metis,
  GovernanceV3Optimism,
  GovernanceV3Plasma,
  GovernanceV3Polygon,
  GovernanceV3Scroll,
  GovernanceV3Soneium,
  GovernanceV3Sonic,
  GovernanceV3ZkSync,
  MiscArbitrum,
  MiscArbitrumSepolia,
  MiscAvalanche,
  MiscBNB,
  MiscBase,
  MiscBaseSepolia,
  MiscBob,
  MiscCelo,
  MiscEthereum,
  MiscFantom,
  MiscFuji,
  MiscGnosis,
  MiscInk,
  MiscInkWhitelabel,
  MiscLinea,
  MiscMantle,
  MiscMetis,
  MiscOptimism,
  MiscOptimismSepolia,
  MiscPlasma,
  MiscPolygon,
  MiscScroll,
  MiscSepolia,
  MiscSoneium,
  MiscSonic,
  MiscZkSync,
  UmbrellaBaseSepolia,
  UmbrellaEthereum
};
