// @bgd-labs/aave-address-book/dist/AaveV3Sonic.mjs
// src/ts/AaveV3Sonic.ts
var POOL_ADDRESSES_PROVIDER = "0x5C2e738F6E27bCE0F7558051Bf90605dD6176900";
var POOL = "0x5362dBb1e601abF3a4c14c22ffEdA64042E5eAA3";
var POOL_CONFIGURATOR = "0x50c70FEB95aBC1A92FC30b9aCc41Bd349E5dE2f0";
var ORACLE = "0xD63f7658C66B2934Bd234D79D06aEF5290734B30";
var ACL_ADMIN = "0x7b62461a3570c6AC8a9f8330421576e417B71EE7";
var ACL_MANAGER = "0x3a790a47c4d531FD333FAD24f70B0ccb521B3b5A";
var AAVE_PROTOCOL_DATA_PROVIDER = "0xc0a344397cfa89dF1e1d3e4fb330834D789cF2CD";
var POOL_IMPL = "0x400081EB35244820e9923d7AA35c9bf2078dD054";
var POOL_CONFIGURATOR_IMPL = "0x5793FE4de34532F162B4e207aF872729880ec2b6";
var DEFAULT_INCENTIVES_CONTROLLER = "0x24bD6e9ca54F1737467DEf82dCA9702925B3Aa59";
var EMISSION_MANAGER = "0xd3A0A19Cdb7d1615f30988763BEA5f8feCc17a87";
var COLLECTOR = "0x1aB55bBdD5DF0782BBCf73553Af93BC6B29A286B";
var DEFAULT_A_TOKEN_IMPL = "0xb754640eCa11e9F89022e755997F4DA9B3F18EC5";
var DEFAULT_VARIABLE_DEBT_TOKEN_IMPL = "0x03978bb90165B8F32De90E6435b32331537Ae946";
var CONFIG_ENGINE = "0x413830C44A88ab6c4B99FeB1975b8E6795E9deE8";
var RISK_STEWARD = "0xdb93e2712a8B36835078f8D28c70fCC95FD6d37c";
var POOL_ADDRESSES_PROVIDER_REGISTRY = "0x5E5D87858592d211fdb4503F09FB2B5cf805cB51";
var UI_INCENTIVE_DATA_PROVIDER = "0xeC859AED64e87004FA724892D5aB474901825B2D";
var UI_POOL_DATA_PROVIDER = "0x9005A69fE088680827f292e8aE885Be4BE1beb2f";
var WALLET_BALANCE_PROVIDER = "0x6637Ef9c6b41Fa62d69b7E88C5Dd9eF30BbEE945";
var WETH_GATEWAY = "0x061D8e131F26512348ee5FA42e2DF1bA9d6505E9";
var STATA_FACTORY = "0xFeeb6FE430B7523fEF2a38327241eE7153779535";
var DUST_BIN = "0x62aDD9C6B511085C67515ebB799b51c4cf2d786F";
var POOL_EXPOSURE_STEWARD = "0x4A137d528444Cc0346320CFa5C80FFc68087D7F9";
var REPAY_WITH_COLLATERAL_ADAPTER = "0x5598BbFA2f4fE8151f45bBA0a3edE1b54B51a0a9";
var SWAP_COLLATERAL_ADAPTER = "0x78F8Bd884C3D738B74B420540659c82f392820e0";
var DEBT_SWAP_ADAPTER = "0x2E549104c516b8657A7D888494DfbAbD7C70b464";
var WITHDRAW_SWAP_ADAPTER = "0x5d4D4007A4c6336550DdAa2a7c0d5e7972eebd16";
var CHAIN_ID = 146;
var ASSETS = {
  WETH: {
    decimals: 18,
    id: 0,
    UNDERLYING: "0x50c42dEAcD8Fc9773493ED674b675bE577f2634b",
    A_TOKEN: "0xe18Ab82c81E7Eecff32B8A82B1b7d2d23F1EcE96",
    V_TOKEN: "0x07B1adFB7d5795Cf21baE8a77Eceb222F2FafBCE",
    INTEREST_RATE_STRATEGY: "0xdFf435BCcf782f11187D3a4454d96702eD78e092",
    ORACLE: "0x824364077993847f71293B24ccA8567c00c2de11",
    STATA_TOKEN: "0xeB5e9B0ae5bb60274786C747A1A2A798c11271E0"
  },
  USDC: {
    decimals: 6,
    id: 1,
    UNDERLYING: "0x29219dd400f2Bf60E5a23d13Be72B486D4038894",
    A_TOKEN: "0x578Ee1ca3a8E1b54554Da1Bf7C583506C4CD11c6",
    V_TOKEN: "0x2273caBAd63b7D247A6b107E723c803fc49953A0",
    INTEREST_RATE_STRATEGY: "0xdFf435BCcf782f11187D3a4454d96702eD78e092",
    ORACLE: "0x7A8443a2a5D772db7f1E40DeFe32db485108F128",
    STATA_TOKEN: "0x6646248971427B80ce531bdD793e2Eb859347E55"
  },
  wS: {
    decimals: 18,
    id: 2,
    UNDERLYING: "0x039e2fB66102314Ce7b64Ce5Ce3E5183bc94aD38",
    A_TOKEN: "0x6C5E14A212c1C3e4Baf6f871ac9B1a969918c131",
    V_TOKEN: "0xF6089B790Fbf8F4812a79a31CFAbeB00B06BA7BD",
    INTEREST_RATE_STRATEGY: "0xdFf435BCcf782f11187D3a4454d96702eD78e092",
    ORACLE: "0xc76dFb89fF298145b417d221B2c747d84952e01d",
    STATA_TOKEN: "0x18B7B8695165290f2767BC63c36D3dFEa4C0F9bB"
  },
  stS: {
    decimals: 18,
    id: 3,
    UNDERLYING: "0xE5DA20F15420aD15DE0fa650600aFc998bbE3955",
    A_TOKEN: "0xeAa74D7F42267eB907092AF4Bc700f667EeD0B8B",
    V_TOKEN: "0x333cFdCB6457C409e4f0C88F3806252bEe5fe425",
    INTEREST_RATE_STRATEGY: "0xdFf435BCcf782f11187D3a4454d96702eD78e092",
    ORACLE: "0x5BA5D5213B47DFE020B1F8d6fB54Db3F74F9ea9a"
  }
};
var E_MODES = {
  "1": {
    label: "stS/wS",
    collateralBitmap: "8",
    collateralAssets: ["0xE5DA20F15420aD15DE0fa650600aFc998bbE3955"],
    borrowableBitmap: "4",
    borrowableAssets: ["0x039e2fB66102314Ce7b64Ce5Ce3E5183bc94aD38"],
    ltv: 8700,
    liquidationThreshold: 9e3,
    liquidationBonus: 10100
  }
};
var EXTERNAL_LIBRARIES = {
  FLASHLOAN_LOGIC: "0x5e84CEe2afb7B37d2AB14722C39A7c1C26F5B0BB",
  BORROW_LOGIC: "0xFeD9871528E713B5038c4c44BbE7a315f56cAdc6",
  E_MODE_LOGIC: "0xD1bddC05A3BB5A7907d82A1b4F1E21dBCE69c3d5",
  LIQUIDATION_LOGIC: "0x36Ae486289bB807C3C79A1427b9c3D934294ef43",
  POOL_LOGIC: "0xE51B69e5722Bf547866A4d7Bc190c6e81b626806",
  SUPPLY_LOGIC: "0x034Fd14b9Ae6bB066a1F9f85A55e990b0b25c168"
};
export {
  AAVE_PROTOCOL_DATA_PROVIDER,
  ACL_ADMIN,
  ACL_MANAGER,
  ASSETS,
  CHAIN_ID,
  COLLECTOR,
  CONFIG_ENGINE,
  DEBT_SWAP_ADAPTER,
  DEFAULT_A_TOKEN_IMPL,
  DEFAULT_INCENTIVES_CONTROLLER,
  DEFAULT_VARIABLE_DEBT_TOKEN_IMPL,
  DUST_BIN,
  EMISSION_MANAGER,
  EXTERNAL_LIBRARIES,
  E_MODES,
  ORACLE,
  POOL,
  POOL_ADDRESSES_PROVIDER,
  POOL_ADDRESSES_PROVIDER_REGISTRY,
  POOL_CONFIGURATOR,
  POOL_CONFIGURATOR_IMPL,
  POOL_EXPOSURE_STEWARD,
  POOL_IMPL,
  REPAY_WITH_COLLATERAL_ADAPTER,
  RISK_STEWARD,
  STATA_FACTORY,
  SWAP_COLLATERAL_ADAPTER,
  UI_INCENTIVE_DATA_PROVIDER,
  UI_POOL_DATA_PROVIDER,
  WALLET_BALANCE_PROVIDER,
  WETH_GATEWAY,
  WITHDRAW_SWAP_ADAPTER
};
