// @bgd-labs/aave-address-book/dist/AaveAddressBook.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/AaveAddressBook.ts
var AaveAddressBook_exports = {};
__export(AaveAddressBook_exports, {
  AaveGovernanceV2: () => AaveGovernanceV2,
  AaveSafetyModule: () => AaveSafetyModule,
  AaveV1: () => AaveV1,
  AaveV2Avalanche: () => AaveV2Avalanche,
  AaveV2Ethereum: () => AaveV2Ethereum,
  AaveV2EthereumAMM: () => AaveV2EthereumAMM,
  AaveV2EthereumArc: () => AaveV2EthereumArc,
  AaveV2Fuji: () => AaveV2Fuji,
  AaveV2Polygon: () => AaveV2Polygon,
  AaveV3Arbitrum: () => AaveV3Arbitrum,
  AaveV3ArbitrumSepolia: () => AaveV3ArbitrumSepolia,
  AaveV3Avalanche: () => AaveV3Avalanche,
  AaveV3BNB: () => AaveV3BNB,
  AaveV3Base: () => AaveV3Base,
  AaveV3BaseSepolia: () => AaveV3BaseSepolia,
  AaveV3BaseSepoliaLido: () => AaveV3BaseSepoliaLido,
  AaveV3Celo: () => AaveV3Celo,
  AaveV3Ethereum: () => AaveV3Ethereum,
  AaveV3EthereumEtherFi: () => AaveV3EthereumEtherFi,
  AaveV3EthereumLido: () => AaveV3EthereumLido,
  AaveV3Fuji: () => AaveV3Fuji,
  AaveV3Gnosis: () => AaveV3Gnosis,
  AaveV3InkWhitelabel: () => AaveV3InkWhitelabel,
  AaveV3Linea: () => AaveV3Linea,
  AaveV3Metis: () => AaveV3Metis,
  AaveV3Optimism: () => AaveV3Optimism,
  AaveV3OptimismSepolia: () => AaveV3OptimismSepolia,
  AaveV3Plasma: () => AaveV3Plasma,
  AaveV3Polygon: () => AaveV3Polygon,
  AaveV3Scroll: () => AaveV3Scroll,
  AaveV3ScrollSepolia: () => AaveV3ScrollSepolia,
  AaveV3Sepolia: () => AaveV3Sepolia,
  AaveV3Soneium: () => AaveV3Soneium,
  AaveV3Sonic: () => AaveV3Sonic,
  AaveV3ZkSync: () => AaveV3ZkSync,
  ChainlinkArbitrum: () => ChainlinkArbitrum,
  ChainlinkAvalanche: () => ChainlinkAvalanche,
  ChainlinkBNB: () => ChainlinkBNB,
  ChainlinkBase: () => ChainlinkBase,
  ChainlinkCelo: () => ChainlinkCelo,
  ChainlinkEthereum: () => ChainlinkEthereum,
  ChainlinkGnosis: () => ChainlinkGnosis,
  ChainlinkLinea: () => ChainlinkLinea,
  ChainlinkMantle: () => ChainlinkMantle,
  ChainlinkMetis: () => ChainlinkMetis,
  ChainlinkOptimism: () => ChainlinkOptimism,
  ChainlinkPolygon: () => ChainlinkPolygon,
  ChainlinkScroll: () => ChainlinkScroll,
  ChainlinkSoneium: () => ChainlinkSoneium,
  ChainlinkSonic: () => ChainlinkSonic,
  ChainlinkZkSync: () => ChainlinkZkSync,
  GhoArbitrum: () => GhoArbitrum,
  GhoAvalanche: () => GhoAvalanche,
  GhoBase: () => GhoBase,
  GhoEthereum: () => GhoEthereum,
  GhoGnosis: () => GhoGnosis,
  GhoInk: () => GhoInk,
  GovernanceV3Arbitrum: () => GovernanceV3Arbitrum,
  GovernanceV3Avalanche: () => GovernanceV3Avalanche,
  GovernanceV3BNB: () => GovernanceV3BNB,
  GovernanceV3Base: () => GovernanceV3Base,
  GovernanceV3Bob: () => GovernanceV3Bob,
  GovernanceV3Celo: () => GovernanceV3Celo,
  GovernanceV3Ethereum: () => GovernanceV3Ethereum,
  GovernanceV3Fuji: () => GovernanceV3Fuji,
  GovernanceV3Gnosis: () => GovernanceV3Gnosis,
  GovernanceV3Ink: () => GovernanceV3Ink,
  GovernanceV3InkWhitelabel: () => GovernanceV3InkWhitelabel,
  GovernanceV3Linea: () => GovernanceV3Linea,
  GovernanceV3Mantle: () => GovernanceV3Mantle,
  GovernanceV3Metis: () => GovernanceV3Metis,
  GovernanceV3Optimism: () => GovernanceV3Optimism,
  GovernanceV3Plasma: () => GovernanceV3Plasma,
  GovernanceV3Polygon: () => GovernanceV3Polygon,
  GovernanceV3Scroll: () => GovernanceV3Scroll,
  GovernanceV3Soneium: () => GovernanceV3Soneium,
  GovernanceV3Sonic: () => GovernanceV3Sonic,
  GovernanceV3ZkSync: () => GovernanceV3ZkSync,
  MiscArbitrum: () => MiscArbitrum,
  MiscArbitrumSepolia: () => MiscArbitrumSepolia,
  MiscAvalanche: () => MiscAvalanche,
  MiscBNB: () => MiscBNB,
  MiscBase: () => MiscBase,
  MiscBaseSepolia: () => MiscBaseSepolia,
  MiscBob: () => MiscBob,
  MiscCelo: () => MiscCelo,
  MiscEthereum: () => MiscEthereum,
  MiscFantom: () => MiscFantom,
  MiscFuji: () => MiscFuji,
  MiscGnosis: () => MiscGnosis,
  MiscInk: () => MiscInk,
  MiscInkWhitelabel: () => MiscInkWhitelabel,
  MiscLinea: () => MiscLinea,
  MiscMantle: () => MiscMantle,
  MiscMetis: () => MiscMetis,
  MiscOptimism: () => MiscOptimism,
  MiscOptimismSepolia: () => MiscOptimismSepolia,
  MiscPlasma: () => MiscPlasma,
  MiscPolygon: () => MiscPolygon,
  MiscScroll: () => MiscScroll,
  MiscSepolia: () => MiscSepolia,
  MiscSoneium: () => MiscSoneium,
  MiscSonic: () => MiscSonic,
  MiscZkSync: () => MiscZkSync,
  UmbrellaBaseSepolia: () => UmbrellaBaseSepolia,
  UmbrellaEthereum: () => UmbrellaEthereum
});
module.exports = __toCommonJS(AaveAddressBook_exports);
var GovernanceV3Ethereum = __toESM(require("./GovernanceV3Ethereum.js"));
var GovernanceV3Arbitrum = __toESM(require("./GovernanceV3Arbitrum.js"));
var GovernanceV3Avalanche = __toESM(require("./GovernanceV3Avalanche.js"));
var GovernanceV3Fuji = __toESM(require("./GovernanceV3Fuji.js"));
var GovernanceV3Optimism = __toESM(require("./GovernanceV3Optimism.js"));
var GovernanceV3Polygon = __toESM(require("./GovernanceV3Polygon.js"));
var GovernanceV3Metis = __toESM(require("./GovernanceV3Metis.js"));
var GovernanceV3Base = __toESM(require("./GovernanceV3Base.js"));
var GovernanceV3BNB = __toESM(require("./GovernanceV3BNB.js"));
var GovernanceV3Gnosis = __toESM(require("./GovernanceV3Gnosis.js"));
var GovernanceV3Scroll = __toESM(require("./GovernanceV3Scroll.js"));
var GovernanceV3ZkSync = __toESM(require("./GovernanceV3ZkSync.js"));
var GovernanceV3Celo = __toESM(require("./GovernanceV3Celo.js"));
var GovernanceV3Linea = __toESM(require("./GovernanceV3Linea.js"));
var GovernanceV3Mantle = __toESM(require("./GovernanceV3Mantle.js"));
var GovernanceV3Sonic = __toESM(require("./GovernanceV3Sonic.js"));
var GovernanceV3Soneium = __toESM(require("./GovernanceV3Soneium.js"));
var GovernanceV3Bob = __toESM(require("./GovernanceV3Bob.js"));
var GovernanceV3Ink = __toESM(require("./GovernanceV3Ink.js"));
var GovernanceV3InkWhitelabel = __toESM(require("./GovernanceV3InkWhitelabel.js"));
var GovernanceV3Plasma = __toESM(require("./GovernanceV3Plasma.js"));
var AaveV1 = __toESM(require("./AaveV1.js"));
var AaveV2EthereumAMM = __toESM(require("./AaveV2EthereumAMM.js"));
var AaveV2EthereumArc = __toESM(require("./AaveV2EthereumArc.js"));
var AaveV2Ethereum = __toESM(require("./AaveV2Ethereum.js"));
var AaveV2Polygon = __toESM(require("./AaveV2Polygon.js"));
var AaveV2Fuji = __toESM(require("./AaveV2Fuji.js"));
var AaveV2Avalanche = __toESM(require("./AaveV2Avalanche.js"));
var AaveV3Ethereum = __toESM(require("./AaveV3Ethereum.js"));
var AaveV3Sepolia = __toESM(require("./AaveV3Sepolia.js"));
var AaveV3Polygon = __toESM(require("./AaveV3Polygon.js"));
var AaveV3Avalanche = __toESM(require("./AaveV3Avalanche.js"));
var AaveV3Fuji = __toESM(require("./AaveV3Fuji.js"));
var AaveV3Base = __toESM(require("./AaveV3Base.js"));
var AaveV3BaseSepolia = __toESM(require("./AaveV3BaseSepolia.js"));
var AaveV3BaseSepoliaLido = __toESM(require("./AaveV3BaseSepoliaLido.js"));
var AaveV3Metis = __toESM(require("./AaveV3Metis.js"));
var AaveV3Gnosis = __toESM(require("./AaveV3Gnosis.js"));
var AaveV3BNB = __toESM(require("./AaveV3BNB.js"));
var AaveV3Arbitrum = __toESM(require("./AaveV3Arbitrum.js"));
var AaveV3ArbitrumSepolia = __toESM(require("./AaveV3ArbitrumSepolia.js"));
var AaveV3Optimism = __toESM(require("./AaveV3Optimism.js"));
var AaveV3OptimismSepolia = __toESM(require("./AaveV3OptimismSepolia.js"));
var AaveV3Scroll = __toESM(require("./AaveV3Scroll.js"));
var AaveV3ScrollSepolia = __toESM(require("./AaveV3ScrollSepolia.js"));
var AaveV3ZkSync = __toESM(require("./AaveV3ZkSync.js"));
var AaveV3EthereumLido = __toESM(require("./AaveV3EthereumLido.js"));
var AaveV3EthereumEtherFi = __toESM(require("./AaveV3EthereumEtherFi.js"));
var AaveV3Linea = __toESM(require("./AaveV3Linea.js"));
var AaveV3Celo = __toESM(require("./AaveV3Celo.js"));
var AaveV3Sonic = __toESM(require("./AaveV3Sonic.js"));
var AaveV3Soneium = __toESM(require("./AaveV3Soneium.js"));
var AaveV3InkWhitelabel = __toESM(require("./AaveV3InkWhitelabel.js"));
var AaveV3Plasma = __toESM(require("./AaveV3Plasma.js"));
var MiscArbitrum = __toESM(require("./MiscArbitrum.js"));
var MiscArbitrumSepolia = __toESM(require("./MiscArbitrumSepolia.js"));
var MiscAvalanche = __toESM(require("./MiscAvalanche.js"));
var MiscFuji = __toESM(require("./MiscFuji.js"));
var MiscBase = __toESM(require("./MiscBase.js"));
var MiscBaseSepolia = __toESM(require("./MiscBaseSepolia.js"));
var MiscEthereum = __toESM(require("./MiscEthereum.js"));
var MiscFantom = __toESM(require("./MiscFantom.js"));
var MiscOptimism = __toESM(require("./MiscOptimism.js"));
var MiscOptimismSepolia = __toESM(require("./MiscOptimismSepolia.js"));
var MiscPolygon = __toESM(require("./MiscPolygon.js"));
var MiscMetis = __toESM(require("./MiscMetis.js"));
var MiscGnosis = __toESM(require("./MiscGnosis.js"));
var MiscBNB = __toESM(require("./MiscBNB.js"));
var MiscCelo = __toESM(require("./MiscCelo.js"));
var MiscScroll = __toESM(require("./MiscScroll.js"));
var MiscSepolia = __toESM(require("./MiscSepolia.js"));
var MiscZkSync = __toESM(require("./MiscZkSync.js"));
var MiscLinea = __toESM(require("./MiscLinea.js"));
var MiscMantle = __toESM(require("./MiscMantle.js"));
var MiscSonic = __toESM(require("./MiscSonic.js"));
var MiscSoneium = __toESM(require("./MiscSoneium.js"));
var MiscBob = __toESM(require("./MiscBob.js"));
var MiscInk = __toESM(require("./MiscInk.js"));
var MiscInkWhitelabel = __toESM(require("./MiscInkWhitelabel.js"));
var MiscPlasma = __toESM(require("./MiscPlasma.js"));
var AaveGovernanceV2 = __toESM(require("./AaveGovernanceV2.js"));
var AaveSafetyModule = __toESM(require("./AaveSafetyModule.js"));
var GhoEthereum = __toESM(require("./GhoEthereum.js"));
var GhoArbitrum = __toESM(require("./GhoArbitrum.js"));
var GhoBase = __toESM(require("./GhoBase.js"));
var GhoAvalanche = __toESM(require("./GhoAvalanche.js"));
var GhoGnosis = __toESM(require("./GhoGnosis.js"));
var GhoInk = __toESM(require("./GhoInk.js"));
var UmbrellaEthereum = __toESM(require("./UmbrellaEthereum.js"));
var UmbrellaBaseSepolia = __toESM(require("./UmbrellaBaseSepolia.js"));
var ChainlinkEthereum = __toESM(require("./ChainlinkEthereum.js"));
var ChainlinkOptimism = __toESM(require("./ChainlinkOptimism.js"));
var ChainlinkBNB = __toESM(require("./ChainlinkBNB.js"));
var ChainlinkGnosis = __toESM(require("./ChainlinkGnosis.js"));
var ChainlinkPolygon = __toESM(require("./ChainlinkPolygon.js"));
var ChainlinkSonic = __toESM(require("./ChainlinkSonic.js"));
var ChainlinkZkSync = __toESM(require("./ChainlinkZkSync.js"));
var ChainlinkMetis = __toESM(require("./ChainlinkMetis.js"));
var ChainlinkSoneium = __toESM(require("./ChainlinkSoneium.js"));
var ChainlinkMantle = __toESM(require("./ChainlinkMantle.js"));
var ChainlinkBase = __toESM(require("./ChainlinkBase.js"));
var ChainlinkArbitrum = __toESM(require("./ChainlinkArbitrum.js"));
var ChainlinkCelo = __toESM(require("./ChainlinkCelo.js"));
var ChainlinkAvalanche = __toESM(require("./ChainlinkAvalanche.js"));
var ChainlinkLinea = __toESM(require("./ChainlinkLinea.js"));
var ChainlinkScroll = __toESM(require("./ChainlinkScroll.js"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AaveGovernanceV2,
  AaveSafetyModule,
  AaveV1,
  AaveV2Avalanche,
  AaveV2Ethereum,
  AaveV2EthereumAMM,
  AaveV2EthereumArc,
  AaveV2Fuji,
  AaveV2Polygon,
  AaveV3Arbitrum,
  AaveV3ArbitrumSepolia,
  AaveV3Avalanche,
  AaveV3BNB,
  AaveV3Base,
  AaveV3BaseSepolia,
  AaveV3BaseSepoliaLido,
  AaveV3Celo,
  AaveV3Ethereum,
  AaveV3EthereumEtherFi,
  AaveV3EthereumLido,
  AaveV3Fuji,
  AaveV3Gnosis,
  AaveV3InkWhitelabel,
  AaveV3Linea,
  AaveV3Metis,
  AaveV3Optimism,
  AaveV3OptimismSepolia,
  AaveV3Plasma,
  AaveV3Polygon,
  AaveV3Scroll,
  AaveV3ScrollSepolia,
  AaveV3Sepolia,
  AaveV3Soneium,
  AaveV3Sonic,
  AaveV3ZkSync,
  ChainlinkArbitrum,
  ChainlinkAvalanche,
  ChainlinkBNB,
  ChainlinkBase,
  ChainlinkCelo,
  ChainlinkEthereum,
  ChainlinkGnosis,
  ChainlinkLinea,
  ChainlinkMantle,
  ChainlinkMetis,
  ChainlinkOptimism,
  ChainlinkPolygon,
  ChainlinkScroll,
  ChainlinkSoneium,
  ChainlinkSonic,
  ChainlinkZkSync,
  GhoArbitrum,
  GhoAvalanche,
  GhoBase,
  GhoEthereum,
  GhoGnosis,
  GhoInk,
  GovernanceV3Arbitrum,
  GovernanceV3Avalanche,
  GovernanceV3BNB,
  GovernanceV3Base,
  GovernanceV3Bob,
  GovernanceV3Celo,
  GovernanceV3Ethereum,
  GovernanceV3Fuji,
  GovernanceV3Gnosis,
  GovernanceV3Ink,
  GovernanceV3InkWhitelabel,
  GovernanceV3Linea,
  GovernanceV3Mantle,
  GovernanceV3Metis,
  GovernanceV3Optimism,
  GovernanceV3Plasma,
  GovernanceV3Polygon,
  GovernanceV3Scroll,
  GovernanceV3Soneium,
  GovernanceV3Sonic,
  GovernanceV3ZkSync,
  MiscArbitrum,
  MiscArbitrumSepolia,
  MiscAvalanche,
  MiscBNB,
  MiscBase,
  MiscBaseSepolia,
  MiscBob,
  MiscCelo,
  MiscEthereum,
  MiscFantom,
  MiscFuji,
  MiscGnosis,
  MiscInk,
  MiscInkWhitelabel,
  MiscLinea,
  MiscMantle,
  MiscMetis,
  MiscOptimism,
  MiscOptimismSepolia,
  MiscPlasma,
  MiscPolygon,
  MiscScroll,
  MiscSepolia,
  MiscSoneium,
  MiscSonic,
  MiscZkSync,
  UmbrellaBaseSepolia,
  UmbrellaEthereum
});
