// @bgd-labs/aave-address-book/dist/ChainlinkAvalanche.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkAvalanche.ts
var ChainlinkAvalanche_exports = {};
__export(ChainlinkAvalanche_exports, {
  AAVE_Network_Emergency_Count_Avalanche: () => AAVE_Network_Emergency_Count_Avalanche,
  AAVE_USD: () => AAVE_USD,
  AAVEe_Proof_of_Reserves: () => AAVEe_Proof_of_Reserves,
  ADA_USD: () => ADA_USD,
  ALPHA_USD: () => ALPHA_USD,
  ARB_USD: () => ARB_USD,
  AUSD_USD: () => AUSD_USD,
  AVAX_USD: () => AVAX_USD,
  AXS_USD: () => AXS_USD,
  BAT_USD: () => BAT_USD,
  BEAM_USD: () => BEAM_USD,
  BNB_USD: () => BNB_USD,
  BTC_USD: () => BTC_USD,
  BTCb_Proof_of_Reserves: () => BTCb_Proof_of_Reserves,
  CHAIN_ID: () => CHAIN_ID,
  CHF_USD: () => CHF_USD,
  CHZ_USD: () => CHZ_USD,
  COMP_USD: () => COMP_USD,
  COQ_USD: () => COQ_USD,
  CRV_USD: () => CRV_USD,
  CVX_USD: () => CVX_USD,
  CZK_USD: () => CZK_USD,
  Calculated_SAVAX_USD: () => Calculated_SAVAX_USD,
  DAI_USD: () => DAI_USD,
  DAIe_Proof_of_Reserves: () => DAIe_Proof_of_Reserves,
  DOT_USD: () => DOT_USD,
  ETH_USD: () => ETH_USD,
  EURC_USD: () => EURC_USD,
  EUR_USD: () => EUR_USD,
  EmCH_Reserves: () => EmCH_Reserves,
  Exchange_Rate_ggAVAX_AVAX: () => Exchange_Rate_ggAVAX_AVAX,
  FIL_USD: () => FIL_USD,
  FRAX_USD: () => FRAX_USD,
  FTM_USD: () => FTM_USD,
  FXS_USD: () => FXS_USD,
  GHO_USD: () => GHO_USD,
  GMX_USD: () => GMX_USD,
  Ion_Digital_Total_Reserve: () => Ion_Digital_Total_Reserve,
  JOE_USD: () => JOE_USD,
  JPY_USD: () => JPY_USD,
  KNC_USD: () => KNC_USD,
  LINK_AVAX: () => LINK_AVAX,
  LINK_USD: () => LINK_USD,
  LINKe_Proof_of_Reserves: () => LINKe_Proof_of_Reserves,
  MANA_USD: () => MANA_USD,
  MATICX_USD_Calculated: () => MATICX_USD_Calculated,
  MATIC_USD: () => MATIC_USD,
  MIMATIC_USD: () => MIMATIC_USD,
  MIM_USD: () => MIM_USD,
  MKR_USD: () => MKR_USD,
  NEAR_USD: () => NEAR_USD,
  PCE_Price_Index_Level: () => PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate: () => PCE_Price_Index_Percent_Change_Annual_Rate,
  POL_USD: () => POL_USD,
  QI_USD: () => QI_USD,
  Re_Offchain_Reserves: () => Re_Offchain_Reserves,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level: () => Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate: () => Real_GDP_Percent_Change_Annual_Rate,
  SAND_USD: () => SAND_USD,
  SGD_USD: () => SGD_USD,
  SHIB_USD: () => SHIB_USD,
  SNX_USD: () => SNX_USD,
  SPELL_USD: () => SPELL_USD,
  SUSHI_USD: () => SUSHI_USD,
  SolvBTCBBN_SolvBTC_Exchange_Rate: () => SolvBTCBBN_SolvBTC_Exchange_Rate,
  TRY_USD: () => TRY_USD,
  TUSD_USD: () => TUSD_USD,
  UNI_USD: () => UNI_USD,
  USDC_USD: () => USDC_USD,
  USDCe_Proof_of_Reserves: () => USDCe_Proof_of_Reserves,
  USDT_USD: () => USDT_USD,
  USDTe_Proof_of_Reserves: () => USDTe_Proof_of_Reserves,
  UST_USD: () => UST_USD,
  WBTC_USD: () => WBTC_USD,
  WBTCe_Proof_of_Reserves: () => WBTCe_Proof_of_Reserves,
  WETHe_Proof_of_Reserves: () => WETHe_Proof_of_Reserves,
  WOO_ETH: () => WOO_ETH,
  WRSETH_ETH_Exchange_Rate: () => WRSETH_ETH_Exchange_Rate,
  WSTETH_ETH: () => WSTETH_ETH,
  WSTETH_USD: () => WSTETH_USD,
  XAG_USD: () => XAG_USD,
  XAU_USD: () => XAU_USD,
  XAVA_USD: () => XAVA_USD,
  YBTCB_BTC: () => YBTCB_BTC,
  YFI_USD: () => YFI_USD,
  deUSD_USD: () => deUSD_USD,
  fsBTC_Frictionless_Bitcoin_Treasury_Fund_Reserves: () => fsBTC_Frictionless_Bitcoin_Treasury_Fund_Reserves,
  savBTC_avBTC_Exchange_Rate: () => savBTC_avBTC_Exchange_Rate,
  savUSD_vUSD_Exchange_Rate: () => savUSD_vUSD_Exchange_Rate,
  solvBTC_BTC: () => solvBTC_BTC,
  syrupUSDC_USDC_Exchange_Rate: () => syrupUSDC_USDC_Exchange_Rate,
  wstETH_stETH_Exchange_Rate: () => wstETH_stETH_Exchange_Rate,
  xSolvBTC_NAV: () => xSolvBTC_NAV
});
module.exports = __toCommonJS(ChainlinkAvalanche_exports);
var WBTCe_Proof_of_Reserves = "0xebEfEAA58636DF9B20a4fAd78Fad8759e6A20e87";
var SAND_USD = "0x6f2A1D4014FED967172FC7caCf7a6e04Cf02752e";
var savUSD_vUSD_Exchange_Rate = "0x1ff75B09dC8B749252F46ec482c8a4700Eb6C644";
var SNX_USD = "0x01752eAAB988ECb0ceBa2C8FC97c4f1d38Bf246D";
var XAU_USD = "0x1F41EF93dece881Ad0b98082B2d44D3f6F0C515B";
var WSTETH_ETH = "0xCF60B4E089eA1ABA29C01b017b38c2f7D69Eb36B";
var ALPHA_USD = "0x7B0ca9A6D03FE0467A31Ca850f5bcA51e027B3aF";
var SHIB_USD = "0x7Ee401373A92f885a7C3A53CF46dbe661eaFb1C0";
var CVX_USD = "0x52F8026423B5E04FdD9E4b5725B68230b71D019b";
var Exchange_Rate_ggAVAX_AVAX = "0xA5ecED13b233BcF5c58BFbada5247d5B2893cB9a";
var DAIe_Proof_of_Reserves = "0x976D7fAc81A49FA71EF20694a3C56B9eFB93c30B";
var deUSD_USD = "0x7E77D9F06056017334e26bEd92176222169DE6C1";
var QI_USD = "0x36E039e6391A5E7A7267650979fdf613f659be5D";
var MIM_USD = "0x54EdAB30a7134A16a54218AE64C73e1DAf48a8Fb";
var EUR_USD = "0x192f2DBA961Bb0277520C082d6bfa87D5961333E";
var BAT_USD = "0xe89B3CE86D25599D1e615C0f6a353B4572FF868D";
var Re_Offchain_Reserves = "0xc79a363a3f849d8b3F6A1932f748eA9d4fB2f607";
var WOO_ETH = "0xfAa665F5a0e13beea63b6DfF601DD634959690Df";
var CHF_USD = "0xA418573AB5226711c8564Eeb449c3618ABFaf677";
var AAVE_Network_Emergency_Count_Avalanche = "0x41185495Bc8297a65DC46f94001DC7233775EbEe";
var KNC_USD = "0x9df2195dc96e6Ef983B1aAC275649F3f28F82Aa1";
var SGD_USD = "0x05950959B6d876ae0fed1BBe5Caa2d74d8659D59";
var Real_GDP_Percent_Change_Annual_Rate = "0x3Eb06AE1Ef3adE7B529Fcce522611c6A044697cC";
var Real_GDP_Level = "0x665Fa131DE7af7606c785F26e66d07cA43a274D9";
var WBTC_USD = "0x86442E3a98558357d46E6182F4b262f76c4fa26F";
var ADA_USD = "0x69C2703b8F1A85a2EF6aBDd085699a9F909BE053";
var BTCb_Proof_of_Reserves = "0x99311B4bf6D8E3D3B4b9fbdD09a1B0F4Ad8e06E9";
var AUSD_USD = "0x5C2d58627Fbe746f5ea24Ef6D618f09f8e3f0122";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate = "0x273e7833604F74a6f80f6B25568D29DE33570e99";
var XAVA_USD = "0x4Cf57DC9028187b9DAaF773c8ecA941036989238";
var AXS_USD = "0x155835C5755205597d62703a5A0b37e57a26Ee5C";
var FRAX_USD = "0xbBa56eF1565354217a3353a466edB82E8F25b08e";
var FXS_USD = "0x12Af94c3716bbf339Aa26BfD927DDdE63B27D50C";
var WRSETH_ETH_Exchange_Rate = "0xFA38289Fe9f043aD8CCD8e81b28C1D02666D51b6";
var JOE_USD = "0x02D35d3a8aC3e1626d3eE09A78Dd87286F5E8e3a";
var LINKe_Proof_of_Reserves = "0x943cEF1B112Ca9FD7EDaDC9A46477d3812a382b6";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Level = "0x982057Fd0A53d52E558A722ba53Ad95E702Da0dF";
var FTM_USD = "0x2dD517B2f9ba49CedB0573131FD97a5AC19ff648";
var USDC_USD = "0xF096872672F44d6EBA71458D74fe67F9a77a23B9";
var LINK_AVAX = "0x1b8a25F73c9420dD507406C3A3816A276b62f56a";
var SUSHI_USD = "0x449A373A090d8A1e5F74c63Ef831Ceff39E94563";
var CZK_USD = "0x545d17579D0F7b422Cd647B9E6a6FA4b45F6e1C5";
var BEAM_USD = "0x3427232b88Ce4e7d62A03289247eE0cA5324f6ba";
var WSTETH_USD = "0xB5f607A2bb2C950053F4DA1F535180e5902EAD16";
var POL_USD = "0x7B0e7d292d414788B080EfCa58b04B6372789639";
var SPELL_USD = "0x4F3ddF9378a4865cf4f28BE51E10AECb83B7daeE";
var TRY_USD = "0xA61bF273688Ea095b5e4c11f1AF5E763F7aEEE91";
var wstETH_stETH_Exchange_Rate = "0x84FF93CAD57DcD39b21895171D30A7A0e7851C75";
var fsBTC_Frictionless_Bitcoin_Treasury_Fund_Reserves = "0xfD2b540D6c147D0889A1F1B97a2A9aB694992677";
var SolvBTCBBN_SolvBTC_Exchange_Rate = "0xc0990e0989141a5F535a2a24b6Af65618CC83d32";
var FIL_USD = "0x2F194315f122d374a27973e259783d5C864A5bf6";
var GHO_USD = "0x076DE3812BDbdAe1330064fc01Adf7f4EAa123f3";
var solvBTC_BTC = "0x5ff6c083B7B2BD0D61beB7173B3757e1a1e6E1C6";
var GMX_USD = "0x3F968A21647d7ca81Fb8A5b69c0A452701d5DCe8";
var ARB_USD = "0xd99bcAdbE216D82f4B77eC54a99ea1b6bA96549b";
var Ion_Digital_Total_Reserve = "0x0AB119EB3f6820BA43c345e9aAbF1e3dE3C09Ce4";
var DAI_USD = "0x51D7180edA2260cc4F6e4EebB82FEF5c3c2B8300";
var LINK_USD = "0x49ccd9ca821EfEab2b98c60dC60F518E765EDe9a";
var ETH_USD = "0x976B3D034E162d8bD72D6b9C989d545b839003b0";
var MKR_USD = "0x3E54eB0475051401D093702A5DB84EFa1Ab77b14";
var MATICX_USD_Calculated = "0xc43861646Cdc051D04CA0c9dF2265FFcFa0dF6f3";
var syrupUSDC_USDC_Exchange_Rate = "0xC63eD21bb4577Ff5a762591c73D4b0BCCfAD7AEb";
var PCE_Price_Index_Level = "0x962Aba596fe6842D644c51b2F112190e6B6e94B4";
var DOT_USD = "0xD73a74314AcCb53b30cAfDA0cb61c9772B57C4a2";
var AAVEe_Proof_of_Reserves = "0x14C4c668E34c09E1FBA823aD5DB47F60aeBDD4F7";
var UNI_USD = "0x9a1372f9b1B71B3A5a72E092AE67E172dBd7Daaa";
var MIMATIC_USD = "0x5D1F504211c17365CA66353442a74D4435A8b778";
var WETHe_Proof_of_Reserves = "0xDDaf9290D057BfA12d7576e6dADC109421F31948";
var NEAR_USD = "0x7FDE7f51dc2580dd051e17A333E28CDC8176da0A";
var XAG_USD = "0xA771e0D1e9E1eCc07C56CC38240779E54337d682";
var MATIC_USD = "0x1db18D41E4AD2403d9f52b5624031a2D9932Fd73";
var JPY_USD = "0xf8B283aD4d969ECFD70005714DD5910160565b94";
var COMP_USD = "0x9D6AA0AC8c4818433bEA7a74F49C73B57BcEC4Ec";
var xSolvBTC_NAV = "0x55b5dc7d7CDD5d3b2Eb189bf11140839076E5d40";
var TUSD_USD = "0x9Cf3Ef104A973b351B2c032AA6793c3A6F76b448";
var YBTCB_BTC = "0x7f7A07FFDe2ab490dad0C0D139d6c730bbe842cC";
var BNB_USD = "0xBb92195Ec95DE626346eeC8282D53e261dF95241";
var BTC_USD = "0x2779D32d5166BAaa2B2b658333bA7e6Ec0C65743";
var UST_USD = "0xf58B78581c480caFf667C63feDd564eCF01Ef86b";
var Calculated_SAVAX_USD = "0x2854Ca10a54800e15A2a25cFa52567166434Ff0a";
var COQ_USD = "0x5B4712ce553E94d9b22bA3CfA10CB6F32fb828E0";
var MANA_USD = "0x774f067083f23cAB866310489419C884a6Dc00a8";
var AAVE_USD = "0x3CA13391E9fb38a75330fb28f8cc2eB3D9ceceED";
var YFI_USD = "0x28043B1Ebd41860B93EC1F1eC19560760B6dB556";
var AVAX_USD = "0x0A77230d17318075983913bC2145DB16C7366156";
var EURC_USD = "0x3368310bC4AeE5D96486A73bae8E6b49FcDE62D3";
var USDTe_Proof_of_Reserves = "0x94D8c2548018C27F1aa078A23C4158206bE1CC72";
var USDT_USD = "0xEBE676ee90Fe1112671f19b6B7459bC678B67e8a";
var CRV_USD = "0x7CF8A6090A9053B01F3DF4D4e6CfEdd8c90d9027";
var savBTC_avBTC_Exchange_Rate = "0x80091AEc326bE58223620D827fd5189FaC8b43C0";
var EmCH_Reserves = "0x0d2807dc7FA52d3B38be564B64a2b37753C49AdD";
var PCE_Price_Index_Percent_Change_Annual_Rate = "0x9A9587Da4Fd5DD3C9AD47C12aFeec5073C8cFc2a";
var USDCe_Proof_of_Reserves = "0x63769951E4cfDbDC653dD9BBde63D2Ce0746e5F2";
var CHZ_USD = "0xC4D7270aCc921DE5A17452437257f075C1298eB3";
var CHAIN_ID = 43114;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_Network_Emergency_Count_Avalanche,
  AAVE_USD,
  AAVEe_Proof_of_Reserves,
  ADA_USD,
  ALPHA_USD,
  ARB_USD,
  AUSD_USD,
  AVAX_USD,
  AXS_USD,
  BAT_USD,
  BEAM_USD,
  BNB_USD,
  BTC_USD,
  BTCb_Proof_of_Reserves,
  CHAIN_ID,
  CHF_USD,
  CHZ_USD,
  COMP_USD,
  COQ_USD,
  CRV_USD,
  CVX_USD,
  CZK_USD,
  Calculated_SAVAX_USD,
  DAI_USD,
  DAIe_Proof_of_Reserves,
  DOT_USD,
  ETH_USD,
  EURC_USD,
  EUR_USD,
  EmCH_Reserves,
  Exchange_Rate_ggAVAX_AVAX,
  FIL_USD,
  FRAX_USD,
  FTM_USD,
  FXS_USD,
  GHO_USD,
  GMX_USD,
  Ion_Digital_Total_Reserve,
  JOE_USD,
  JPY_USD,
  KNC_USD,
  LINK_AVAX,
  LINK_USD,
  LINKe_Proof_of_Reserves,
  MANA_USD,
  MATICX_USD_Calculated,
  MATIC_USD,
  MIMATIC_USD,
  MIM_USD,
  MKR_USD,
  NEAR_USD,
  PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate,
  POL_USD,
  QI_USD,
  Re_Offchain_Reserves,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate,
  SAND_USD,
  SGD_USD,
  SHIB_USD,
  SNX_USD,
  SPELL_USD,
  SUSHI_USD,
  SolvBTCBBN_SolvBTC_Exchange_Rate,
  TRY_USD,
  TUSD_USD,
  UNI_USD,
  USDC_USD,
  USDCe_Proof_of_Reserves,
  USDT_USD,
  USDTe_Proof_of_Reserves,
  UST_USD,
  WBTC_USD,
  WBTCe_Proof_of_Reserves,
  WETHe_Proof_of_Reserves,
  WOO_ETH,
  WRSETH_ETH_Exchange_Rate,
  WSTETH_ETH,
  WSTETH_USD,
  XAG_USD,
  XAU_USD,
  XAVA_USD,
  YBTCB_BTC,
  YFI_USD,
  deUSD_USD,
  fsBTC_Frictionless_Bitcoin_Treasury_Fund_Reserves,
  savBTC_avBTC_Exchange_Rate,
  savUSD_vUSD_Exchange_Rate,
  solvBTC_BTC,
  syrupUSDC_USDC_Exchange_Rate,
  wstETH_stETH_Exchange_Rate,
  xSolvBTC_NAV
});
