// @bgd-labs/aave-address-book/dist/ChainlinkCelo.mjs
// src/ts/ChainlinkCelo.ts
var KES_USD = "0x0826492a24b1dBd1d8fcB4701b38C557CE685e9D";
var CUSD_USD = "0xe38A27BE4E7d866327e09736F3C570F256FFd048";
var BRL_USD = "0xe8EcaF727080968Ed5F6DBB595B91e50eEb9F8B3";
var USDT_USD = "0x5e37AF40A7A344ec9b03CCD34a250F3dA9a20B02";
var LINK_USD = "0x6b6a4c71ec3858A024f3f0Ee44bb0AdcBEd3DcC2";
var ETH_USD = "0x1FcD30A73D67639c1cD89ff5746E7585731c083B";
var CELO_USD = "0x0568fD19986748cEfF3301e55c0eb1E729E0Ab7e";
var GBP_USD = "0xe76FE54dfeD2ce8B4d1AC63c982DfF7CFc92bf82";
var USDC_USD = "0xc7A353BaE210aed958a1A2928b654938EC59DaB2";
var JPY_USD = "0xf323563241BF8B77a2979e9edC1181788A98EcB2";
var KRW_USD = "0x04cE1b865a45cAF8e5218114C9DD03540753C68a";
var CAD_USD = "0x2f6d6cB9e01d63e1a1873BACc5BfD4e7d4e461d1";
var AUD_USD = "0xf2Bd4FAa89f5A360cDf118bccD183307fDBcB6F5";
var GHS_USD = "0x2719B648DB57C5601Bd4cB2ea934Dec6F4262cD8";
var CHF_USD = "0xfd49bFcb3dc4aAa713c25e7d23B14BB39C4B8857";
var EUR_USD = "0x3D207061Dbe8E2473527611BFecB87Ff12b28dDa";
var COP_USD = "0x97b770B0200CCe161907a9cbe0C6B177679f8F7C";
var BTC_USD = "0x128fE88eaa22bFFb868Bb3A584A54C96eE24014b";
var NGN_USD = "0x235e5c8697177931459fA7D19fba7256d29F17DA";
var ZAR_USD = "0x11b7221a0DD025778A95e9E0B87b477522C32E02";
var INR_USD = "0x85d4Ec34339478F73c153710B19f2D5C402dce6F";
var AAVE_Network_Emergency_Count_Celo = "0x91b21900E91CD302EBeD05E45D8f270ddAED944d";
var PHP_USD = "0x4ce8e628Bb82Ea5271908816a6C580A71233a66c";
var XOF_USD = "0x1626095f9548291cA67A6Aa743c30A1BB9380c9d";
var LINK_CELO = "0xBa45f0a1a2fa3FB62a4D6dC135741E2aeb1b14e7";
var EURC_USD = "0x9a48d9b0AF457eF040281A9Af3867bc65522Fecd";
var CHAIN_ID = 42220;
export {
  AAVE_Network_Emergency_Count_Celo,
  AUD_USD,
  BRL_USD,
  BTC_USD,
  CAD_USD,
  CELO_USD,
  CHAIN_ID,
  CHF_USD,
  COP_USD,
  CUSD_USD,
  ETH_USD,
  EURC_USD,
  EUR_USD,
  GBP_USD,
  GHS_USD,
  INR_USD,
  JPY_USD,
  KES_USD,
  KRW_USD,
  LINK_CELO,
  LINK_USD,
  NGN_USD,
  PHP_USD,
  USDC_USD,
  USDT_USD,
  XOF_USD,
  ZAR_USD
};
