// @bgd-labs/aave-address-book/dist/ChainlinkSonic.mjs
// src/ts/ChainlinkSonic.ts
var GHO_USD = "0xb1061E9ab9fC2e9e2f7a036f8e3c287D7395d671";
var DAI_USD = "0x1654Df3d2543717534eE1c38eb9aF5F0407Ec708";
var mooBIFI_USD = "0xc9541134848f525f8eaEA262599a1668A3a9b6eD";
var AAVE_Network_Emergency_Count_Sonic = "0xECB564e91f620fBFb59d0C4A41d7f10aDb0D1934";
var S_USD = "0xc76dFb89fF298145b417d221B2c747d84952e01d";
var SCETH_USD = "0x19A95E6203A0611b6be322c25b63Ec989fFE15c1";
var sfrxUSD_USD = "0xCC8DA199b159f25E7782494Df9140d2ce0Fe048B";
var LINK_USD = "0x26e450ca14D7bF598C89f212010c691434486119";
var LBTC_BTC = "0xA63b1614D17536C22fDB4c1a58023E35d08Cccef";
var BTC_USD = "0x8Bcd59Cb7eEEea8e2Da3080C891609483dae53EF";
var wstkscUSD_stkscUSD_Exchange_Rate = "0xe5bd703E6C4C7679e10D429D87EF4550a9fA6fF4";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Level = "0xD83083D9D899fB185CcaA8EC63F0582A2970373E";
var SWPX_USD = "0x5e7B8000f967ada54eDb8696d21cCFEB7eC36EC6";
var ANON_USD = "0x30FCcFde9857B94B647F9a344c5F4BaCEaB67d72";
var WBTC_USD = "0x61140C09956495F1ce49D28E125Ed4035e1CAE95";
var wstkscBTC_stkscBTC_Exchange_Rate = "0xdA49fdf6a5606Ca7FF180a9B71928Db01a00b0a8";
var xSolvBTC_SolvBTC_Exchange_Rate = "0x21f037F67946A199b1DBF1Cb7DFa21bFde477196";
var YETH_ETH_Exchange_Rate = "0xf0766006e9032295995D60254a5BB035aAF8b88C";
var ETH_USD = "0x824364077993847f71293B24ccA8567c00c2de11";
var X33_USD = "0x0afB9d28170cB7459AEC634a6dD96591315cA42a";
var USDC_USD = "0x55bCa887199d5520B3Ce285D41e6dC10C08716C9";
var beS_S_Exchange_Rate = "0x15643d909F07e4083fCE3e3204F7e1A0A37D52f4";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate = "0xB8deBF17f62838F56e4abc923aef4ad4716C105B";
var SCUSD_USD = "0xACE5e348a341a740004304c2c228Af1A4581920F";
var EUR_USD = "0x0fceF1123FDBEdC89a0189B15D35B7A33B7694c0";
var PCE_Price_Index_Level = "0xba7e8ea9B592c0E6731000b9a186581224a09f69";
var PCE_Price_Index_Percent_Change_Annual_Rate = "0x852aE0B1Af1aAeDB0fC4428B4B24420780976ca8";
var OUSDT_USD = "0x790181e93e9F4Eedb5b864860C12e4d2CffFe73B";
var solvBTC_BTC = "0xCdF021EbD115b1B225801F4056c40F4Ff14A574d";
var vyUSD_USD_Exchange_Rate = "0x419BB1910D6e13DCF8Fcd87E01C5B826493c1845";
var stS_S_Exchange_Rate = "0xf97A2074fCCFDcD2FF567faEbfE235eCF0091c3D";
var wstkscETH_stkscETH_Exchange_Rate = "0x356637474C2124cdD46d8c5fFA17365ad2B42d45";
var Real_GDP_Level = "0xA1d23D3FE32Ef45774D0f2170a7BEf21da09566b";
var YBTC_BTC_Exchange_Rate = "0xA8bed2F8cddD243D894fa629bf1F845A7100ecc8";
var stS_USD = "0xdB17996a889706Bd67771dEa59E0Bf9453aF0CE4";
var xSolvBTC_NAV = "0x5c042362ecB555D9fb554E4ee1821Aa6762D9239";
var wanS_anS_Exchange_Rate = "0x3c6490e4c60BD8246071A014AaCE7A596d392642";
var Real_GDP_Percent_Change_Annual_Rate = "0x791b0Ece46914A7995CEbe6a62d6Be8A3c7b7FAF";
var WRSETH_ETH_Exchange_Rate = "0x7f9506FcBBc9881a316a0CacD0ae6958EE2b5c7d";
var yUSD_USD_Exchange_Rate = "0xB5882f52428d10C99BbF56539C02820dda809Ccf";
var PumpBTC_BTC_Exchange_Rate = "0xd3b87e6F953B1126056b914A8CBE78035fB3E241";
var OS_S = "0x30caC44b395eB969C9CA0d44dF39e6E0aE8f8D94";
var EURC_USD = "0x8f0240dc9C7644D121D370d8ee61FF9F2356521E";
var USD1_USD = "0xB6D739Ba498a0D8d1e92F81345B764C995Cc4882";
var uniBTC_BTC_Exchange_Rate = "0x960BDD1dFD20d7c98fa482D793C3dedD73A113a3";
var scBTC_BTC = "0x3cf9E4860bafe8e29eC8e1bC073436C5519684f4";
var USDT_USD = "0x76F4C040A792aFB7F6dBadC7e30ca3EEa140D216";
var SHADOW_USD = "0xc26dEEfBF72CFC862610Db23830970Fc6663CcC3";
var wOS_OS_Exchange_Rate = "0xEE04fA54F0aDFB6a0d7791EA8236F4BbC5d07E97";
var Calculated_stS_USD = "0xb00C13A3a919259D1A2F47181ECD32C3b7df8636";
var CHAIN_ID = 146;
export {
  AAVE_Network_Emergency_Count_Sonic,
  ANON_USD,
  BTC_USD,
  CHAIN_ID,
  Calculated_stS_USD,
  DAI_USD,
  ETH_USD,
  EURC_USD,
  EUR_USD,
  GHO_USD,
  LBTC_BTC,
  LINK_USD,
  OS_S,
  OUSDT_USD,
  PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate,
  PumpBTC_BTC_Exchange_Rate,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate,
  SCETH_USD,
  SCUSD_USD,
  SHADOW_USD,
  SWPX_USD,
  S_USD,
  USD1_USD,
  USDC_USD,
  USDT_USD,
  WBTC_USD,
  WRSETH_ETH_Exchange_Rate,
  X33_USD,
  YBTC_BTC_Exchange_Rate,
  YETH_ETH_Exchange_Rate,
  beS_S_Exchange_Rate,
  mooBIFI_USD,
  scBTC_BTC,
  sfrxUSD_USD,
  solvBTC_BTC,
  stS_S_Exchange_Rate,
  stS_USD,
  uniBTC_BTC_Exchange_Rate,
  vyUSD_USD_Exchange_Rate,
  wOS_OS_Exchange_Rate,
  wanS_anS_Exchange_Rate,
  wstkscBTC_stkscBTC_Exchange_Rate,
  wstkscETH_stkscETH_Exchange_Rate,
  wstkscUSD_stkscUSD_Exchange_Rate,
  xSolvBTC_NAV,
  xSolvBTC_SolvBTC_Exchange_Rate,
  yUSD_USD_Exchange_Rate
};
