// @bgd-labs/aave-address-book/dist/ChainlinkMetis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkMetis.ts
var ChainlinkMetis_exports = {};
__export(ChainlinkMetis_exports, {
  AAVE_USD: () => AAVE_USD,
  BTC_USD: () => BTC_USD,
  CHAIN_ID: () => CHAIN_ID,
  DAI_USD: () => DAI_USD,
  ETH_USD: () => ETH_USD,
  L2_Sequencer_Uptime_Status_Feed: () => L2_Sequencer_Uptime_Status_Feed,
  LINK_USD: () => LINK_USD,
  METIS_USD: () => METIS_USD,
  MIMATIC_USD: () => MIMATIC_USD,
  USDC_USD: () => USDC_USD,
  USDT_USD: () => USDT_USD,
  frxETH_ETH_Exchange_Rate_High: () => frxETH_ETH_Exchange_Rate_High,
  frxETH_ETH_Exchange_Rate_Low: () => frxETH_ETH_Exchange_Rate_Low,
  sfrxETH_frxETH_Exchange_Rate: () => sfrxETH_frxETH_Exchange_Rate
});
module.exports = __toCommonJS(ChainlinkMetis_exports);
var BTC_USD = "0x51Ed8Fecf96813826F727CaBDF01b3cF6a61373e";
var DAI_USD = "0xe0351cAAE70B5AdBD0107cD5331AD1D79c4c1CA1";
var AAVE_USD = "0x54389e89A5Ec1d4312d5B5C48055d6E56a177BF9";
var USDT_USD = "0x51864b8948Aa5e35aace2BaDaF901D63418A3b9D";
var METIS_USD = "0xD4a5Bb03B5D66d9bf81507379302Ac2C2DFDFa6D";
var USDC_USD = "0x663855969c85F3BE415807250414Ca9129533a5f";
var sfrxETH_frxETH_Exchange_Rate = "0x732bEa04DF4184Fb50B5648bf1A74553f01Fd850";
var L2_Sequencer_Uptime_Status_Feed = "0x58218ea7422255EBE94e56b504035a784b7AA204";
var LINK_USD = "0x4A4F382A2FF9685de9f0418F1375cE16D0727637";
var ETH_USD = "0x3BBe70e2F96c87aEce7F67A2b0178052f62E37fE";
var frxETH_ETH_Exchange_Rate_Low = "0x2aB80F9a26bc87c1359D86DDF89ce331648aE91E";
var MIMATIC_USD = "0x116542f62410Ac122C73ED3bC478937e781c5333";
var frxETH_ETH_Exchange_Rate_High = "0xa7CBaB1722a6631665863d624A81ab4C4eD0e876";
var CHAIN_ID = 1088;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_USD,
  BTC_USD,
  CHAIN_ID,
  DAI_USD,
  ETH_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LINK_USD,
  METIS_USD,
  MIMATIC_USD,
  USDC_USD,
  USDT_USD,
  frxETH_ETH_Exchange_Rate_High,
  frxETH_ETH_Exchange_Rate_Low,
  sfrxETH_frxETH_Exchange_Rate
});
