// @bgd-labs/aave-address-book/dist/ChainlinkMetis.mjs
// src/ts/ChainlinkMetis.ts
var BTC_USD = "0x51Ed8Fecf96813826F727CaBDF01b3cF6a61373e";
var DAI_USD = "0xe0351cAAE70B5AdBD0107cD5331AD1D79c4c1CA1";
var AAVE_USD = "0x54389e89A5Ec1d4312d5B5C48055d6E56a177BF9";
var USDT_USD = "0x51864b8948Aa5e35aace2BaDaF901D63418A3b9D";
var METIS_USD = "0xD4a5Bb03B5D66d9bf81507379302Ac2C2DFDFa6D";
var USDC_USD = "0x663855969c85F3BE415807250414Ca9129533a5f";
var sfrxETH_frxETH_Exchange_Rate = "0x732bEa04DF4184Fb50B5648bf1A74553f01Fd850";
var L2_Sequencer_Uptime_Status_Feed = "0x58218ea7422255EBE94e56b504035a784b7AA204";
var LINK_USD = "0x4A4F382A2FF9685de9f0418F1375cE16D0727637";
var ETH_USD = "0x3BBe70e2F96c87aEce7F67A2b0178052f62E37fE";
var frxETH_ETH_Exchange_Rate_Low = "0x2aB80F9a26bc87c1359D86DDF89ce331648aE91E";
var MIMATIC_USD = "0x116542f62410Ac122C73ED3bC478937e781c5333";
var frxETH_ETH_Exchange_Rate_High = "0xa7CBaB1722a6631665863d624A81ab4C4eD0e876";
var CHAIN_ID = 1088;
export {
  AAVE_USD,
  BTC_USD,
  CHAIN_ID,
  DAI_USD,
  ETH_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LINK_USD,
  METIS_USD,
  MIMATIC_USD,
  USDC_USD,
  USDT_USD,
  frxETH_ETH_Exchange_Rate_High,
  frxETH_ETH_Exchange_Rate_Low,
  sfrxETH_frxETH_Exchange_Rate
};
