// @bgd-labs/aave-address-book/dist/ChainlinkSoneium.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkSoneium.ts
var ChainlinkSoneium_exports = {};
__export(ChainlinkSoneium_exports, {
  ASTR_USD: () => ASTR_USD,
  BTC_USD: () => BTC_USD,
  CHAIN_ID: () => CHAIN_ID,
  Calculated_wstASTR_USD: () => Calculated_wstASTR_USD,
  ETH_USD: () => ETH_USD,
  L2_Sequencer_Uptime_Status_Feed: () => L2_Sequencer_Uptime_Status_Feed,
  LINK_ETH: () => LINK_ETH,
  LINK_USD: () => LINK_USD,
  OUSDT_USD: () => OUSDT_USD,
  RLP_USD_Exchange_Rate: () => RLP_USD_Exchange_Rate,
  SolvBTCBBN_SolvBTC_Exchange_Rate: () => SolvBTCBBN_SolvBTC_Exchange_Rate,
  USD1_USD: () => USD1_USD,
  USDC_USD: () => USDC_USD,
  USDT_USD: () => USDT_USD,
  USR_USD: () => USR_USD,
  WBTC_USD: () => WBTC_USD,
  pufETH_ETH: () => pufETH_ETH,
  solvBTC_BTC: () => solvBTC_BTC,
  stETH_ETH: () => stETH_ETH,
  wstETH_ETH: () => wstETH_ETH,
  wstUSR_stUSR_Exchange_Rate: () => wstUSR_stUSR_Exchange_Rate,
  xSolvBTC_NAV: () => xSolvBTC_NAV
});
module.exports = __toCommonJS(ChainlinkSoneium_exports);
var pufETH_ETH = "0xEEDE303b8602Db7556eD57BDf7133A6bAADa294D";
var BTC_USD = "0x884E2819193d8c8993036290D5a8778Dbe3942f2";
var SolvBTCBBN_SolvBTC_Exchange_Rate = "0x564eC0aeEb910e99dA4aC9341f3C01052F80A74B";
var L2_Sequencer_Uptime_Status_Feed = "0xaDE1b9AbB98c6A542E4B49db2588a3Ec4bF7Cdf0";
var solvBTC_BTC = "0x97D2996B726F774B7c5BF77C7A74CFe98174a9d3";
var USD1_USD = "0x251Ef178518aba5aa1df24713344EA1C6d9ccc67";
var LINK_ETH = "0x6D414dC3F9D8822D947B737CA8F75b1c4d6C8542";
var WBTC_USD = "0x75196350e8F9Ffa8e6a917E3C6E073D24F458781";
var Calculated_wstASTR_USD = "0x2580FA33f9dfD9D26F1aC418E09113f3E9c316c2";
var USR_USD = "0x316DefA0f4BA9567aF55E66212e66b4fD6646eEE";
var OUSDT_USD = "0xF049354158e13F6794A3cCD236694E0e1BD9AF79";
var wstETH_ETH = "0x4516fDa3259110EefAfC6fBcE266417B3F3b985a";
var RLP_USD_Exchange_Rate = "0x4f80347FE3579859232e1e8E0735020C008408cB";
var USDT_USD = "0xE92d289831823c96C22592952C1cfA2584a65038";
var ETH_USD = "0x291cF980BA12505D65ee01BDe0882F1d5e533525";
var USDC_USD = "0x46522a7fD5bD5E7aaFF862C17E116152e07d7158";
var LINK_USD = "0x5f01955B7048d1aC6F8023A5508963617f9FE4f8";
var ASTR_USD = "0xBa5C28f78eFdC03C37e2C46880314386aFf43228";
var wstUSR_stUSR_Exchange_Rate = "0xD5046A1128a57BDc609332A41Cfda72dAd623BC5";
var stETH_ETH = "0x931602D8758E7A891Dce57A31c2A23A98931Fe1c";
var xSolvBTC_NAV = "0x42237D7d96A4178bB22498bb3B2C689D502DE847";
var CHAIN_ID = 1868;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ASTR_USD,
  BTC_USD,
  CHAIN_ID,
  Calculated_wstASTR_USD,
  ETH_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LINK_ETH,
  LINK_USD,
  OUSDT_USD,
  RLP_USD_Exchange_Rate,
  SolvBTCBBN_SolvBTC_Exchange_Rate,
  USD1_USD,
  USDC_USD,
  USDT_USD,
  USR_USD,
  WBTC_USD,
  pufETH_ETH,
  solvBTC_BTC,
  stETH_ETH,
  wstETH_ETH,
  wstUSR_stUSR_Exchange_Rate,
  xSolvBTC_NAV
});
