// @bgd-labs/aave-address-book/dist/ChainlinkGnosis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkGnosis.ts
var ChainlinkGnosis_exports = {};
__export(ChainlinkGnosis_exports, {
  AAVE_Network_Emergency_Count_Gnosis_Chain: () => AAVE_Network_Emergency_Count_Gnosis_Chain,
  AAVE_USD: () => AAVE_USD,
  AVAX_USD: () => AVAX_USD,
  BAL_USD: () => BAL_USD,
  BNB_USD: () => BNB_USD,
  BTC_USD: () => BTC_USD,
  CHAIN_ID: () => CHAIN_ID,
  CHF_USD: () => CHF_USD,
  COMP_USD: () => COMP_USD,
  CRV_USD: () => CRV_USD,
  DAI_USD: () => DAI_USD,
  DOGE_USD: () => DOGE_USD,
  DOT_USD: () => DOT_USD,
  ETH_USD: () => ETH_USD,
  EUR_USD: () => EUR_USD,
  FTT_USD: () => FTT_USD,
  GHO_USD: () => GHO_USD,
  GNO_USD: () => GNO_USD,
  GRT_USD: () => GRT_USD,
  JPY_USD: () => JPY_USD,
  LINK_DAI: () => LINK_DAI,
  LINK_USD: () => LINK_USD,
  MKR_USD: () => MKR_USD,
  MXN_USD: () => MXN_USD,
  PERP_USD: () => PERP_USD,
  REN_USD: () => REN_USD,
  SNX_USD: () => SNX_USD,
  SOL_USD: () => SOL_USD,
  STETH_USD: () => STETH_USD,
  SUSHI_USD: () => SUSHI_USD,
  UMA_USD: () => UMA_USD,
  UNI_USD: () => UNI_USD,
  USDC_USD: () => USDC_USD,
  USDT_USD: () => USDT_USD,
  WBTC_USD: () => WBTC_USD,
  XAU_USD: () => XAU_USD,
  YFI_USD: () => YFI_USD,
  ZIL_USD: () => ZIL_USD,
  _1INCH_USD: () => _1INCH_USD,
  osETH_ETH_Exchange_Rate: () => osETH_ETH_Exchange_Rate,
  wstETH_ETH_Exchange_Rate: () => wstETH_ETH_Exchange_Rate
});
module.exports = __toCommonJS(ChainlinkGnosis_exports);
var ETH_USD = "0xa767f745331D267c7751297D982b050c93985627";
var GRT_USD = "0xeBbd67a84e33791F1bcFde74CDc22a71e332e2F1";
var GHO_USD = "0x855193775eF2ABb12CF95DE3086229a14Ab8E0E3";
var SOL_USD = "0xB7B7d008c49295A0FF6Eed6dF4Ad3052Fd39d5e6";
var UNI_USD = "0xd98735d78266c62277Bb4dBf3e3bCdd3694782F4";
var USDC_USD = "0x26C31ac71010aF62E6B486D1132E266D6298857D";
var SUSHI_USD = "0xC0a6Bf8d5D408B091D022C3C0653d4056D4B9c01";
var SNX_USD = "0x3b84d6e6976D5826500572600eB44f9f1753827b";
var DOT_USD = "0x3c30c5c415B2410326297F0f65f5Cbb32f3aefCc";
var XAU_USD = "0x4a5AB0F60d12a4420d36D3eD9A1F77d8c47EB94c";
var WBTC_USD = "0x00288135bE38B83249F380e9b6b9a04c90EC39eE";
var YFI_USD = "0x14030d5a0C9e63D9606C6f2c8771Fc95b34b07e0";
var CHF_USD = "0xFb00261Af80ADb1629D3869E377ae1EEC7bE659F";
var wstETH_ETH_Exchange_Rate = "0x0064AC007fF665CF8D0D3Af5E0AD1c26a3f853eA";
var EUR_USD = "0xab70BCB260073d036d1660201e9d5405F5829b7a";
var AAVE_Network_Emergency_Count_Gnosis_Chain = "0xF937ffAeA1363e4Fa260760bDFA2aA8Fc911F84D";
var FTT_USD = "0x0CaE8f5c10931f0Ce87Ed9BbB71391C6E93C2C26";
var COMP_USD = "0xBa95Bc8418Ebcdf8a690924E1d4aD5292139F2EA";
var LINK_USD = "0xed322A5ac55BAE091190dFf9066760b86751947B";
var DOGE_USD = "0x824b4A1A0443609A2ADd94a700b770FA5bE31287";
var MKR_USD = "0x51e4024255d0cBd1F4C79AEe6BDB6565Df2C5d1b";
var CRV_USD = "0xC77B83ac3Dd2a761073bD0f281f7b880B2DDDe18";
var BNB_USD = "0x6D42cc26756C34F26BEcDD9b30a279cE9Ea8296E";
var REN_USD = "0x27d4D36968a2BD1Cc3406D99cB1DF50561dBf2a4";
var ZIL_USD = "0x2997eBa3d9c2447c36107bB0F082b8c33566b49c";
var AVAX_USD = "0x911e08A32A6b7671A80387F93147Ab29063DE9A2";
var osETH_ETH_Exchange_Rate = "0xc5f7665e7FdC5059B93Db8bEAB83F5ffA05Eb18e";
var AAVE_USD = "0x2b481Dc923Aa050E009113Dca8dcb0daB4B68cDF";
var MXN_USD = "0xe9cea51a7b1b9B32E057ff62762a2066dA933cD2";
var BAL_USD = "0x1b723C855F7D2c2785F99486973271355e782d77";
var JPY_USD = "0x2AfB993C670C01e9dA1550c58e8039C1D8b8A317";
var BTC_USD = "0x6C1d7e76EF7304a40e8456ce883BC56d3dEA3F7d";
var PERP_USD = "0x76e76F7E73F3BD42E3c2b4282B50b36E78130B4A";
var UMA_USD = "0xF826E3ff8c0481D2e58DB9d301936F94Cd4fa916";
var DAI_USD = "0x678df3415fc31947dA4324eC63212874be5a82f8";
var USDT_USD = "0x68811D7DF835B1c33e6EEae8E7C141eF48d48cc7";
var GNO_USD = "0x22441d81416430A54336aB28765abd31a792Ad37";
var STETH_USD = "0x229e486Ee0D35b7A9f668d10a1e6029eEE6B77E0";
var LINK_DAI = "0xc89077976e03d19057eb296215C15afAe4A7464B";
var _1INCH_USD = "0xFDF9EB5fafc11Efa65f6FD144898da39a7920Ae8";
var CHAIN_ID = 100;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_Network_Emergency_Count_Gnosis_Chain,
  AAVE_USD,
  AVAX_USD,
  BAL_USD,
  BNB_USD,
  BTC_USD,
  CHAIN_ID,
  CHF_USD,
  COMP_USD,
  CRV_USD,
  DAI_USD,
  DOGE_USD,
  DOT_USD,
  ETH_USD,
  EUR_USD,
  FTT_USD,
  GHO_USD,
  GNO_USD,
  GRT_USD,
  JPY_USD,
  LINK_DAI,
  LINK_USD,
  MKR_USD,
  MXN_USD,
  PERP_USD,
  REN_USD,
  SNX_USD,
  SOL_USD,
  STETH_USD,
  SUSHI_USD,
  UMA_USD,
  UNI_USD,
  USDC_USD,
  USDT_USD,
  WBTC_USD,
  XAU_USD,
  YFI_USD,
  ZIL_USD,
  _1INCH_USD,
  osETH_ETH_Exchange_Rate,
  wstETH_ETH_Exchange_Rate
});
