// @bgd-labs/aave-address-book/dist/ChainlinkPolygon.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkPolygon.ts
var ChainlinkPolygon_exports = {};
__export(ChainlinkPolygon_exports, {
  AAPL_USD: () => AAPL_USD,
  AAVE_ETH: () => AAVE_ETH,
  AAVE_Network_Emergency_Count_Polygon: () => AAVE_Network_Emergency_Count_Polygon,
  AAVE_USD: () => AAVE_USD,
  ADA_USD: () => ADA_USD,
  AGEUR_USD: () => AGEUR_USD,
  ALCX_USD: () => ALCX_USD,
  ALGO_USD: () => ALGO_USD,
  AMZN_USD: () => AMZN_USD,
  APE_USD: () => APE_USD,
  AUD_USD: () => AUD_USD,
  AVAX_USD: () => AVAX_USD,
  AXS_USD: () => AXS_USD,
  BADGER_ETH: () => BADGER_ETH,
  BADGER_USD: () => BADGER_USD,
  BAL_ETH: () => BAL_ETH,
  BAL_USD: () => BAL_USD,
  BAT_USD: () => BAT_USD,
  BCH_USD: () => BCH_USD,
  BNB_USD: () => BNB_USD,
  BRL_USD: () => BRL_USD,
  BTC_ETH: () => BTC_ETH,
  BTC_USD: () => BTC_USD,
  BTC_USD_Total_Marketcap: () => BTC_USD_Total_Marketcap,
  CAD_USD: () => CAD_USD,
  CBETH_ETH: () => CBETH_ETH,
  CGT_Proof_of_Reserves_ETH: () => CGT_Proof_of_Reserves_ETH,
  CHAIN_ID: () => CHAIN_ID,
  CHF_USD: () => CHF_USD,
  CHZ_USD: () => CHZ_USD,
  CNY_USD: () => CNY_USD,
  COMP_USD: () => COMP_USD,
  COPW_Reserves: () => COPW_Reserves,
  COP_USD: () => COP_USD,
  CRV_ETH: () => CRV_ETH,
  CRV_USD: () => CRV_USD,
  CVX_USD: () => CVX_USD,
  Calculated_MaticX_USD: () => Calculated_MaticX_USD,
  Calculated_stMATIC_USD: () => Calculated_stMATIC_USD,
  DAI_ETH: () => DAI_ETH,
  DAI_USD: () => DAI_USD,
  DASH_USD: () => DASH_USD,
  DGB_USD: () => DGB_USD,
  DODO_USD: () => DODO_USD,
  DOGE_USD: () => DOGE_USD,
  DOGE_USD_Total_Marketcap: () => DOGE_USD_Total_Marketcap,
  DOT_USD: () => DOT_USD,
  DPI_ETH: () => DPI_ETH,
  DPI_USD: () => DPI_USD,
  EOS_USD: () => EOS_USD,
  ETC_USD: () => ETC_USD,
  ETH_USD: () => ETH_USD,
  ETH_USD_Total_Marketcap: () => ETH_USD_Total_Marketcap,
  EUR_USD: () => EUR_USD,
  FB_USD: () => FB_USD,
  FIL_USD: () => FIL_USD,
  FRAX_USD: () => FRAX_USD,
  FTM_USD: () => FTM_USD,
  FTT_USD: () => FTT_USD,
  FXS_USD: () => FXS_USD,
  GBP_USD: () => GBP_USD,
  GHST_ETH: () => GHST_ETH,
  GHST_USD: () => GHST_USD,
  GNS_USD: () => GNS_USD,
  GOOGL_USD: () => GOOGL_USD,
  GRT_USD: () => GRT_USD,
  HBAR_USD: () => HBAR_USD,
  ICP_USD: () => ICP_USD,
  IDR_USD: () => IDR_USD,
  ILS_USD: () => ILS_USD,
  ILV_ETH: () => ILV_ETH,
  INR_USD: () => INR_USD,
  JPY_USD: () => JPY_USD,
  KAVA_USD: () => KAVA_USD,
  KNC_USD: () => KNC_USD,
  KRW_USD: () => KRW_USD,
  LINK_ETH: () => LINK_ETH,
  LINK_MATIC: () => LINK_MATIC,
  LINK_USD: () => LINK_USD,
  LTC_USD: () => LTC_USD,
  MANA_USD: () => MANA_USD,
  MATIC_ETH: () => MATIC_ETH,
  MATIC_USD: () => MATIC_USD,
  MIMATIC_USD: () => MIMATIC_USD,
  MKR_ETH: () => MKR_ETH,
  MKR_USD: () => MKR_USD,
  MLN_ETH: () => MLN_ETH,
  MSFT_USD: () => MSFT_USD,
  MXN_USD: () => MXN_USD,
  MaticX_POL_Exchange_Rate: () => MaticX_POL_Exchange_Rate,
  NZD_USD: () => NZD_USD,
  OGN_USD: () => OGN_USD,
  OM_USD: () => OM_USD,
  PAXG_USD: () => PAXG_USD,
  PHP_USD: () => PHP_USD,
  PLN_USD: () => PLN_USD,
  QNT_USD: () => QNT_USD,
  QUICK_USD: () => QUICK_USD,
  RYT_NAV: () => RYT_NAV,
  RYT_Polygon_Proof_of_Reserves: () => RYT_Polygon_Proof_of_Reserves,
  SAND_USD: () => SAND_USD,
  SEK_USD: () => SEK_USD,
  SGD_USD: () => SGD_USD,
  SHIB_USD: () => SHIB_USD,
  SLP_USD: () => SLP_USD,
  SNX_USD: () => SNX_USD,
  SOL_USD: () => SOL_USD,
  SPYUS: () => SPYUS,
  STORJ_USD: () => STORJ_USD,
  SUSHI_ETH: () => SUSHI_ETH,
  SUSHI_USD: () => SUSHI_USD,
  THB_USD: () => THB_USD,
  THETA_USD: () => THETA_USD,
  TRUMATIC_MATIC_Exchange_Rate: () => TRUMATIC_MATIC_Exchange_Rate,
  TRX_USD: () => TRX_USD,
  TRY_USD: () => TRY_USD,
  TSLA_USD: () => TSLA_USD,
  TUSD_USD: () => TUSD_USD,
  UMA_USD: () => UMA_USD,
  UNI_ETH: () => UNI_ETH,
  UNI_USD: () => UNI_USD,
  USDC_ETH: () => USDC_ETH,
  USDC_USD: () => USDC_USD,
  USDS_USD: () => USDS_USD,
  USDT_ETH: () => USDT_ETH,
  USDT_USD: () => USDT_USD,
  USDe_USD: () => USDe_USD,
  VanEck_SMH_MID_iNAV: () => VanEck_SMH_MID_iNAV,
  WBTC_ETH: () => WBTC_ETH,
  WBTC_USD: () => WBTC_USD,
  WOO_USD: () => WOO_USD,
  WSTETH_ETH: () => WSTETH_ETH,
  XAG_USD: () => XAG_USD,
  XAU_USD: () => XAU_USD,
  XLM_USD: () => XLM_USD,
  XMR_USD: () => XMR_USD,
  XPT_USD: () => XPT_USD,
  XRP_USD: () => XRP_USD,
  XTZ_USD: () => XTZ_USD,
  YFI_ETH: () => YFI_ETH,
  YFI_USD: () => YFI_USD,
  ZAR_USD: () => ZAR_USD,
  ZEC_USD: () => ZEC_USD,
  _1INCH_USD: () => _1INCH_USD,
  bC3M_Reserves: () => bC3M_Reserves,
  bCSPX_Reserves: () => bCSPX_Reserves,
  bIB01_Reserves: () => bIB01_Reserves,
  bIBTA_Reserves: () => bIBTA_Reserves,
  bNVDA_Reserves_Proof_of_Reserves: () => bNVDA_Reserves_Proof_of_Reserves,
  ibBTC_PricePerShare: () => ibBTC_PricePerShare,
  sUSDe_USD: () => sUSDe_USD,
  sUSDe_USDe_Exchange_Rate: () => sUSDe_USDe_Exchange_Rate,
  wstETH_stETH_Exchange_Rate: () => wstETH_stETH_Exchange_Rate,
  wstPOL_POL_Exchange_Rate: () => wstPOL_POL_Exchange_Rate
});
module.exports = __toCommonJS(ChainlinkPolygon_exports);
var BTC_USD = "0xc907E116054Ad103354f2D350FD2514433D57F6f";
var WBTC_USD = "0xDE31F8bFBD8c84b5360CFACCa3539B938dd78ae6";
var VanEck_SMH_MID_iNAV = "0xD2D1BdD220C35C27f70a2bE1eD6462cECAc76384";
var SGD_USD = "0x8CE3cAc0E6635ce04783709ca3CC4F5fc5304299";
var PHP_USD = "0x218231089Bebb2A31970c3b77E96eCfb3BA006D1";
var QUICK_USD = "0x2251169D32E7538652a9a8c86bf0c43bFcd956f1";
var wstPOL_POL_Exchange_Rate = "0x2A5e28b4F9Ecb01807CeEC25F7b0fcCafF3F6721";
var XAG_USD = "0x461c7B8D370a240DdB46B402748381C3210136b3";
var CVX_USD = "0x5ec151834040B4D453A1eA46aA634C1773b36084";
var MANA_USD = "0xA1CbF3Fe43BC3501e3Fc4b573e822c70e76A7512";
var XAU_USD = "0x0C466540B2ee1a31b441671eac0ca886e051E410";
var GHST_ETH = "0xe638249AF9642CdA55A92245525268482eE4C67b";
var MATIC_USD = "0xAB594600376Ec9fD91F8e885dADF0CE036862dE0";
var DODO_USD = "0x59161117086a4C7A9beDA16C66e40Bdaa1C5a8B6";
var UNI_ETH = "0x162d8c5bF15eB6BEe003a1ffc4049C92114bc931";
var WOO_USD = "0x6a99EC84819FB7007dd5D032068742604E755c56";
var MSFT_USD = "0xC43081d9EA6d1c53f1F0e525504d47Dd60de12da";
var KRW_USD = "0x24B820870F726dA9B0D83B0B28a93885061dbF50";
var OM_USD = "0xc86105DccF9BD629Cea7Fd41f94c6050bF96D57F";
var KAVA_USD = "0x7899dd75C329eFe63e35b02bC7d60D3739FB23c5";
var TUSD_USD = "0x7C5D415B64312D38c56B54358449d0a4058339d2";
var ETH_USD_Total_Marketcap = "0x67935f65D1577ced9f4929D3679A157E95C1c02c";
var AMZN_USD = "0xf9184b8E5da48C19fA4E06f83f77742e748cca96";
var AGEUR_USD = "0x9b88d07B2354eF5f4579690356818e07371c7BeD";
var DOGE_USD = "0xbaf9327b6564454F4a3364C33eFeEf032b4b4444";
var SUSHI_ETH = "0x17414Eb5159A082e8d41D243C1601c2944401431";
var BNB_USD = "0x82a6c4AF830caa6c97bb504425f6A66165C2c26e";
var bNVDA_Reserves_Proof_of_Reserves = "0x0fB2beD999da86Cb1Fdd97E746600A96141EeA09";
var DAI_ETH = "0xFC539A559e170f848323e19dfD66007520510085";
var FIL_USD = "0xa07703E5C2eD1516107c7c72A494493Dcb99C676";
var SOL_USD = "0x10C8264C0935b3B9870013e057f330Ff3e9C56dC";
var COPW_Reserves = "0x1d22c334621364F16f050076eE15Acd5eb8225Ce";
var COP_USD = "0xfAA9147190c2C2cc5B8387B4f49016bDB3380572";
var NZD_USD = "0xa302a0B8a499fD0f00449df0a490DedE21105955";
var CHZ_USD = "0x2409987e514Ad8B0973C2b90ee1D95051DF0ECB9";
var MXN_USD = "0x171b16562EA3476F5C61d1b8dad031DbA0768545";
var QNT_USD = "0xF7F291042F6Cbc4deC0Ad75c17786511a530dbe8";
var LINK_USD = "0xd9FFdb71EbE7496cC440152d43986Aae0AB76665";
var TSLA_USD = "0x567E67f456c7453c583B6eFA6F18452cDee1F5a8";
var AAVE_ETH = "0xbE23a3AA13038CfC28aFd0ECe4FdE379fE7fBfc4";
var DPI_USD = "0x2e48b7924FBe04d575BA229A59b64547d9da16e9";
var bIBTA_Reserves = "0x4517002fCD31062Ea38680dF9Ee37f29528C2707";
var sUSDe_USDe_Exchange_Rate = "0xd1d85Bf11c28F878EE75BdC39387a2a15E0390b7";
var IDR_USD = "0x80a5cb83ce268Ed11a6EFC4bBF0beC39dF35Db21";
var GNS_USD = "0x9cb43aa3D036Cb035a694Ba0AAa91f8875B16cE1";
var EUR_USD = "0x73366Fe0AA0Ded304479862808e02506FE556a98";
var WSTETH_ETH = "0x10f964234cae09cB6a9854B56FF7D4F38Cda5E6a";
var USDS_USD = "0x3B21a277babcB73b5a2325106FC04Df8AA1aC5b6";
var bC3M_Reserves = "0x648E0fF6A36D58F6FCE5927cB77601b73cAdc2Af";
var ILS_USD = "0x8d5eB34C509261533235b91350d359EdcB969D33";
var TRX_USD = "0x307cCF7cBD17b69A487b9C3dbe483931Cf3E1833";
var GRT_USD = "0x3FabBfb300B1e2D7c9B84512fe9D30aeDF24C410";
var GHST_USD = "0xDD229Ce42f11D8Ee7fFf29bDB71C7b81352e11be";
var SPYUS = "0x187c42f6C0e7395AeA00B1B30CB0fF807ef86d5d";
var KNC_USD = "0x10e5f3DFc81B3e5Ef4e648C4454D04e79E1E41E2";
var LTC_USD = "0xEB99F173cf7d9a6dC4D889C2Ad7103e8383b6Efa";
var GOOGL_USD = "0x1b32682C033b2DD7EFdC615FA82d353e254F39b5";
var ILV_ETH = "0x3636B780588328dc3F5df075De5627DBc9A6BA10";
var bIB01_Reserves = "0xad4395fc414Fc1575A7a38C20B0Bfdbdb09ee41A";
var ZAR_USD = "0xd4a120c26d57B910C56c910CdD13EeBFA3135502";
var MLN_ETH = "0xB89D583B72aBF9C3a7e6e093251C2fCad3365312";
var CNY_USD = "0x04bB437Aa63E098236FA47365f0268547f6EAB32";
var PLN_USD = "0xB34BCE11040702f71c11529D00179B2959BcE6C0";
var BAL_ETH = "0x03CD157746c61F44597dD54C6f6702105258C722";
var ETH_USD = "0xF9680D99D6C9589e2a93a78A04A279e509205945";
var ibBTC_PricePerShare = "0xc3E676E68dB28c9Fb2199f25B60560723237cc76";
var COMP_USD = "0x2A8758b7257102461BC958279054e372C2b1bDE6";
var TRUMATIC_MATIC_Exchange_Rate = "0xed7051d55eeaF85E36053D6b0f2C51E59625B047";
var YFI_USD = "0x9d3A43c111E7b2C6601705D9fcF7a70c95b1dc55";
var PAXG_USD = "0x0f6914d8e7e1214CDb3A4C6fbf729b75C69DF608";
var LINK_MATIC = "0x5787BefDc0ECd210Dfa948264631CD53E68F7802";
var FB_USD = "0x5b4586C911144A947D7814Fd71fe0872b8334748";
var WBTC_ETH = "0xA338e0492B2F944E9F8C0653D3AD1484f2657a37";
var AUD_USD = "0x062Df9C4efd2030e243ffCc398b652e8b8F95C6f";
var MaticX_POL_Exchange_Rate = "0x864377d3620da9d42451f98674593e067a997308";
var MIMATIC_USD = "0xd8d483d813547CfB624b8Dc33a00F2fcbCd2D428";
var DOT_USD = "0xacb51F1a83922632ca02B25a8164c10748001BdE";
var XMR_USD = "0xBE6FB0AB6302B693368D0E9001fAF77ecc6571db";
var INR_USD = "0xDA0F8Df6F5dB15b346f4B8D1156722027E194E60";
var THB_USD = "0x5164Ad28fb12a5e55946090Ec3eE1B748AFb3785";
var FTM_USD = "0x58326c0F831b2Dbf7234A4204F28Bba79AA06d5f";
var USDT_USD = "0x0A6513e40db6EB1b165753AD52E80663aeA50545";
var BAL_USD = "0xD106B538F2A868c28Ca1Ec7E298C3325E0251d66";
var XTZ_USD = "0x691e26AB58ff05800E028b0876A41B720b26FC65";
var wstETH_stETH_Exchange_Rate = "0x3Ea1eC855fBda8bA0396975eC260AD2e9B2Bc01c";
var AAVE_Network_Emergency_Count_Polygon = "0xDAFA1989A504c48Ee20a582f2891eeB25E2fA23F";
var XPT_USD = "0x76631863c2ae7367aF8f37Cd10d251DA7f1DE186";
var XRP_USD = "0x785ba89291f676b5386652eB12b30cF361020694";
var RYT_Polygon_Proof_of_Reserves = "0x6b20c92A1D73A9232383DEcD698Ff559eFa0c31A";
var sUSDe_USD = "0xe71A04ACe06e2667c7e1c802E998CBf44A6bf53C";
var DOGE_USD_Total_Marketcap = "0xbd238a35Fb47aE22F0cC551f14ffB8E8f04FCA21";
var MKR_USD = "0xa070427bF5bA5709f70e98b94Cb2F435a242C46C";
var CAD_USD = "0xACA44ABb8B04D07D883202F99FA5E3c53ed57Fb5";
var BRL_USD = "0xB90DA3ff54C3ED09115abf6FbA0Ff4645586af2c";
var UNI_USD = "0xdf0Fb4e4F928d2dCB76f438575fDD8682386e13C";
var EOS_USD = "0xd6285F06203D938ab713Fa6A315e7d23247DDE95";
var TRY_USD = "0xd78325DcA0F90F0FFe53cCeA1B02Bb12E1bf8FdB";
var SNX_USD = "0xbF90A5D9B6EE9019028dbFc2a9E50056d5252894";
var FXS_USD = "0x6C0fe985D3cAcbCdE428b84fc9431792694d0f51";
var APE_USD = "0x2Ac3F3Bfac8fC9094BC3f0F9041a51375235B992";
var BTC_USD_Total_Marketcap = "0x18E4058491C3F58bC2f747A9E64cA256Ed6B318d";
var USDT_ETH = "0xf9d5AAC6E5572AEFa6bd64108ff86a222F69B64d";
var BADGER_ETH = "0x82C9d4E88862f194C2bd874a106a90dDD0D35AAB";
var AXS_USD = "0x9c371aE34509590E10aB98205d2dF5936A1aD875";
var YFI_ETH = "0x9896A1eA7A00F5f32Ab131eBbeE07487B0af31D0";
var AAVE_USD = "0x72484B12719E23115761D5DA1646945632979bB6";
var SEK_USD = "0xbd92B4919ae82be8473859295dEF0e778A626302";
var SUSHI_USD = "0x49B0c695039243BBfEb8EcD054EB70061fd54aa0";
var CBETH_ETH = "0x0a6a03CdF7d0b48d4e4BA8e362A4FfC3aAC4f3c0";
var CGT_Proof_of_Reserves_ETH = "0x4e9fc7480c16F3FE5d956C0759eE6b4808d1F5D7";
var CRV_USD = "0x336584C8E6Dc19637A5b36206B1c79923111b405";
var THETA_USD = "0x38611b09F8f2D520c14eA973765C225Bf57B9Eac";
var MKR_ETH = "0x807b59d12520830D1864286FA0271c27baa94197";
var USDC_ETH = "0xefb7e6be8356cCc6827799B6A7348eE674A80EaE";
var OGN_USD = "0x8Ec0eC2e0F26D8253ABf39Db4B1793D76B49C6D5";
var BCH_USD = "0x327d9822e9932996f55b39F557AEC838313da8b7";
var FTT_USD = "0x817A7D43f0277Ca480AE03Ec76Fc63A2EC7114bA";
var JPY_USD = "0xD647a6fC9BC6402301583C91decC5989d8Bc382D";
var Calculated_stMATIC_USD = "0x97371dF4492605486e23Da797fA68e55Fc38a13f";
var DASH_USD = "0xD94427eDee70E4991b4b8DdCc848f2B58ED01C0b";
var FRAX_USD = "0x00DBeB1e45485d53DF7C2F0dF1Aa0b6Dc30311d3";
var SLP_USD = "0xBB3eF70953fC3766bec4Ab7A9BF05B6E4caf89c6";
var SHIB_USD = "0x3710abeb1A0Fc7C2EC59C26c8DAA7a448ff6125A";
var SAND_USD = "0x3D49406EDd4D52Fb7FFd25485f32E073b529C924";
var BTC_ETH = "0x19b0F0833C78c0848109E3842D34d2fDF2cA69BA";
var BAT_USD = "0x2346Ce62bd732c62618944E51cbFa09D985d86D2";
var AVAX_USD = "0xe01eA2fbd8D76ee323FbEd03eB9a8625EC981A10";
var MATIC_ETH = "0x327e23A4855b6F663a28c5161541d69Af8973302";
var _1INCH_USD = "0x443C5116CdF663Eb387e72C688D276e702135C87";
var USDe_USD = "0x5c3890e86f3E7Ed7F5390532De147953580f1605";
var BADGER_USD = "0xF626964Ba5e81405f47e8004F0b276Bb974742B5";
var XLM_USD = "0x692AE5510cA9070095A496dbcFBCDA99D4024Cd9";
var bCSPX_Reserves = "0x55e75d35c44A9EE1A5b05416640965EbcA4a8D33";
var ETC_USD = "0xDf3f72Be10d194b58B1BB56f2c4183e661cB2114";
var DPI_ETH = "0xC70aAF9092De3a4E5000956E672cDf5E996B4610";
var ALGO_USD = "0x03Bc6D9EFed65708D35fDaEfb25E87631a0a3437";
var STORJ_USD = "0x0F1d5Bd7be9B30Fc09E110cd6504Bd450e53cb0E";
var HBAR_USD = "0xC5878bDf8a89FA3bF0DC8389ae8EE6DE601D01bC";
var USDC_USD = "0xfE4A8cc5b5B2366C1B58Bea3858e81843581b2F7";
var GBP_USD = "0x099a2540848573e94fb1Ca0Fa420b00acbBc845a";
var AAPL_USD = "0x7E7B45b08F68EC69A99AAb12e42FcCB078e10094";
var ALCX_USD = "0x5DB6e61B6159B20F068dc15A47dF2E5931b14f29";
var ADA_USD = "0x882554df528115a743c4537828DA8D5B58e52544";
var DAI_USD = "0x4746DeC9e833A82EC7C2C1356372CcF2cfcD2F3D";
var LINK_ETH = "0xb77fa460604b9C6435A235D057F7D319AC83cb53";
var DGB_USD = "0x4205eC5fd179A843caa7B0860a8eC7D980013359";
var ZEC_USD = "0xBC08c639e579a391C4228F20d0C29d0690092DF0";
var UMA_USD = "0x33D9B1BAaDcF4b26ab6F8E83e9cb8a611B2B3956";
var CRV_ETH = "0x1CF68C76803c9A415bE301f50E82e44c64B7F1D4";
var Calculated_MaticX_USD = "0x5d37E4b374E6907de8Fc7fb33EE3b0af403C7403";
var RYT_NAV = "0xdefcBDC0a5f5fBBdF520dC97be8d6d025B18da05";
var ICP_USD = "0x84227A76a04289473057BEF706646199D7C58c34";
var CHF_USD = "0xc76f762CedF0F78a439727861628E0fdfE1e70c2";
var CHAIN_ID = 137;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAPL_USD,
  AAVE_ETH,
  AAVE_Network_Emergency_Count_Polygon,
  AAVE_USD,
  ADA_USD,
  AGEUR_USD,
  ALCX_USD,
  ALGO_USD,
  AMZN_USD,
  APE_USD,
  AUD_USD,
  AVAX_USD,
  AXS_USD,
  BADGER_ETH,
  BADGER_USD,
  BAL_ETH,
  BAL_USD,
  BAT_USD,
  BCH_USD,
  BNB_USD,
  BRL_USD,
  BTC_ETH,
  BTC_USD,
  BTC_USD_Total_Marketcap,
  CAD_USD,
  CBETH_ETH,
  CGT_Proof_of_Reserves_ETH,
  CHAIN_ID,
  CHF_USD,
  CHZ_USD,
  CNY_USD,
  COMP_USD,
  COPW_Reserves,
  COP_USD,
  CRV_ETH,
  CRV_USD,
  CVX_USD,
  Calculated_MaticX_USD,
  Calculated_stMATIC_USD,
  DAI_ETH,
  DAI_USD,
  DASH_USD,
  DGB_USD,
  DODO_USD,
  DOGE_USD,
  DOGE_USD_Total_Marketcap,
  DOT_USD,
  DPI_ETH,
  DPI_USD,
  EOS_USD,
  ETC_USD,
  ETH_USD,
  ETH_USD_Total_Marketcap,
  EUR_USD,
  FB_USD,
  FIL_USD,
  FRAX_USD,
  FTM_USD,
  FTT_USD,
  FXS_USD,
  GBP_USD,
  GHST_ETH,
  GHST_USD,
  GNS_USD,
  GOOGL_USD,
  GRT_USD,
  HBAR_USD,
  ICP_USD,
  IDR_USD,
  ILS_USD,
  ILV_ETH,
  INR_USD,
  JPY_USD,
  KAVA_USD,
  KNC_USD,
  KRW_USD,
  LINK_ETH,
  LINK_MATIC,
  LINK_USD,
  LTC_USD,
  MANA_USD,
  MATIC_ETH,
  MATIC_USD,
  MIMATIC_USD,
  MKR_ETH,
  MKR_USD,
  MLN_ETH,
  MSFT_USD,
  MXN_USD,
  MaticX_POL_Exchange_Rate,
  NZD_USD,
  OGN_USD,
  OM_USD,
  PAXG_USD,
  PHP_USD,
  PLN_USD,
  QNT_USD,
  QUICK_USD,
  RYT_NAV,
  RYT_Polygon_Proof_of_Reserves,
  SAND_USD,
  SEK_USD,
  SGD_USD,
  SHIB_USD,
  SLP_USD,
  SNX_USD,
  SOL_USD,
  SPYUS,
  STORJ_USD,
  SUSHI_ETH,
  SUSHI_USD,
  THB_USD,
  THETA_USD,
  TRUMATIC_MATIC_Exchange_Rate,
  TRX_USD,
  TRY_USD,
  TSLA_USD,
  TUSD_USD,
  UMA_USD,
  UNI_ETH,
  UNI_USD,
  USDC_ETH,
  USDC_USD,
  USDS_USD,
  USDT_ETH,
  USDT_USD,
  USDe_USD,
  VanEck_SMH_MID_iNAV,
  WBTC_ETH,
  WBTC_USD,
  WOO_USD,
  WSTETH_ETH,
  XAG_USD,
  XAU_USD,
  XLM_USD,
  XMR_USD,
  XPT_USD,
  XRP_USD,
  XTZ_USD,
  YFI_ETH,
  YFI_USD,
  ZAR_USD,
  ZEC_USD,
  _1INCH_USD,
  bC3M_Reserves,
  bCSPX_Reserves,
  bIB01_Reserves,
  bIBTA_Reserves,
  bNVDA_Reserves_Proof_of_Reserves,
  ibBTC_PricePerShare,
  sUSDe_USD,
  sUSDe_USDe_Exchange_Rate,
  wstETH_stETH_Exchange_Rate,
  wstPOL_POL_Exchange_Rate
});
