// @bgd-labs/aave-address-book/dist/ChainlinkArbitrum.mjs
// src/ts/ChainlinkArbitrum.ts
var AAPL_USD = "0x8d0CC5f38f9E802475f2CFf4F9fc7000C2E1557c";
var SPY_USD = "0x46306F3795342117721D8DEd50fbcF6DF2b3cc10";
var OP_USD = "0x205aaD468a11fd5D34fA7211bC6Bad5b3deB9b98";
var NVDA_USD = "0x4881A4418b5F2460B21d6F08CD5aA0678a7f262F";
var XAG_USD = "0xC56765f04B248394CF1619D20dB8082Edbfa75b1";
var rETH_ETH_Exchange_Rate = "0xF3272CAfe65b190e76caAF483db13424a3e23dD2";
var MIM_USD = "0x87121F6c9A9F6E90E59591E4Cf4804873f54A95b";
var HYPE_USD = "0xf9ce4fE2F0EcE0362cb416844AE179a49591D567";
var WSTUSR_STUSR_Exchange_Rate = "0x6BC7fF3C21f005684ef834FBA4d88459e988CcdC";
var GLV_WETH_USDC_USD = "0xD0E31451f58d3991045D68C479ea80D37e418851";
var WLFI_USD = "0x4b13Dd76De990Db9A2Dab58D35C2c02E5e3AE848";
var HSK_USD = "0x866fa212dfc5AAf321379874eCe95631F6d12e00";
var WSTETH_ETH = "0xb523AE262D20A936BC152e6023996e46FDC2A95D";
var instETH_ETH_Exchange_Rate = "0x4050bD8263771f8BBded08C299BD944488a91AaD";
var TSLA_USD = "0x3609baAa0a9b1f0FE4d6CC01884585d0e191C3E3";
var USDS_USD = "0x37833E5b3fbbEd4D613a3e0C354eF91A42B81eeB";
var USD1_USD = "0xcAc0138592090762385CCB7bb94C9401F734Eb30";
var USDG_USD = "0xE9Bd4Ca2CDacb56cd06c9C4C5231e69c33340B3c";
var DOLO_USD = "0x17d8D87dF3E279c737568aB0C5cC3fF750aB763e";
var GOOGL_USD = "0x1D1a83331e9D255EB1Aaf75026B60dFD00A252ba";
var RETH_ETH = "0xD6aB2298946840262FcC278fF31516D39fF611eF";
var WBTC_BTC = "0x0017abAc5b6f291F9164e35B1234CA1D697f9CF4";
var STX_USD = "0x3a9659C071dD3C37a8b1A2363409A8D41B2Feae3";
var wOETH_OETH_Exchange_Rate = "0x03a1f4b19aaeA6e68f0f104dc4346dA3E942cC45";
var XRP_USD = "0xB4AD57B52aB9141de9926a3e0C8dc6264c2ef205";
var CNY_USD = "0xcC3370Bde6AFE51e1205a5038947b9836371eCCb";
var KNC_USD = "0xbF539d4c2106dd4D9AB6D56aed3d9023529Db145";
var LUSD_USD = "0x0411D28c94d85A36bC72Cb0f875dfA8371D8fFfF";
var KRW_USD = "0x85bb02E0Ae286600d1c68Bb6Ce22Cc998d411916";
var SUI_USD = "0x4a85B128EBDaFC24d5CB611e161376ffDECeB289";
var USDC_USD = "0x50834F3163758fcC1Df9973b6e91f0F0F0434aD3";
var aBTC_Proof_of_Reserves = "0x5F25Ea8Cf2a3aB12E4D6fB58FB63C7938473B036";
var APE_USD = "0x221912ce795669f628c51c69b7d0873eDA9C03bB";
var DOT_USD = "0xa6bC5bAF2000424e90434bA7104ee399dEe80DEc";
var swETH_ETH_Exchange_Rate = "0xEcD471ef663eCEFFC7D731a0C7e51007433e6d6e";
var deUSD_USD = "0xa2b9BFb5CE1c7ABBC6609de17e5968eA2c20a5f1";
var PAXG_USD = "0x2BA975D4D7922cD264267Af16F3bD177F206FE3c";
var LTC_USD = "0x5698690a7B7B84F6aa985ef7690A8A7288FBc9c8";
var ANON_USD = "0x38ACD8CA9612737D2Ba42881b544626A8E18b061";
var ezETH_ETH = "0x11E1836bFF2ce9d6A5bec9cA79dc998210f3886d";
var RYT_Arbitrum_Proof_of_Reserves = "0xb966F82183E41208f594586BBF04bb6163571C48";
var FBTC_Proof_of_Reserves = "0xfb010c715Ff2B85DB9c896f0C030bF9329638905";
var USDPlus_USD = "0x6548a81E640C000150e06AB413fB3F772682e9c5";
var stBTC_BTC_Exchange_Rate = "0x7A58648CAeA88e042767823723aF266DC6c4c4D9";
var ETH_USD = "0x639Fe6ab55C921f74e7fac1ee960C0B6293ba612";
var sUSDe_USD = "0xf2215b9c35b1697B5f47e407c917a40D055E68d7";
var SNX_USD = "0x054296f0D036b95531B4E14aFB578B80CFb41252";
var SWETH_ETH = "0x05Bc6e5Fb110589bb366A3Cd7CdBe143EeBA2168";
var LBTC_BTC = "0x9fa74925F21ad6C86d8f402EF490cFbA2Fa5e9bE";
var yUSD_USD_Exchange_Rate = "0x875C69Da7AE092E4Dc277171e9c373718DDB25E2";
var vyUSD_USD_Exchange_Rate = "0x640F0B3e8Ee59bD7310C3E16B93f4c6612CC1130";
var wrsETH_rsETH_Exchange_Rate = "0x8f1dF6D7F2db73eECE86a18b4381F4707b918FB1";
var AAVE_USD = "0xaD1d5344AaDE45F43E596773Bcc4c423EAbdD034";
var BTC_ETH = "0xc5a90A6d7e4Af242dA238FFe279e9f2BA0c64B2e";
var CVX_USD = "0x851175a919f36c8e30197c09a9A49dA932c2CC00";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Level = "0x926Eed7973EBc57a8767659C5b35b9a0bAB6302f";
var GHO_USD = "0x3c786e934F23375Ca345C9b8D5aD54838796E8e7";
var AUD_USD = "0x9854e9a850e7C354c1de177eA953a6b1fba8Fc22";
var RPL_USD = "0xF0b7159BbFc341Cc41E7Cb182216F62c6d40533D";
var ankrETH_ETH_Exchange_Rate = "0x5Fb73f7Af8a29297953d3611422826039338E5b4";
var sUSDe_USDe_Exchange_Rate = "0x605EA726F0259a30db5b7c9ef39Df9fE78665C44";
var ARB_USD = "0xb2A824043730FE05F3DA2efaFa1CBbe83fa548D6";
var ORDER_USD = "0xE2A3216D8e4BdFA2Ee78F2e55B995e787e6Ce500";
var agETH_ETH_Exchange_Rate = "0x6303C35D49B13016A6acef5414893Fa7519E6da0";
var wstUSR_stUSR_Exchange_Rate = "0x9BC7E5a6f1EED1C3217d2c63ad680DF83D84a906";
var CAKE_USD = "0x256654437f1ADA8057684b18d742eFD14034C400";
var rswETH_ETH_Exchange_Rate = "0x683989a7A6424122678164Ad26736bA484055B35";
var WTI_USD = "0x594b919AD828e693B935705c3F816221729E7AE8";
var YETH_ETH_Exchange_Rate = "0x703389807836463cff0CB01142dE903a74E509C4";
var XTZ_USD = "0xf3451Fd5eddE08cDAE95A6233BaD69DE95552a61";
var gmARB_USD = "0x5d046567b97B0d7322F2402e3b34Bf789cE329f5";
var STG_USD = "0xe74d69E233faB0d8F48921f2D93aDfDe44cEb3B7";
var PENDLE_USD = "0x66853E19d73c0F9301fe099c324A1E9726953433";
var GNS_USD = "0xE89E98CE4E19071E59Ed4780E0598b541CE76486";
var ibBTC_PricePerShare = "0x80dd57c45B73f3c70feAF1BFe1bcdF384703E558";
var CAD_USD = "0xf6DA27749484843c4F02f5Ad1378ceE723dD61d4";
var COMP_USD = "0xe7C53FFd03Eb6ceF7d208bC4C13446c76d1E5884";
var BAL_USD = "0xBE5eA816870D11239c543F84b71439511D70B94f";
var RYT_NAV = "0xBf4C6D59070F5649670E3d4f9033673D3489c4E9";
var MKR_USD = "0xdE9f0894670c4EFcacF370426F10C3AD2Cdf147e";
var frxETH_ETH_Exchange_Rate_High = "0x5C3e80763862CB777Aa07BDDBcCE0123104e1c34";
var AMZN_USD = "0xd6a77691f071E98Df7217BED98f38ae6d2313EBA";
var RSR_USD = "0xcfF9349ec6d027f20fC9360117fef4a1Ad38B488";
var MELANIA_USD = "0xE2CB592D636c500a6e469628054F09d58e4d91BB";
var FRAX_USD = "0x0809E3d38d1B4214958faf06D8b1B1a2b73f2ab8";
var PHP_USD = "0xfF82AAF635645fD0bcc7b619C3F28004cDb58574";
var MSFT_USD = "0xDde33fb9F21739602806580bdd73BAd831DcA867";
var iBTC_Proof_of_Reserves = "0x0543Bb6b5A57E3e7569237b67321eAB773e41401";
var MAGIC_USD = "0x47E55cCec6582838E173f252D08Afd8116c2202d";
var USR_USD = "0xBafDf5307EaEF8dCd18c4b7b3977BEEB73b4AFE4";
var BRL_USD = "0x04b7384473A2aDF1903E3a98aCAc5D62ba8C2702";
var sfrxETH_frxETH_Exchange_Rate = "0x98E5a52fB741347199C08a7a3fcF017364284431";
var COIN_USD = "0x950DC95D4E537A14283059bADC2734977C454498";
var SHIB_USD = "0x0E278D14B4bf6429dDB0a1B353e2Ae8A4e128C93";
var YFI_USD = "0x745Ab5b69E01E2BE1104Ca84937Bb71f96f5fB21";
var FBTC_BTC_Exchange_Rate = "0x84ad1Cb9FDE786c00a77b7539344ad3C66A616f5";
var GRASS_USD = "0x3791D5eFF50Ab65fD82BBb16BCba42BEA556E7aA";
var PEPE_USD = "0x02DEd5a7EDDA750E3Eb240b54437a54d57b74dBE";
var YBTC_BTC_Exchange_Rate = "0x726a96F31846f586E6a4C72F73c3dB671986a002";
var AXL_USD = "0x84e8237CC1418Ea1B4A1e0C3e7F48c3A5fbC81AF";
var IOTX_USD = "0x484A1b29ED1Ea038dBd75D7c7293714343363122";
var MVI_USD = "0x87B679C03e9672Af516a9E08085e1F4FA1722A3D";
var WIF_USD = "0xF7Ee427318d2Bd0EEd3c63382D0d52Ad8A68f90D";
var ynETH_ETH_Exchange_Rate = "0xc2430cD1214F8452f4040473b7587195f9c565bD";
var solvBTC_BTC = "0x9ca3D04Bdf90Ddc0074Ea57d3e2D331CD4b2d5a2";
var FTM_USD = "0xFeaC1A3936514746e70170c0f539e70b23d36F19";
var StaFi_Staked_ETH_rETH_ETH_Exchange_Rate = "0x052d4200b624b07262F574af26C71A6553996Ab5";
var stBTC_Proof_of_Reserves = "0xA0118D8948FA85624608d94eC6B52973578082fA";
var GRT_USD = "0x0F38D86FceF4955B705F35c9e41d1A16e0637c73";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate = "0x5Fb8E23065C362080E540e01e0bd4119B6cB53fe";
var ATOM_USD = "0xCDA67618e51762235eacA373894F0C79256768fa";
var ATH_USD = "0x920E09A98f83b82701207BA7E433a5b53eCB888A";
var GBP_USD = "0x9C4424Fd84C6661F97D8d6b3fc3C1aAc2BeDd137";
var egETH_ETH_Exchange_Rate = "0xD3631AC9D81eD560D61957a55E9c992cdE497eb6";
var CRV_USD = "0xaebDA2c976cfd1eE1977Eac079B4382acb849325";
var EUR_USD = "0xA14d53bC1F1c0F31B4aA3BD109344E5009051a84";
var SHIB_ETH = "0x639b7cD102A1c5BaB71c54f88D41E894215c54E2";
var Total_Marketcap_USD = "0x4763b84cdBc5211B9e0a57D5E39af3B3b2440012";
var TIA_USD = "0x4096b9bfB4c34497B7a3939D4f629cf65EBf5634";
var cbETH_ETH_Exchange_Rate = "0x0518673439245BB95A58688Bc31cd513F3D5bDd6";
var SGD_USD = "0xF0d38324d1F86a176aC727A4b0c43c9F9d9c5EB1";
var syrupUSDC_USDC_Exchange_Rate = "0xF8722c901675C4F2F7824E256B8A6477b2c105FB";
var gmBTC_USD = "0x395D5c5D552Df670dc4B2B1cef0c4EABfFba492f";
var WOO_USD = "0x5e2b5C5C07cCA3437c4D724225Bb42c7E55d1597";
var BTC_USD_Total_Marketcap = "0x7519bCA20e21725557Bb98d9032124f8885a26C2";
var frxUSD_USD = "0xF6a5c5480A9aC736ea567454e2E570889E0DB4Bc";
var TRY_USD = "0xE8f8AfE4b56c6C421F691bfAc225cE61b2C7CD05";
var ULTI_USD = "0x8883045300Eaf3b1Bb1b3b17F9B4d70EfF50212a";
var ENA_USD = "0x9eE96caa9972c801058CAA8E23419fc6516FbF7e";
var SOL_USD = "0x24ceA4b8ce57cdA5058b924B9B9987992450590c";
var xSolvBTC_NAV = "0xAE8B4179389059A735821A77417eACcA60f3e4B4";
var PUMP_USD = "0x0C997958ccE7A0403AEA7E34d14bbaDA897B5bb3";
var sUSDS_USDS_Exchange_Rate = "0x2483326d19f780Fb082f333Fe124e4C075B207ba";
var XVS_USD = "0x300b0990Ba191a1AeBef6e5Ed8B5B308C0B2d0c9";
var TBTC_BTC = "0xe4c31C5B118d8aA92433eD1c7EC70afb430cd730";
var ezETH_ETH_Exchange_Rate = "0x989a480b6054389075CBCdC385C18CfB6FC08186";
var BERA_USD = "0x4f861F14246229530a881D32C8d26D78b8c48BE6";
var weETH_eETH_Exchange_Rate = "0x20bAe7e1De9c596f5F7615aeaa1342Ba99294e12";
var BTC_USD = "0x6ce185860a4963106506C203335A2910413708e9";
var RDNT_USD = "0x20d0Fcab0ECFD078B036b6CAf1FaC69A6453b352";
var MLN_USD = "0xD07de6e37A011CCAfD375d7eb130205E0fa24d69";
var LINK_USD = "0x86E53CF1B870786351Da77A57575e79CB55812CB";
var USDX_USD = "0x68148201f743eFE9AfC2Bc62F9d6D5672d118Bb3";
var BNB_USD = "0x6970460aabF80C5BE983C6b74e5D06dEDCA95D4A";
var weETH_ETH = "0xE141425bc1594b8039De6390db1cDaf4397EA22b";
var AXS_USD = "0x5B58aA6E0651Ae311864876A55411F481aD86080";
var MATIC_USD = "0x52099D4523531f678Dfc568a7B1e5038aadcE1d6";
var PLUME_USD = "0x004f214586CaFE0F1321Ad13A93bca77C385b196";
var ZAR_USD = "0xA9cC9B5Ea2584239365Ea6b985868D121CB7Aea6";
var TAO_USD = "0x6aCcBB82aF71B8a576B4C05D4aF92A83A035B991";
var dlcBTC_Proof_of_Reserves = "0x47A2fBEb46553F01E7133686Fb1b5349d4823a0C";
var META_USD = "0xcd1bd86fDc33080DCF1b5715B6FCe04eC6F85845";
var MNT_USD = "0x37DDEE84dE03d039e1Bf809b7a01EDd2c4665771";
var Real_GDP_Level = "0x326fA29b208DE897981C78827b8ffDbfa355590b";
var STETH_ETH = "0xded2c52b75B24732e9107377B7Ba93eC1fFa4BAf";
var USDe_USD = "0x88AC7Bca36567525A866138F03a6F6844868E0Bc";
var uniETH_ETH_Exchange_Rate = "0xF18BE32dB91591A7256d738C166FA195a17457DA";
var L2_Sequencer_Uptime_Status_Feed = "0xFdB631F5EE196F0ed6FAa767959853A9F217697D";
var ETHx_ETH = "0xB4AC4078DDA43d0eB6Bb9e08b8C12A73f9FEAA7d";
var ETH_USD_Total_Marketcap = "0xB1f70A229FE7cceD0428245db8B1f6C48c7Ea82a";
var LDO_USD = "0xA43A34030088E6510FecCFb77E88ee5e7ed0fE64";
var USR_USD_Exchange_Rate = "0x229a9322D050BB6aC91C40fFbB8e3bef79c48071";
var ZRO_USD = "0x1940fEd49cDBC397941f2D336eb4994D599e568B";
var CRO_USD = "0xea320E4d688B143A3bFBF1b4a5cc4B986fCa086c";
var TBTC_USD = "0xE808488e8627F6531bA79a13A9E0271B39abEb1C";
var USDT_USD = "0x3f3f5dF88dC9F13eac63DF89EC16ef6e7E25DdE7";
var osETH_ETH_Exchange_Rate = "0xB4102D5E72c402D537C9f024F4bd9c3709FE200d";
var TRUMP_USD = "0x373510BDa1ab7e873c731968f4D81B685f520E4B";
var zBTC_Proof_of_Reserves = "0xd9344493d99153Ad4353D604A1d80d4089004c5D";
var rsETH_ETH_Exchange_Rate = "0x3A917e6B5732dFCc4A45257e3930979fAE6a3737";
var STETH_USD = "0x07C5b924399cc23c24a95c8743DE4006a32b7f2a";
var TUSD_USD = "0x6fAbee62266Da6686EE2744C6f15bb8352d2f28D";
var EURC_USD = "0xCF9752295D0ac9215461fA095faFEC1B854b849B";
var GLV_BTC_USDC_USD = "0xc6984fF1047Ddd0AB52D2dFa78381CC677921A85";
var CHF_USD = "0xe32AccC8c4eC03F6E75bd3621BfC9Fbb234E1FC3";
var inETH_ETH_Exchange_Rate = "0x3C5C5329b028E674F7a124b18527B94BE74A66cF";
var GMX_USD = "0xDB98056FecFff59D032aB628337A4887110df3dB";
var HKD_USD = "0x5082d93991DfCc69F57a27CEC63EDEe1B8C22757";
var DAI_USD = "0xc5C8E77B397E531B8EC06BFb0048328B30E9eCfB";
var BONE_USD = "0x8b7C8726F93063b88Db512f34b90291AEB1E884B";
var XAI_USD = "0x806c532D543352e7C344ba6C7F3F00Bfbd309Af1";
var CRVUSD_USD = "0x0a32255dd4BB6177C994bAAc73E0606fDD568f66";
var LINK_ETH = "0xb7c8Fb1dB45007F98A68Da0588e1AA524C317f27";
var Real_GDP_Percent_Change_Annual_Rate = "0xDC1a8F77Cdc46D1Fb7F22F94dFCA4dDBf566fA23";
var JOE_USD = "0x04180965a782E487d0632013ABa488A472243542";
var ETHx_ETH_Exchange_Rate = "0x1f5C0C2CD2e9Ad1eE475660AF0bBa27aE7d87f5e";
var tETH_wstETH_Exchange_Rate = "0x98a977Ba31C72aeF2e15B950Eb5Ae3158863D856";
var _1INCH_USD = "0x4bC735Ef24bf286983024CAd5D03f0738865Aaef";
var XDC_USD = "0x3b9c34dC5B6BfC1dE896496aFCEf1ca705D0c762";
var JPY_USD = "0x3dD6e51CB9caE717d5a8778CF79A04029f9cFDF8";
var WBTC_USD = "0xd0C7101eACbB49F3deCcCc166d238410D6D46d57";
var wstETH_stETH_Exchange_Rate = "0xB1552C5e96B312d0Bf8b554186F846C40614a540";
var CV_Index = "0xbcD8bEA7831f392bb019ef3a672CC15866004536";
var APT_USD = "0xdc49F292ad1bb3DAb6C11363d74ED06F38b9bd9C";
var NEAR_USD = "0xBF5C3fB2633e924598A46B9D07a174a9DBcF57C0";
var DOGE_USD = "0x9A7FB1b3950837a8D9b40517626E11D4127C098C";
var CBETH_ETH = "0xa668682974E3f121185a3cD94f00322beC674275";
var RON_USD = "0x29D57534598BF8Adda2CC2FbDe4B7502387B8177";
var DODO_USD = "0xA33a06c119EC08F92735F9ccA37e07Af08C4f281";
var MIMATIC_USD = "0x59644ec622243878d1464A9504F9e9a31294128a";
var USDL_USD = "0x18DdDDEe3682853c746E5203e0495CAaC5054a75";
var RLP_USD_Exchange_Rate = "0x3861d83ed006A6570cE8Eb9878F4Df0dF889A2da";
var ORDI_USD = "0x76998C22eEa325A11dc6971Cedcf533E9740F854";
var uniBTC_BTC_Exchange_Rate = "0xa677622735c38001fd1183c92dc5b889128EC234";
var AVAX_USD = "0x8bf61728eeDCE2F32c456454d87B5d6eD6150208";
var SEK_USD = "0xdE89a55d04DEd40A410877ab87d4F567ee66a1f0";
var SEI_USD = "0xCc9742d77622eE9abBF1Df03530594f9097bDcB3";
var XAU_USD = "0x1F954Dc24a49708C26E0C1777f16750B5C6d5a2c";
var JLP_USD = "0x702609AFaDda5b357bc7b0C5174645a4438A99F3";
var ASTR_USD = "0x70E48a135F76bA31B47FE944e769E052A8FeB849";
var USDD_USD = "0x4Ee1f9ec1048979930aC832a3C1d18a0b4955a02";
var POL_USD = "0x82BA56a2fADF9C14f17D08bc51bDA0bDB83A8934";
var sFRAX_FRAX_Exchange_Rate = "0x03e4054B11ad01915257bE53Af03A32Abf7837b9";
var SUSHI_USD = "0xb2A8BA74cbca38508BA1632761b56C897060147C";
var apxETH_pxETH_Exchange_Rate = "0x41f8459f911658e9C627c5D1d2Fb18c7dbc8ceBe";
var CORE_USD = "0x037100C673E8A63fBF8F6f78A8FBeEbb683593a8";
var DPI_USD = "0x1e431E56118bE414bD91f6392414ad3833d21B0D";
var RSETH_ETH = "0xb0EA543f9F8d4B818550365d13F66Da747e1476A";
var GM_BTC_USD_WBTC_WBTC = "0x19eCDd6DDc12597ec4A522fB1E25b1A580B605B7";
var PCE_Price_Index_Percent_Change_Annual_Rate = "0x552c92c0C09A73d569Ab52FF4C17a153C3472BE3";
var gmETH_USD = "0xfB3264D1129824933a52374c2C1696F4470D041e";
var ADA_USD = "0xD9f615A9b820225edbA2d821c4A696a0924051c6";
var UNI_USD = "0x9C917083fDb403ab5ADbEC26Ee294f6EcAda2720";
var ynETHx_ETH_Exchange_Rate = "0xF6897122530ca89beB919f5990FDd3F7Cc9837C4";
var M_NAV = "0xD2385f41995499bBF255cE0A1486ebCe3Dca0e5D";
var ultraETHs_ETH_Exchange_Rate = "0x20b0159404886F4BFE51A1ad9c14cFc3Cde5995c";
var GM_ETH_USD_wETH_wETH = "0xEAeFFF521cb36dFb414E8580f8635BFB44d96255";
var PCE_Price_Index_Level = "0x066Ba4567C40cB315EAd257Bd86585bb4a74004f";
var SPELL_USD = "0x383b3624478124697BEF675F07cA37570b73992f";
var FXS_USD = "0x36a121448D74Fa81450c992A1a44B9b7377CD3a5";
var WEMIX_USD = "0x6FfBc6339DD46a7e0513D4887106349214C05505";
var GAS_USD = "0xCF7a7A1Fb746E926Bbe5C88a702a89167a94C51C";
var frxETH_ETH_Exchange_Rate_Low = "0x1bD872f3A606471787B1a304cE0356e4e87Af930";
var CHAIN_ID = 42161;
export {
  AAPL_USD,
  AAVE_USD,
  ADA_USD,
  AMZN_USD,
  ANON_USD,
  APE_USD,
  APT_USD,
  ARB_USD,
  ASTR_USD,
  ATH_USD,
  ATOM_USD,
  AUD_USD,
  AVAX_USD,
  AXL_USD,
  AXS_USD,
  BAL_USD,
  BERA_USD,
  BNB_USD,
  BONE_USD,
  BRL_USD,
  BTC_ETH,
  BTC_USD,
  BTC_USD_Total_Marketcap,
  CAD_USD,
  CAKE_USD,
  CBETH_ETH,
  CHAIN_ID,
  CHF_USD,
  CNY_USD,
  COIN_USD,
  COMP_USD,
  CORE_USD,
  CRO_USD,
  CRVUSD_USD,
  CRV_USD,
  CVX_USD,
  CV_Index,
  DAI_USD,
  DODO_USD,
  DOGE_USD,
  DOLO_USD,
  DOT_USD,
  DPI_USD,
  ENA_USD,
  ETH_USD,
  ETH_USD_Total_Marketcap,
  ETHx_ETH,
  ETHx_ETH_Exchange_Rate,
  EURC_USD,
  EUR_USD,
  FBTC_BTC_Exchange_Rate,
  FBTC_Proof_of_Reserves,
  FRAX_USD,
  FTM_USD,
  FXS_USD,
  GAS_USD,
  GBP_USD,
  GHO_USD,
  GLV_BTC_USDC_USD,
  GLV_WETH_USDC_USD,
  GMX_USD,
  GM_BTC_USD_WBTC_WBTC,
  GM_ETH_USD_wETH_wETH,
  GNS_USD,
  GOOGL_USD,
  GRASS_USD,
  GRT_USD,
  HKD_USD,
  HSK_USD,
  HYPE_USD,
  IOTX_USD,
  JLP_USD,
  JOE_USD,
  JPY_USD,
  KNC_USD,
  KRW_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LBTC_BTC,
  LDO_USD,
  LINK_ETH,
  LINK_USD,
  LTC_USD,
  LUSD_USD,
  MAGIC_USD,
  MATIC_USD,
  MELANIA_USD,
  META_USD,
  MIMATIC_USD,
  MIM_USD,
  MKR_USD,
  MLN_USD,
  MNT_USD,
  MSFT_USD,
  MVI_USD,
  M_NAV,
  NEAR_USD,
  NVDA_USD,
  OP_USD,
  ORDER_USD,
  ORDI_USD,
  PAXG_USD,
  PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate,
  PENDLE_USD,
  PEPE_USD,
  PHP_USD,
  PLUME_USD,
  POL_USD,
  PUMP_USD,
  RDNT_USD,
  RETH_ETH,
  RLP_USD_Exchange_Rate,
  RON_USD,
  RPL_USD,
  RSETH_ETH,
  RSR_USD,
  RYT_Arbitrum_Proof_of_Reserves,
  RYT_NAV,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate,
  SEI_USD,
  SEK_USD,
  SGD_USD,
  SHIB_ETH,
  SHIB_USD,
  SNX_USD,
  SOL_USD,
  SPELL_USD,
  SPY_USD,
  STETH_ETH,
  STETH_USD,
  STG_USD,
  STX_USD,
  SUI_USD,
  SUSHI_USD,
  SWETH_ETH,
  StaFi_Staked_ETH_rETH_ETH_Exchange_Rate,
  TAO_USD,
  TBTC_BTC,
  TBTC_USD,
  TIA_USD,
  TRUMP_USD,
  TRY_USD,
  TSLA_USD,
  TUSD_USD,
  Total_Marketcap_USD,
  ULTI_USD,
  UNI_USD,
  USD1_USD,
  USDC_USD,
  USDD_USD,
  USDG_USD,
  USDL_USD,
  USDPlus_USD,
  USDS_USD,
  USDT_USD,
  USDX_USD,
  USDe_USD,
  USR_USD,
  USR_USD_Exchange_Rate,
  WBTC_BTC,
  WBTC_USD,
  WEMIX_USD,
  WIF_USD,
  WLFI_USD,
  WOO_USD,
  WSTETH_ETH,
  WSTUSR_STUSR_Exchange_Rate,
  WTI_USD,
  XAG_USD,
  XAI_USD,
  XAU_USD,
  XDC_USD,
  XRP_USD,
  XTZ_USD,
  XVS_USD,
  YBTC_BTC_Exchange_Rate,
  YETH_ETH_Exchange_Rate,
  YFI_USD,
  ZAR_USD,
  ZRO_USD,
  _1INCH_USD,
  aBTC_Proof_of_Reserves,
  agETH_ETH_Exchange_Rate,
  ankrETH_ETH_Exchange_Rate,
  apxETH_pxETH_Exchange_Rate,
  cbETH_ETH_Exchange_Rate,
  deUSD_USD,
  dlcBTC_Proof_of_Reserves,
  egETH_ETH_Exchange_Rate,
  ezETH_ETH,
  ezETH_ETH_Exchange_Rate,
  frxETH_ETH_Exchange_Rate_High,
  frxETH_ETH_Exchange_Rate_Low,
  frxUSD_USD,
  gmARB_USD,
  gmBTC_USD,
  gmETH_USD,
  iBTC_Proof_of_Reserves,
  ibBTC_PricePerShare,
  inETH_ETH_Exchange_Rate,
  instETH_ETH_Exchange_Rate,
  osETH_ETH_Exchange_Rate,
  rETH_ETH_Exchange_Rate,
  rsETH_ETH_Exchange_Rate,
  rswETH_ETH_Exchange_Rate,
  sFRAX_FRAX_Exchange_Rate,
  sUSDS_USDS_Exchange_Rate,
  sUSDe_USD,
  sUSDe_USDe_Exchange_Rate,
  sfrxETH_frxETH_Exchange_Rate,
  solvBTC_BTC,
  stBTC_BTC_Exchange_Rate,
  stBTC_Proof_of_Reserves,
  swETH_ETH_Exchange_Rate,
  syrupUSDC_USDC_Exchange_Rate,
  tETH_wstETH_Exchange_Rate,
  ultraETHs_ETH_Exchange_Rate,
  uniBTC_BTC_Exchange_Rate,
  uniETH_ETH_Exchange_Rate,
  vyUSD_USD_Exchange_Rate,
  wOETH_OETH_Exchange_Rate,
  weETH_ETH,
  weETH_eETH_Exchange_Rate,
  wrsETH_rsETH_Exchange_Rate,
  wstETH_stETH_Exchange_Rate,
  wstUSR_stUSR_Exchange_Rate,
  xSolvBTC_NAV,
  yUSD_USD_Exchange_Rate,
  ynETH_ETH_Exchange_Rate,
  ynETHx_ETH_Exchange_Rate,
  zBTC_Proof_of_Reserves
};
