// @bgd-labs/aave-address-book/dist/ChainlinkEthereum.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkEthereum.ts
var ChainlinkEthereum_exports = {};
__export(ChainlinkEthereum_exports, {
  AAVE_ETH: () => AAVE_ETH,
  AAVE_USD: () => AAVE_USD,
  ALCX_ETH: () => ALCX_ETH,
  AMPL_ETH: () => AMPL_ETH,
  AMPL_USD: () => AMPL_USD,
  APE_ETH: () => APE_ETH,
  APE_USD: () => APE_USD,
  ARB_USD: () => ARB_USD,
  ATH_USD: () => ATH_USD,
  AUD_USD: () => AUD_USD,
  AVAIL_USD: () => AVAIL_USD,
  AVAX_USD: () => AVAX_USD,
  Arbitrum_Healthcheck: () => Arbitrum_Healthcheck,
  BADGER_ETH: () => BADGER_ETH,
  BAL_ETH: () => BAL_ETH,
  BAL_USD: () => BAL_USD,
  BAT_ETH: () => BAT_ETH,
  BAT_USD: () => BAT_USD,
  BGBTC_Proof_of_Reserves: () => BGBTC_Proof_of_Reserves,
  BNB_USD: () => BNB_USD,
  BTC_ETH: () => BTC_ETH,
  BTC_USD: () => BTC_USD,
  BUIDL_NAV: () => BUIDL_NAV,
  BUSD_USD: () => BUSD_USD,
  Base_Healthcheck: () => Base_Healthcheck,
  C3M_EUR: () => C3M_EUR,
  CAD_USD: () => CAD_USD,
  CBETH_ETH: () => CBETH_ETH,
  CETH_Reserves: () => CETH_Reserves,
  CHAIN_ID: () => CHAIN_ID,
  CHF_USD: () => CHF_USD,
  CNY_USD: () => CNY_USD,
  COMP_ETH: () => COMP_ETH,
  COMP_USD: () => COMP_USD,
  CRVUSD_USD: () => CRVUSD_USD,
  CRV_ETH: () => CRV_ETH,
  CRV_USD: () => CRV_USD,
  CSPX_USD: () => CSPX_USD,
  CVX_ETH: () => CVX_ETH,
  CVX_USD: () => CVX_USD,
  Calculated_XSUSHI_ETH: () => Calculated_XSUSHI_ETH,
  Celo_Healthcheck: () => Celo_Healthcheck,
  Consumer_Price_Index: () => Consumer_Price_Index,
  DAI_ETH: () => DAI_ETH,
  DAI_USD: () => DAI_USD,
  DOLO_USD: () => DOLO_USD,
  DPI_ETH: () => DPI_ETH,
  DPI_USD: () => DPI_USD,
  EIGEN_USD: () => EIGEN_USD,
  ENJ_ETH: () => ENJ_ETH,
  ENS_USD: () => ENS_USD,
  ETHFI_USD: () => ETHFI_USD,
  ETH_BTC: () => ETH_BTC,
  ETH_USD: () => ETH_USD,
  ETHx_ETH: () => ETHx_ETH,
  EURC_USD: () => EURC_USD,
  EUR_USD: () => EUR_USD,
  EUTBL_NAV: () => EUTBL_NAV,
  FBTC_BTC_Exchange_Rate: () => FBTC_BTC_Exchange_Rate,
  FBTC_Proof_of_Reserves: () => FBTC_Proof_of_Reserves,
  FDUSD_USD: () => FDUSD_USD,
  FIL_ETH: () => FIL_ETH,
  FRAX_ETH: () => FRAX_ETH,
  FRAX_USD: () => FRAX_USD,
  FTM_ETH: () => FTM_ETH,
  FTT_ETH: () => FTT_ETH,
  FXS_USD: () => FXS_USD,
  Fast_Gas_Gwei: () => Fast_Gas_Gwei,
  GBP_USD: () => GBP_USD,
  GHO_USD: () => GHO_USD,
  GRT_ETH: () => GRT_ETH,
  GRT_USD: () => GRT_USD,
  HBTC_Proof_of_Reserves: () => HBTC_Proof_of_Reserves,
  HIGH_USD: () => HIGH_USD,
  IB01_USD: () => IB01_USD,
  IBTA_USD: () => IBTA_USD,
  IDR_USD: () => IDR_USD,
  IMX_USD: () => IMX_USD,
  Ink_Healthcheck: () => Ink_Healthcheck,
  JAAA_NAV_Aave_LlamaRisk: () => JAAA_NAV_Aave_LlamaRisk,
  JPY_USD: () => JPY_USD,
  JTRSY_NAV_Aave_LlamaRisk: () => JTRSY_NAV_Aave_LlamaRisk,
  KNC_ETH: () => KNC_ETH,
  KNC_USD: () => KNC_USD,
  KRW_USD: () => KRW_USD,
  LBTC_BTC: () => LBTC_BTC,
  LDO_ETH: () => LDO_ETH,
  LINK_ETH: () => LINK_ETH,
  LINK_USD: () => LINK_USD,
  LRC_ETH: () => LRC_ETH,
  LUSD_USD: () => LUSD_USD,
  LsETH_ETH_Exchange_Rate: () => LsETH_ETH_Exchange_Rate,
  MANA_ETH: () => MANA_ETH,
  MATIC_USD: () => MATIC_USD,
  MAVIA_USD: () => MAVIA_USD,
  MIM_USD: () => MIM_USD,
  MKR_ETH: () => MKR_ETH,
  MKR_USD: () => MKR_USD,
  MLN_ETH: () => MLN_ETH,
  MUSD_USD: () => MUSD_USD,
  M_NAV: () => M_NAV,
  Mantle_Healthcheck: () => Mantle_Healthcheck,
  Metis_Healthcheck: () => Metis_Healthcheck,
  NEIRO_USD: () => NEIRO_USD,
  NZD_USD: () => NZD_USD,
  Nexus_wETH_Reserves: () => Nexus_wETH_Reserves,
  OETH_ETH: () => OETH_ETH,
  OHMv2_ETH: () => OHMv2_ETH,
  OUSDT_USD: () => OUSDT_USD,
  Optimism_Healthcheck: () => Optimism_Healthcheck,
  PAXG_USD: () => PAXG_USD,
  PCE_Price_Index_Level: () => PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate: () => PCE_Price_Index_Percent_Change_Annual_Rate,
  PERP_ETH: () => PERP_ETH,
  PHP_USD: () => PHP_USD,
  PYUSD_USD: () => PYUSD_USD,
  PumpBTC_BTC_Exchange_Rate: () => PumpBTC_BTC_Exchange_Rate,
  PumpBTC_Proof_of_Reserves: () => PumpBTC_Proof_of_Reserves,
  RAI_ETH: () => RAI_ETH,
  RDNT_USD: () => RDNT_USD,
  REN_ETH: () => REN_ETH,
  RETH_ETH: () => RETH_ETH,
  RLUSD_USD: () => RLUSD_USD,
  RPL_USD: () => RPL_USD,
  RSETH_ETH: () => RSETH_ETH,
  RSR_USD: () => RSR_USD,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level: () => Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate: () => Real_GDP_Percent_Change_Annual_Rate,
  SAND_USD: () => SAND_USD,
  SGD_USD: () => SGD_USD,
  SHIB_ETH: () => SHIB_ETH,
  SHV_USD: () => SHV_USD,
  SKY_USD: () => SKY_USD,
  SNX_ETH: () => SNX_ETH,
  SNX_USD: () => SNX_USD,
  SOLVBTC_BTC_Exchange_Rate: () => SOLVBTC_BTC_Exchange_Rate,
  SOL_USD: () => SOL_USD,
  SPELL_USD: () => SPELL_USD,
  STBT_Proof_of_Reserves: () => STBT_Proof_of_Reserves,
  STETH_ETH: () => STETH_ETH,
  STETH_USD: () => STETH_USD,
  STG_USD: () => STG_USD,
  SUSD1Plus_USD1_Exchange_Rate: () => SUSD1Plus_USD1_Exchange_Rate,
  SUSD_ETH: () => SUSD_ETH,
  SUSHI_ETH: () => SUSHI_ETH,
  SUSHI_USD: () => SUSHI_USD,
  SVR_AAVE_USD: () => SVR_AAVE_USD,
  SVR_AAVE_USD_BASE: () => SVR_AAVE_USD_BASE,
  SVR_BTC_USD: () => SVR_BTC_USD,
  SVR_BTC_USD_BASE: () => SVR_BTC_USD_BASE,
  SVR_COMP_USD: () => SVR_COMP_USD,
  SVR_COMP_USD_BASE: () => SVR_COMP_USD_BASE,
  SVR_CRCLon_USD: () => SVR_CRCLon_USD,
  SVR_CRCLon_USD_BASE: () => SVR_CRCLon_USD_BASE,
  SVR_ETH_USD: () => SVR_ETH_USD,
  SVR_ETH_USD_BASE: () => SVR_ETH_USD_BASE,
  SVR_LINK_USD: () => SVR_LINK_USD,
  SVR_LINK_USD_BASE: () => SVR_LINK_USD_BASE,
  SVR_QQQon_USD: () => SVR_QQQon_USD,
  SVR_QQQon_USD_BASE: () => SVR_QQQon_USD_BASE,
  SVR_SPYon_USD: () => SVR_SPYon_USD,
  SVR_SPYon_USD_BASE: () => SVR_SPYon_USD_BASE,
  SVR_TSLAon_USD: () => SVR_TSLAon_USD,
  SVR_TSLAon_USD_BASE: () => SVR_TSLAon_USD_BASE,
  SVR_USDC_USD: () => SVR_USDC_USD,
  SVR_USDC_USD_BASE: () => SVR_USDC_USD_BASE,
  SVR_USDT_USD: () => SVR_USDT_USD,
  SVR_USDT_USD_BASE: () => SVR_USDT_USD_BASE,
  SWELL_ETH: () => SWELL_ETH,
  SWETH_ETH: () => SWETH_ETH,
  SXT_USD: () => SXT_USD,
  Scroll_Healthcheck: () => Scroll_Healthcheck,
  Secured_Overnight_Financing_Rate: () => Secured_Overnight_Financing_Rate,
  SolvBTCBBN_SolvBTC_Exchange_Rate: () => SolvBTCBBN_SolvBTC_Exchange_Rate,
  SolvBTC_Proof_of_Reserves: () => SolvBTC_Proof_of_Reserves,
  Soneium_Healthcheck: () => Soneium_Healthcheck,
  Swell_ETH_Proof_of_Reserves: () => Swell_ETH_Proof_of_Reserves,
  Swell_Restaked_ETH_Proof_of_Reserves: () => Swell_Restaked_ETH_Proof_of_Reserves,
  Synthetix_Aggregator_Debt_Ratio: () => Synthetix_Aggregator_Debt_Ratio,
  Synthetix_Aggregator_Issued_Synths: () => Synthetix_Aggregator_Issued_Synths,
  TAO_USD: () => TAO_USD,
  TBILL_NAV: () => TBILL_NAV,
  TBTC_USD: () => TBTC_USD,
  TRUMP_USD: () => TRUMP_USD,
  TRY_USD: () => TRY_USD,
  TUSD_ETH: () => TUSD_ETH,
  TUSD_Reserves: () => TUSD_Reserves,
  TUSD_USD: () => TUSD_USD,
  Total_Marketcap_USD: () => Total_Marketcap_USD,
  UNI_ETH: () => UNI_ETH,
  UNI_USD: () => UNI_USD,
  USCC_NAV_Aave_LlamaRisk: () => USCC_NAV_Aave_LlamaRisk,
  USCC_NAV_per_Share: () => USCC_NAV_per_Share,
  USD0PlusPlus_USD: () => USD0PlusPlus_USD,
  USD0_USD: () => USD0_USD,
  USD1_USD: () => USD1_USD,
  USDC_ETH: () => USDC_ETH,
  USDC_USD: () => USDC_USD,
  USDG_USD: () => USDG_USD,
  USDL_USD: () => USDL_USD,
  USDP_USD: () => USDP_USD,
  USDS_USD: () => USDS_USD,
  USDT_ETH: () => USDT_ETH,
  USDT_USD: () => USDT_USD,
  USDe_USD: () => USDe_USD,
  USDf_USD: () => USDf_USD,
  USDtb_USD: () => USDtb_USD,
  USR_USD: () => USR_USD,
  USTBL_NAV: () => USTBL_NAV,
  USTB_AUM: () => USTB_AUM,
  USTB_NAV_Aave_LlamaRisk: () => USTB_NAV_Aave_LlamaRisk,
  USTB_NAV_per_Share: () => USTB_NAV_per_Share,
  UST_ETH: () => UST_ETH,
  USYC_NAV_Aave_LlamaRisk: () => USYC_NAV_Aave_LlamaRisk,
  Unichain_Healthcheck: () => Unichain_Healthcheck,
  WAMPL_USD: () => WAMPL_USD,
  WBTC_BTC: () => WBTC_BTC,
  WBTC_Proof_of_Reserves: () => WBTC_Proof_of_Reserves,
  WLFI_USD: () => WLFI_USD,
  WTGXX_NAV: () => WTGXX_NAV,
  XAG_USD: () => XAG_USD,
  XAU_USD: () => XAU_USD,
  XCN_USD: () => XCN_USD,
  YFI_ETH: () => YFI_ETH,
  YFI_USD: () => YFI_USD,
  ZBU_USD: () => ZBU_USD,
  ZKsync_Healthcheck: () => ZKsync_Healthcheck,
  ZRX_ETH: () => ZRX_ETH,
  ZRX_USD: () => ZRX_USD,
  _1INCH_ETH: () => _1INCH_ETH,
  _1INCH_USD: () => _1INCH_USD,
  _21BTC_Proof_of_Reserves: () => _21BTC_Proof_of_Reserves,
  cUSD_AUM: () => cUSD_AUM,
  cbBTC_Reserves: () => cbBTC_Reserves,
  cbBTC_USD: () => cbBTC_USD,
  deUSD_USD: () => deUSD_USD,
  eETH_Proof_of_Reserves: () => eETH_Proof_of_Reserves,
  ezETH_ETH: () => ezETH_ETH,
  frxUSD_USD: () => frxUSD_USD,
  goUSD_Reserves: () => goUSD_Reserves,
  lvlUSD_USD: () => lvlUSD_USD,
  mETH_ETH: () => mETH_ETH,
  pufETH_ETH: () => pufETH_ETH,
  rsETH_ETH_Exchange_Rate: () => rsETH_ETH_Exchange_Rate,
  rswETH_ETH: () => rswETH_ETH,
  sUSDe_USD: () => sUSDe_USD,
  sUSDe_USD_Calculated: () => sUSDe_USD_Calculated,
  sUSDf_USDf_Exchange_Rate: () => sUSDf_USDf_Exchange_Rate,
  solvBTC_BTC: () => solvBTC_BTC,
  stBTC_BTC_Exchange_Rate: () => stBTC_BTC_Exchange_Rate,
  stBTC_Proof_of_Reserves: () => stBTC_Proof_of_Reserves,
  stETH_Proof_of_Reserves: () => stETH_Proof_of_Reserves,
  tETH_wstETH_Exchange_Rate: () => tETH_wstETH_Exchange_Rate,
  uniBTC_BTC_Exchange_Rate: () => uniBTC_BTC_Exchange_Rate,
  uniBTC_Proof_of_Reserves: () => uniBTC_Proof_of_Reserves,
  weETH_ETH: () => weETH_ETH,
  wstETH_USD_Calculated: () => wstETH_USD_Calculated,
  xSolvBTC_NAV: () => xSolvBTC_NAV,
  xSolvBTC_Proof_of_Reserves: () => xSolvBTC_Proof_of_Reserves
});
module.exports = __toCommonJS(ChainlinkEthereum_exports);
var FBTC_Proof_of_Reserves = "0x75A0d3264a949C2C920d7F25Df174af1FaF73399";
var RAI_ETH = "0x4ad7B025127e89263242aB68F0f9c4E5C033B489";
var rswETH_ETH = "0xb613CfebD0b6e95abDDe02677d6bC42394FdB857";
var FIL_ETH = "0x0606Be69451B1C9861Ac6b3626b99093b713E801";
var ETHFI_USD = "0x19678515847d8DE85034dAD0390e09c3048d31cd";
var USDf_USD = "0xb177857a1799aA5F7fEb5799Fdf12CbE8fdF78B1";
var deUSD_USD = "0x471a6299C027Bd81ed4D66069dc510Bd0569f4F8";
var FDUSD_USD = "0xfAA9147190c2C2cc5B8387B4f49016bDB3380572";
var rsETH_ETH_Exchange_Rate = "0x9d2F2f96B24C444ee32E57c04F7d944bcb8c8549";
var UNI_ETH = "0xD6aA3D25116d8dA79Ea0246c4826EB951872e02e";
var NEIRO_USD = "0x771cf56aE75bC907193177237b423A9DA831280A";
var USDT_ETH = "0xEe9F2375b4bdF6387aa8265dD4FB8F16512A1d46";
var ezETH_ETH = "0x636A000262F6aA9e1F094ABF0aD8f645C44f641C";
var Scroll_Healthcheck = "0x7fb9B4a05e7B4F0c1Ac0B0046784cc0aCE8CbBC5";
var CSPX_USD = "0xF4E1B57FB228879D057ac5AE33973e8C53e4A0e0";
var weETH_ETH = "0x5c9C449BbC9a6075A2c061dF312a35fd1E05fF22";
var JAAA_NAV_Aave_LlamaRisk = "0x1E41Ef40AC148706c114534E8192Ca608f80fC48";
var BAT_ETH = "0x0d16d4528239e9ee52fa531af613AcdB23D88c94";
var stBTC_Proof_of_Reserves = "0x7Eaa6f116Ab9E57f58d9A9Ce88cc3f1c2476ECfA";
var Synthetix_Aggregator_Debt_Ratio = "0x0981af0C002345c9C5AD5efd26242D0cBe5aCA99";
var USDT_USD = "0x3E7d1eAB13ad0104d2750B8863b489D65364e32D";
var XAU_USD = "0x214eD9Da11D2fbe465a6fc601a91E62EbEc1a0D6";
var SUSHI_ETH = "0xe572CeF69f43c2E488b33924AF04BDacE19079cf";
var KNC_USD = "0xf8fF43E991A81e6eC886a3D281A2C6cC19aE70Fc";
var TUSD_Reserves = "0xBE456fd14720C3aCCc30A2013Bffd782c9Cb75D5";
var Real_GDP_Percent_Change_Annual_Rate = "0x1e9cb00c0AC8d2a171f44E63bE7532AD7224F6D1";
var AVAX_USD = "0xFF3EEb22B5E3dE6e705b44749C2559d704923FD7";
var C3M_EUR = "0xD41390267Afec3fA5b4c0B3aA6c706556CCE75ec";
var PERP_ETH = "0x3b41D5571468904D4e53b6a8d93A6BaC43f02dC9";
var CBETH_ETH = "0xF017fcB346A1885194689bA23Eff2fE6fA5C483b";
var BAT_USD = "0x9441D7556e7820B5ca42082cfa99487D56AcA958";
var COMP_ETH = "0x1B39Ee86Ec5979ba5C322b826B3ECb8C79991699";
var PCE_Price_Index_Level = "0x2F1494543bffb3022bf8Cb18C251D2286C98A85F";
var Synthetix_Aggregator_Issued_Synths = "0xbCF5792575bA3A875D8C406F4E7270f51a902539";
var Optimism_Healthcheck = "0x59c2287c8E848310c809C061a1Be0d1556eFF4e2";
var COMP_USD = "0xdbd020CAeF83eFd542f4De03e3cF0C28A4428bd5";
var KRW_USD = "0x01435677FB11763550905594A16B645847C1d0F3";
var USDC_ETH = "0x986b5E1e1755e3C2440e960477f25201B0a8bbD4";
var STETH_USD = "0xCfE54B5cD566aB89272946F602D76Ea879CAb4a8";
var DPI_USD = "0xD2A593BF7594aCE1faD597adb697b5645d5edDB2";
var BAL_USD = "0xdF2917806E30300537aEB49A7663062F4d1F2b5F";
var _1INCH_ETH = "0x72AFAECF99C9d9C8215fF44C77B94B99C28741e8";
var Calculated_XSUSHI_ETH = "0xF05D9B6C08757EAcb1fbec18e36A1B7566a13DEB";
var MAVIA_USD = "0x29d26C008e8f201eD0D864b1Fd9392D29d0C8e96";
var LINK_ETH = "0xDC530D9457755926550b59e8ECcdaE7624181557";
var AAVE_ETH = "0x6Df09E975c830ECae5bd4eD9d90f3A95a4f88012";
var ZRX_ETH = "0x2Da4983a622a8498bb1a21FaE9D8F6C664939962";
var LUSD_USD = "0x3D7aE7E594f2f2091Ad8798313450130d0Aba3a0";
var tETH_wstETH_Exchange_Rate = "0x7B2Fb2c667af80Bccc0B2556378352dFDE2be914";
var TAO_USD = "0x1c88503c9A52aE6aaE1f9bb99b3b7e9b8Ab35459";
var AUD_USD = "0x77F9710E7d0A19669A13c055F62cd80d313dF022";
var PYUSD_USD = "0x8f1dF6D7F2db73eECE86a18b4381F4707b918FB1";
var USCC_NAV_Aave_LlamaRisk = "0x19e2d716288751c5A59deaB61af012D5DF895962";
var Real_GDP_Level = "0x2b94a8B3E478a2984A0b50a4bB0F19827Ad0cef3";
var XCN_USD = "0xeb988B77b94C186053282BfcD8B7ED55142D3cAB";
var LBTC_BTC = "0x5c29868C58b6e15e2b962943278969Ab6a7D3212";
var ZBU_USD = "0x617689cAB8329d57fEa64f4C086190E6797b8B5e";
var cbBTC_Reserves = "0xcBe87Dc0Cf9d807848a3E703B01A90B28eCFb2a7";
var solvBTC_BTC = "0x936B31C428C29713343E05D631e69304f5cF5f49";
var Mantle_Healthcheck = "0x88E588b2f3C2224fBd9441C49eeF61761a3449e0";
var RSR_USD = "0x759bBC1be8F90eE6457C44abc7d443842a976d02";
var SVR_COMP_USD = "0x69B50fF403E995d9c4441a303438D9049dAC8cCD";
var SVR_COMP_USD_BASE = "0x203e994f3908cF886C6155c31742557D82c9B4a2";
var AVAIL_USD = "0xEBca574f1cE4d17cd02c20F47Ef8210C08Cc4255";
var ALCX_ETH = "0x194a9AaF2e0b67c35915cD01101585A33Fe25CAa";
var SVR_CRCLon_USD = "0x393899FeF73E3C08100BF9b8a78bA84769Fa5B96";
var SVR_CRCLon_USD_BASE = "0xC0457F67cac4eb0567A208955C332897a597A207";
var SVR_TSLAon_USD = "0xED4E679adAFA9aBC97a3fA797dEE000d7b0ed247";
var SVR_TSLAon_USD_BASE = "0x737401E0D1299D8A85b653Fd52823501f4FE0be0";
var SVR_LINK_USD = "0xC7e9b623ed51F033b32AE7f1282b1AD62C28C183";
var SVR_LINK_USD_BASE = "0x76F8C9E423C228E83DCB11d17F0Bd8aEB0Ca01bb";
var BTC_USD = "0xF4030086522a5bEEa4988F8cA5B36dbC97BeE88c";
var SWETH_ETH = "0xec21B3e882CE09928cb397DcfF31B15cBBD1e1C3";
var GRT_ETH = "0x17D054eCac33D91F7340645341eFB5DE9009F1C1";
var LRC_ETH = "0x160AC928A16C93eD4895C2De6f81ECcE9a7eB7b4";
var pufETH_ETH = "0xDe3f7Dd92C4701BCf59F47235bCb61e727c45f80";
var PumpBTC_Proof_of_Reserves = "0xe80baC615c38D01c101B0B1b943b05E20C8c2f76";
var Total_Marketcap_USD = "0xEC8761a0A73c34329CA5B1D3Dc7eD07F30e836e2";
var YFI_USD = "0xA027702dbb89fbd58938e4324ac03B58d812b0E1";
var MIM_USD = "0x7A364e8770418566e3eb2001A96116E6138Eb32F";
var Metis_Healthcheck = "0x3425455fe737cdaE8564640df27bbF2eCD56E584";
var TUSD_ETH = "0x3886BA987236181D98F2401c507Fb8BeA7871dF2";
var USDtb_USD = "0x66704DAD467A7cA508B3be15865D9B9F3E186c90";
var DOLO_USD = "0x5c7a079dbf188dAbc66e389482849b05ACB17421";
var GBP_USD = "0x5c0Ab2d9b5a7ed9f470386e82BB36A3613cDd4b5";
var CHF_USD = "0x449d117117838fFA61263B61dA6301AA2a88B13A";
var USDS_USD = "0xfF30586cD0F29eD462364C7e81375FC0C71219b1";
var EIGEN_USD = "0xf2917e602C2dCa458937fad715bb1E465305A4A1";
var USCC_NAV_per_Share = "0xAfFd8F5578E8590665de561bdE9E7BAdb99300d9";
var eETH_Proof_of_Reserves = "0xC8cd82067eA907EA4af81b625d2bB653E21b5156";
var BGBTC_Proof_of_Reserves = "0xADcc914F882965Ef1B2f1043522b3B81ED081491";
var Consumer_Price_Index = "0x9a51192e065ECC6BDEafE5e194ce54702DE4f1f5";
var Soneium_Healthcheck = "0x498b4e83001acc364b2a73f827e9aB4B034ea215";
var cbBTC_USD = "0x2665701293fCbEB223D11A08D826563EDcCE423A";
var xSolvBTC_Proof_of_Reserves = "0x461790bDAF5aeD3df6a88cB97Dec42DD0EFA73c0";
var SUSD1Plus_USD1_Exchange_Rate = "0x171e3Ec9377FeBad8d26967f1B1d6383Ae037997";
var frxUSD_USD = "0x9B4a96210bc8D9D55b1908B465D8B0de68B7fF83";
var ENJ_ETH = "0x24D9aB51950F3d62E9144fdC2f3135DAA6Ce8D1B";
var SKY_USD = "0xee10fE5E7aa92dd7b136597449c3d5813cFC5F18";
var USD1_USD = "0xF0d9bb015Cd7BfAb877B7156146dc09Bf461370d";
var WBTC_Proof_of_Reserves = "0xa81FE04086865e63E12dD3776978E49DEEa2ea4e";
var SUSHI_USD = "0xCc70F09A6CC17553b2E31954cD36E4A2d89501f7";
var PAXG_USD = "0x9944D86CEB9160aF5C5feB251FD671923323f8C3";
var _1INCH_USD = "0xc929ad75B72593967DE83E7F7Cda0493458261D9";
var SAND_USD = "0x35E3f7E558C04cE7eEE1629258EcbbA03B36Ec56";
var BUIDL_NAV = "0x6029F53A5Df21E5e70C460F5E4c9Aad20427B090";
var ENS_USD = "0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16";
var MKR_ETH = "0x24551a8Fb2A7211A25a17B1481f043A8a8adC7f2";
var RSETH_ETH = "0x03c68933f7a3F76875C0bc670a58e69294cDFD01";
var SVR_BTC_USD = "0xb41E773f507F7a7EA890b1afB7d2b660c30C8B0A";
var SVR_BTC_USD_BASE = "0x85355da30ee4b35F4B30759Bd49a1EBE3fc41Bdb";
var DAI_USD = "0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9";
var KNC_ETH = "0x656c0544eF4C98A6a98491833A89204Abb045d6b";
var JTRSY_NAV_Aave_LlamaRisk = "0x23adce82907D20c509101E2Af0723A9e16224EFb";
var USR_USD = "0x34ad75691e25A8E9b681AAA85dbeB7ef6561B42c";
var SXT_USD = "0x2D27d9e1b74936D8E83c4BA118F09A4c4a897f62";
var Ink_Healthcheck = "0x9656D3A00402bD9c18EeF89b29ca6810734ABCAd";
var BUSD_USD = "0x833D8Eb16D306ed1FbB5D7A2E019e106B960965A";
var ETH_BTC = "0xAc559F25B1619171CbC396a50854A3240b6A4e99";
var SVR_ETH_USD = "0xc0053f3FBcCD593758258334Dfce24C2A9A673aD";
var SVR_ETH_USD_BASE = "0xd82562bb17557231Cd871e1B2525F3AB8d63D409";
var ETH_USD = "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419";
var RLUSD_USD = "0x26C46B7aD0012cA71F2298ada567dC9Af14E7f2A";
var sUSDf_USDf_Exchange_Rate = "0xe471bc940AA9831a0AeA21E6F40C1A1236EB4BB3";
var SWELL_ETH = "0x2a638b1203a3B62FF003598B7165Fc5cd5b13B00";
var FTM_ETH = "0x2DE7E4a9488488e0058B95854CC2f7955B35dC9b";
var sUSDe_USD_Calculated = "0xeD9960f685C3c4d6aa937E56169a41C19D0aC9c6";
var DPI_ETH = "0x029849bbc0b1d93b85a8b6190e979fd38F5760E2";
var LsETH_ETH_Exchange_Rate = "0xE858728eB31a25C4AcCcE17d01B68dCFC3A0ED2C";
var _21BTC_Proof_of_Reserves = "0x43921Ca0eca1EA69722c048A6afbc2CAd0BB80e9";
var SHV_USD = "0xc04611C43842220fd941515F86d1DDdB15F04e46";
var SOLVBTC_BTC_Exchange_Rate = "0x4e25cfdc5f53c2CdeD13c1472E628eFd35c99C83";
var CNY_USD = "0xeF8A4aF35cd47424672E3C590aBD37FBB7A7759a";
var BAL_ETH = "0xC1438AA3823A6Ba0C159CfA8D98dF5A994bA120b";
var SNX_ETH = "0x79291A9d692Df95334B1a0B3B4AE6bC606782f8c";
var DAI_ETH = "0x773616E4d11A78F511299002da57A0a94577F1f4";
var USTB_AUM = "0xf3Ab3e54DE3b4BCaEbef6C66Cec0a3a88f262757";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Level = "0xA5265f5B54D0AdD0D48BE88BC3CB0Cdb1a54B816";
var APE_USD = "0xD10aBbC76679a20055E167BB80A24ac851b37056";
var FRAX_USD = "0xB9E1E3A9feFf48998E45Fa90847ed4D467E8BcfD";
var stETH_Proof_of_Reserves = "0xAd410E655C0fE4741F573152592eeb766e686CE7";
var HIGH_USD = "0x5C8D8AaB4ffa4652753Df94f299330Bb4479bF85";
var YFI_ETH = "0x7c5d4F8345e66f68099581Db340cd65B078C41f4";
var MANA_ETH = "0x82A44D92D6c329826dc557c5E1Be6ebeC5D5FeB9";
var RDNT_USD = "0x393CC05baD439c9B36489384F11487d9C8410471";
var USD0_USD = "0x7e891DEbD8FA0A4Cf6BE58Ddff5a8ca174FebDCB";
var PumpBTC_BTC_Exchange_Rate = "0x6CE4Ef3689F26edD40ed3ccbE3Cc29dab62C915f";
var RPL_USD = "0x4E155eD98aFE9034b7A5962f6C84c86d869daA9d";
var USDL_USD = "0x8697D23509F328F90135933de0Dde38Eb86d4893";
var Unichain_Healthcheck = "0x5d8017CDC949AA0a6c00999B970892456d4f5fF3";
var SolvBTCBBN_SolvBTC_Exchange_Rate = "0x1f34794A16D644b9810477EbF3f0b3870141E2e3";
var GRT_USD = "0x86cF33a451dE9dc61a2862FD94FF4ad4Bd65A5d2";
var UST_ETH = "0xa20623070413d42a5C01Db2c8111640DD7A5A03a";
var EUR_USD = "0xb49f677943BC038e9857d61E7d053CaA2C1734C1";
var ZKsync_Healthcheck = "0x987517b50230De2042750570D8fE3783ee115975";
var LINK_USD = "0x2c1d072e956AFFC0D435Cb7AC38EF18d24d9127c";
var MLN_ETH = "0xDaeA8386611A157B08829ED4997A8A62B557014C";
var SUSD_ETH = "0x8e0b7e6062272B5eF4524250bFFF8e5Bd3497757";
var OETH_ETH = "0x703118C4CbccCBF2AB31913e0f8075fbbb15f563";
var SPELL_USD = "0x8c110B94C5f1d347fAcF5E1E938AB2db60E3c9a8";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate = "0xB06A26DF4Ab5853AaCCF2d21FAa00B82362684C3";
var STBT_Proof_of_Reserves = "0xad4A9bED9a5E2c1c9a6E43D35Db53c83873dd901";
var FTT_ETH = "0xF0985f7E2CaBFf22CecC5a71282a89582c382EFE";
var USYC_NAV_Aave_LlamaRisk = "0xE8E65Fb9116875012F5990Ecaab290B3531DbeB9";
var WAMPL_USD = "0xAcFCF155a0de611414C510D43446c27a4EE6a758";
var PHP_USD = "0x3C7dB4D25deAb7c89660512C5494Dc9A3FC40f78";
var USTB_NAV_Aave_LlamaRisk = "0xde49c7B5C0E54b1624ED21C7D88bA6593d444Aa0";
var SVR_USDT_USD = "0x62c2ab773B7324ad9e030D777989B3b5d5c54c0A";
var SVR_USDT_USD_BASE = "0x7bB7bF4ca536DbC49545704BFAcaa13633D18718";
var BADGER_ETH = "0x58921Ac140522867bf50b9E009599Da0CA4A2379";
var JPY_USD = "0xBcE206caE7f0ec07b545EddE332A47C2F75bbeb3";
var TRUMP_USD = "0x7f0347903F413a6C4c540f39145E2a2249639931";
var CVX_ETH = "0xC9CbF687f43176B302F03f5e58470b77D07c61c6";
var WBTC_BTC = "0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23";
var EURC_USD = "0x04F84020Fdf10d9ee64D1dcC2986EDF2F556DA11";
var lvlUSD_USD = "0xe5181aA707feD57781e764437f021b633B34659E";
var BNB_USD = "0x14e613AC84a31f709eadbdF89C6CC390fDc9540A";
var Arbitrum_Healthcheck = "0x32EaFC72772821936BCc9b8A32dC394fEFcDBfD9";
var xSolvBTC_NAV = "0x46cE854814ea38A4857AeA23aE7759b3A7970e4a";
var stBTC_BTC_Exchange_Rate = "0xD93571A6201978976e37c4A0F7bE17806f2Feab2";
var XAG_USD = "0x379589227b15F1a12195D3f2d90bBc9F31f95235";
var Base_Healthcheck = "0x48D9DA600EC48DDd6ce7FC1D47D683818e511c81";
var TRY_USD = "0xB09fC5fD3f11Cf9eb5E1C5Dba43114e3C9f477b5";
var IDR_USD = "0x91b99C9b75aF469a71eE1AB528e8da994A5D7030";
var USTBL_NAV = "0x477e363c51Ab0C4D13B22CD6B57D56d4a3Cb7Abe";
var MATIC_USD = "0x7bAC85A8a13A4BcD8abb3eB7d6b4d632c5a57676";
var CVX_USD = "0xd962fC30A72A84cE50161031391756Bf2876Af5D";
var WTGXX_NAV = "0x1DE472E766E87eB993d0C7904367014a913E55ee";
var ETHx_ETH = "0xC5f8c4aB091Be1A899214c0C3636ca33DcA0C547";
var STETH_ETH = "0x86392dC19c0b719886221c78AB11eb8Cf5c52812";
var SVR_QQQon_USD = "0xE6417a3B82438f783DC8fd2B1cB6B0808585030B";
var SVR_QQQon_USD_BASE = "0xE5DF423251c67D85B2D70787Af76069d96BC4D4C";
var CAD_USD = "0xa34317DB73e77d453b1B8d04550c44D10e981C8e";
var SolvBTC_Proof_of_Reserves = "0xda9258AFc797Cd64d1b6FC651051224cdAB1B25E";
var wstETH_USD_Calculated = "0x164b276057258d81941e97B0a900D4C7B358bCe0";
var USDG_USD = "0x14f0737d6b705259e521EA6E9E3506AC78dBd311";
var TBTC_USD = "0x8350b7De6a6a2C1368E7D4Bd968190e13E354297";
var Swell_ETH_Proof_of_Reserves = "0x60cbE8D88EF519cF3C62414D76f50818D211fea1";
var IB01_USD = "0x32d1463EB53b73C095625719Afa544D5426354cB";
var SVR_SPYon_USD = "0xc9c10271B2B76767C385Ac389C05d77c319DC41C";
var SVR_SPYon_USD_BASE = "0x6EcC1b902dB35eAFE95332443802774Fd1D72576";
var uniBTC_BTC_Exchange_Rate = "0x861d15F8a4059cb918bD6F3670adAEB1220B298f";
var Nexus_wETH_Reserves = "0xCc72039A141c6e34a779eF93AEF5eB4C82A893c7";
var FBTC_BTC_Exchange_Rate = "0xe5346a4Fd329768A99455d969724768a00CA63FB";
var M_NAV = "0xC28198Df9aee1c4990994B35ff51eFA4C769e534";
var MUSD_USD = "0xc90E3460424fb8ea79775089E9053113FEE34Ed0";
var OHMv2_ETH = "0x9a72298ae3886221820B1c878d12D872087D3a23";
var Celo_Healthcheck = "0x7Fa89217C9bA5eb1307Aca6B84E99ed9220e076F";
var ATH_USD = "0xd95044Eb392CdE502b1F903Fc978317462049E52";
var PCE_Price_Index_Percent_Change_Annual_Rate = "0x64c4c1BF5ec145CB6095A289249592B5bb007098";
var USD0PlusPlus_USD = "0xFC9e30Cf89f8A00dba3D34edf8b65BCDAdeCC1cB";
var STG_USD = "0x7A9f34a0Aa917D438e9b6E630067062B7F8f6f3d";
var REN_ETH = "0x3147D7203354Dc06D9fd350c7a2437bcA92387a4";
var IBTA_USD = "0xd27e6D02b72eB6FCe04Ad5690C419196B4EF2885";
var Swell_Restaked_ETH_Proof_of_Reserves = "0x0c89c488e763AC2d69cB058CCAC7A8B283EE3DbA";
var Fast_Gas_Gwei = "0x169E633A2D1E6c10dD91238Ba11c4A708dfEF37C";
var SOL_USD = "0x4ffC43a60e009B551865A93d232E33Fce9f01507";
var BTC_ETH = "0xdeb288F737066589598e9214E782fa5A8eD689e8";
var CRV_ETH = "0x8a12Be339B0cD1829b91Adc01977caa5E9ac121e";
var HBTC_Proof_of_Reserves = "0x0A8cD0115B1EE87EbA5b8E06A9a15ED93e230f7a";
var USDP_USD = "0x09023c0DA49Aaf8fc3fA3ADF34C6A7016D38D5e3";
var uniBTC_Proof_of_Reserves = "0xc590D9fb8eE78a0909dFF341ccf717000b7b7fF2";
var NZD_USD = "0x3977CFc9e4f29C184D4675f4EB8e0013236e5f3e";
var CETH_Reserves = "0xa7d76167900493Acf2650Dc001fb2Bc5256579B0";
var FXS_USD = "0x6Ebc52C8C1089be9eB3945C4350B68B8E4C2233f";
var IMX_USD = "0xBAEbEFc1D023c0feCcc047Bff42E75F15Ff213E6";
var FRAX_ETH = "0x14d04Fff8D21bd62987a5cE9ce543d2F1edF5D3E";
var SNX_USD = "0xDC3EA94CD0AC27d9A86C180091e7f78C683d3699";
var RETH_ETH = "0x536218f9E9Eb48863970252233c8F271f554C2d0";
var USDC_USD = "0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6";
var APE_ETH = "0xc7de7f4d4C9c991fF62a07D18b3E31e349833A18";
var USDe_USD = "0xa569d910839Ae8865Da8F8e70FfFb0cBA869F961";
var AMPL_ETH = "0x492575FDD11a0fCf2C6C719867890a7648d526eB";
var SHIB_ETH = "0x8dD1CD88F43aF196ae478e91b9F5E4Ac69A97C61";
var OUSDT_USD = "0xE3fd61Cd8935EAE81e1c31AF36D7134e411490Cd";
var SVR_AAVE_USD = "0xF02C1e2A3B77c1cacC72f72B44f7d0a4c62e4a85";
var SVR_AAVE_USD_BASE = "0xbd7F896e60B650C01caf2d7279a1148189A68884";
var cUSD_AUM = "0x16caE6d6ffb4AE01e206b928de925Ac0C8C8116A";
var goUSD_Reserves = "0x269f871c80b50a5cF34cDfCfEC11460adA4D66f1";
var Secured_Overnight_Financing_Rate = "0xBc8f5e8B47D69fa10A4B64eC5B45b466f81B77CA";
var AMPL_USD = "0xe20CA8D7546932360e37E9D72c1a47334af57706";
var USTB_NAV_per_Share = "0x289B5036cd942e619E1Ee48670F98d214E745AAC";
var AAVE_USD = "0x547a514d5e3769680Ce22B2361c10Ea13619e8a9";
var CRV_USD = "0xCd627aA160A6fA45Eb793D19Ef54f5062F20f33f";
var UNI_USD = "0x553303d460EE0afB37EdFf9bE42922D8FF63220e";
var EUTBL_NAV = "0xfD628af590c4150A9651C1f4ddD0b4f532B703ae";
var sUSDe_USD = "0xFF3BC18cCBd5999CE63E788A1c250a88626aD099";
var SVR_USDC_USD = "0xEa674bBC33AE708Bc9EB4ba348b04E4eB55b496b";
var SVR_USDC_USD_BASE = "0xfB6471ACD42c91FF265344Ff73E88353521d099F";
var ZRX_USD = "0x2885d15b8Af22648b98B122b22FDF4D2a56c6023";
var MKR_USD = "0xec1D1B3b0443256cc3860e24a46F108e699484Aa";
var WLFI_USD = "0x14E5FC91Ddb3f97C33013Cc9fA74F54062Ad1Aa1";
var TBILL_NAV = "0xAbE7a3643615Ed32d3431e11E0Ee5A486Cb27d48";
var CRVUSD_USD = "0xEEf0C605546958c1f899b6fB336C20671f9cD49F";
var mETH_ETH = "0x5b563107C8666d2142C216114228443B94152362";
var ARB_USD = "0x31697852a68433DbCc2Ff612c516d69E3D9bd08F";
var LDO_ETH = "0x4e844125952D32AcdF339BE976c98E22F6F318dB";
var SGD_USD = "0xe25277fF4bbF9081C75Ab0EB13B4A13a721f3E13";
var GHO_USD = "0x3f12643D3f6f874d39C2a4c9f2Cd6f2DbAC877FC";
var TUSD_USD = "0xec746eCF986E2927Abd291a2A1716c940100f8Ba";
var CHAIN_ID = 1;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_ETH,
  AAVE_USD,
  ALCX_ETH,
  AMPL_ETH,
  AMPL_USD,
  APE_ETH,
  APE_USD,
  ARB_USD,
  ATH_USD,
  AUD_USD,
  AVAIL_USD,
  AVAX_USD,
  Arbitrum_Healthcheck,
  BADGER_ETH,
  BAL_ETH,
  BAL_USD,
  BAT_ETH,
  BAT_USD,
  BGBTC_Proof_of_Reserves,
  BNB_USD,
  BTC_ETH,
  BTC_USD,
  BUIDL_NAV,
  BUSD_USD,
  Base_Healthcheck,
  C3M_EUR,
  CAD_USD,
  CBETH_ETH,
  CETH_Reserves,
  CHAIN_ID,
  CHF_USD,
  CNY_USD,
  COMP_ETH,
  COMP_USD,
  CRVUSD_USD,
  CRV_ETH,
  CRV_USD,
  CSPX_USD,
  CVX_ETH,
  CVX_USD,
  Calculated_XSUSHI_ETH,
  Celo_Healthcheck,
  Consumer_Price_Index,
  DAI_ETH,
  DAI_USD,
  DOLO_USD,
  DPI_ETH,
  DPI_USD,
  EIGEN_USD,
  ENJ_ETH,
  ENS_USD,
  ETHFI_USD,
  ETH_BTC,
  ETH_USD,
  ETHx_ETH,
  EURC_USD,
  EUR_USD,
  EUTBL_NAV,
  FBTC_BTC_Exchange_Rate,
  FBTC_Proof_of_Reserves,
  FDUSD_USD,
  FIL_ETH,
  FRAX_ETH,
  FRAX_USD,
  FTM_ETH,
  FTT_ETH,
  FXS_USD,
  Fast_Gas_Gwei,
  GBP_USD,
  GHO_USD,
  GRT_ETH,
  GRT_USD,
  HBTC_Proof_of_Reserves,
  HIGH_USD,
  IB01_USD,
  IBTA_USD,
  IDR_USD,
  IMX_USD,
  Ink_Healthcheck,
  JAAA_NAV_Aave_LlamaRisk,
  JPY_USD,
  JTRSY_NAV_Aave_LlamaRisk,
  KNC_ETH,
  KNC_USD,
  KRW_USD,
  LBTC_BTC,
  LDO_ETH,
  LINK_ETH,
  LINK_USD,
  LRC_ETH,
  LUSD_USD,
  LsETH_ETH_Exchange_Rate,
  MANA_ETH,
  MATIC_USD,
  MAVIA_USD,
  MIM_USD,
  MKR_ETH,
  MKR_USD,
  MLN_ETH,
  MUSD_USD,
  M_NAV,
  Mantle_Healthcheck,
  Metis_Healthcheck,
  NEIRO_USD,
  NZD_USD,
  Nexus_wETH_Reserves,
  OETH_ETH,
  OHMv2_ETH,
  OUSDT_USD,
  Optimism_Healthcheck,
  PAXG_USD,
  PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate,
  PERP_ETH,
  PHP_USD,
  PYUSD_USD,
  PumpBTC_BTC_Exchange_Rate,
  PumpBTC_Proof_of_Reserves,
  RAI_ETH,
  RDNT_USD,
  REN_ETH,
  RETH_ETH,
  RLUSD_USD,
  RPL_USD,
  RSETH_ETH,
  RSR_USD,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate,
  SAND_USD,
  SGD_USD,
  SHIB_ETH,
  SHV_USD,
  SKY_USD,
  SNX_ETH,
  SNX_USD,
  SOLVBTC_BTC_Exchange_Rate,
  SOL_USD,
  SPELL_USD,
  STBT_Proof_of_Reserves,
  STETH_ETH,
  STETH_USD,
  STG_USD,
  SUSD1Plus_USD1_Exchange_Rate,
  SUSD_ETH,
  SUSHI_ETH,
  SUSHI_USD,
  SVR_AAVE_USD,
  SVR_AAVE_USD_BASE,
  SVR_BTC_USD,
  SVR_BTC_USD_BASE,
  SVR_COMP_USD,
  SVR_COMP_USD_BASE,
  SVR_CRCLon_USD,
  SVR_CRCLon_USD_BASE,
  SVR_ETH_USD,
  SVR_ETH_USD_BASE,
  SVR_LINK_USD,
  SVR_LINK_USD_BASE,
  SVR_QQQon_USD,
  SVR_QQQon_USD_BASE,
  SVR_SPYon_USD,
  SVR_SPYon_USD_BASE,
  SVR_TSLAon_USD,
  SVR_TSLAon_USD_BASE,
  SVR_USDC_USD,
  SVR_USDC_USD_BASE,
  SVR_USDT_USD,
  SVR_USDT_USD_BASE,
  SWELL_ETH,
  SWETH_ETH,
  SXT_USD,
  Scroll_Healthcheck,
  Secured_Overnight_Financing_Rate,
  SolvBTCBBN_SolvBTC_Exchange_Rate,
  SolvBTC_Proof_of_Reserves,
  Soneium_Healthcheck,
  Swell_ETH_Proof_of_Reserves,
  Swell_Restaked_ETH_Proof_of_Reserves,
  Synthetix_Aggregator_Debt_Ratio,
  Synthetix_Aggregator_Issued_Synths,
  TAO_USD,
  TBILL_NAV,
  TBTC_USD,
  TRUMP_USD,
  TRY_USD,
  TUSD_ETH,
  TUSD_Reserves,
  TUSD_USD,
  Total_Marketcap_USD,
  UNI_ETH,
  UNI_USD,
  USCC_NAV_Aave_LlamaRisk,
  USCC_NAV_per_Share,
  USD0PlusPlus_USD,
  USD0_USD,
  USD1_USD,
  USDC_ETH,
  USDC_USD,
  USDG_USD,
  USDL_USD,
  USDP_USD,
  USDS_USD,
  USDT_ETH,
  USDT_USD,
  USDe_USD,
  USDf_USD,
  USDtb_USD,
  USR_USD,
  USTBL_NAV,
  USTB_AUM,
  USTB_NAV_Aave_LlamaRisk,
  USTB_NAV_per_Share,
  UST_ETH,
  USYC_NAV_Aave_LlamaRisk,
  Unichain_Healthcheck,
  WAMPL_USD,
  WBTC_BTC,
  WBTC_Proof_of_Reserves,
  WLFI_USD,
  WTGXX_NAV,
  XAG_USD,
  XAU_USD,
  XCN_USD,
  YFI_ETH,
  YFI_USD,
  ZBU_USD,
  ZKsync_Healthcheck,
  ZRX_ETH,
  ZRX_USD,
  _1INCH_ETH,
  _1INCH_USD,
  _21BTC_Proof_of_Reserves,
  cUSD_AUM,
  cbBTC_Reserves,
  cbBTC_USD,
  deUSD_USD,
  eETH_Proof_of_Reserves,
  ezETH_ETH,
  frxUSD_USD,
  goUSD_Reserves,
  lvlUSD_USD,
  mETH_ETH,
  pufETH_ETH,
  rsETH_ETH_Exchange_Rate,
  rswETH_ETH,
  sUSDe_USD,
  sUSDe_USD_Calculated,
  sUSDf_USDf_Exchange_Rate,
  solvBTC_BTC,
  stBTC_BTC_Exchange_Rate,
  stBTC_Proof_of_Reserves,
  stETH_Proof_of_Reserves,
  tETH_wstETH_Exchange_Rate,
  uniBTC_BTC_Exchange_Rate,
  uniBTC_Proof_of_Reserves,
  weETH_ETH,
  wstETH_USD_Calculated,
  xSolvBTC_NAV,
  xSolvBTC_Proof_of_Reserves
});
