// @bgd-labs/aave-address-book/dist/ChainlinkScroll.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkScroll.ts
var ChainlinkScroll_exports = {};
__export(ChainlinkScroll_exports, {
  AAVE_USD: () => AAVE_USD,
  AVAX_USD: () => AVAX_USD,
  BNB_USD: () => BNB_USD,
  BTC_USD: () => BTC_USD,
  CHAIN_ID: () => CHAIN_ID,
  COMP_USD: () => COMP_USD,
  CRV_USD: () => CRV_USD,
  DAI_USD: () => DAI_USD,
  DOGE_USD: () => DOGE_USD,
  ETH_USD: () => ETH_USD,
  L2_Sequencer_Uptime_Status_Feed: () => L2_Sequencer_Uptime_Status_Feed,
  LINK_ETH: () => LINK_ETH,
  LINK_USD: () => LINK_USD,
  RETH_ETH: () => RETH_ETH,
  RSETH_ETH: () => RSETH_ETH,
  SCR_USD: () => SCR_USD,
  SOL_USD: () => SOL_USD,
  STETH_USD: () => STETH_USD,
  STG_USD: () => STG_USD,
  STONE_ETH_Exchange_Rate: () => STONE_ETH_Exchange_Rate,
  USDC_USD: () => USDC_USD,
  USDT_USD: () => USDT_USD,
  USDe_USD: () => USDe_USD,
  WBTC_BTC: () => WBTC_BTC,
  WBTC_USD: () => WBTC_USD,
  WSTETH_ETH: () => WSTETH_ETH,
  agETH_ETH_Exchange_Rate: () => agETH_ETH_Exchange_Rate,
  ezETH_ETH: () => ezETH_ETH,
  pufETH_ETH_Exchange_Rate: () => pufETH_ETH_Exchange_Rate,
  rsETH_ETH_Exchange_Rate: () => rsETH_ETH_Exchange_Rate,
  rswETH_ETH_Exchange_Rate: () => rswETH_ETH_Exchange_Rate,
  stBTC_BTC_Exchange_Rate: () => stBTC_BTC_Exchange_Rate,
  stBTC_Proof_of_Reserves: () => stBTC_Proof_of_Reserves,
  uniETH_ETH_Exchange_Rate: () => uniETH_ETH_Exchange_Rate,
  weETH_eETH_Exchange_Rate: () => weETH_eETH_Exchange_Rate,
  wstETH_stETH_Exchange_Rate: () => wstETH_stETH_Exchange_Rate,
  ylstETH_stETH_exchange_rate: () => ylstETH_stETH_exchange_rate,
  ynETH_ETH_Exchange_Rate: () => ynETH_ETH_Exchange_Rate,
  ynETHx_ETH_Exchange_Rate: () => ynETHx_ETH_Exchange_Rate
});
module.exports = __toCommonJS(ChainlinkScroll_exports);
var pufETH_ETH_Exchange_Rate = "0x7C6Da2C92caCe9F77274379Dc32a1eEE0B4C5FfD";
var ylstETH_stETH_exchange_rate = "0x8B5947e03894cEAD4f96283cFa19d88D6bc73d8F";
var weETH_eETH_Exchange_Rate = "0x57bd9E614f542fB3d6FeF2B744f3B813f0cc1258";
var STONE_ETH_Exchange_Rate = "0x0E4d8D665dA14D35444f0eCADc82F78a804A5F95";
var WSTETH_ETH = "0xe428fbdbd61CC1be6C273dC0E27a1F43124a86F3";
var COMP_USD = "0x6726C678feE07B25BBE67bC720728652E4129369";
var USDe_USD = "0x812Ef236A7240D8d0d59d61A9E7d67C43eFDB20e";
var STG_USD = "0x9019Be7Aa8f66551E94d6508EA48856386945E80";
var BNB_USD = "0x1AC823FdC79c30b1aB1787FF5e5766D6f29235E1";
var LINK_USD = "0x227a4E5E9239CAc88022DF86B1Ad9B24A7616e60";
var ETH_USD = "0x6bF14CB0A831078629D993FDeBcB182b21A8774C";
var SCR_USD = "0x26f6F7C468EE309115d19Aa2055db5A74F8cE7A5";
var stBTC_BTC_Exchange_Rate = "0x667f449F2B57C8FB3DaaB04D41A85216D83EbaCB";
var uniETH_ETH_Exchange_Rate = "0x16F46406457D35fc40BBAA306d8a81aa414e8482";
var SOL_USD = "0xDf3F55B6bd57084DD4a72a41853C0a2487CB757F";
var stBTC_Proof_of_Reserves = "0x6dCa10bce66548e26DBd1FCF4E8692796BeDdd7E";
var AVAX_USD = "0xB4b121ebE4DdCdFD3378b9519A101678829fE8c6";
var DAI_USD = "0x203322e1d15EB3Dff541a5aF0288D951c4a8d3eA";
var L2_Sequencer_Uptime_Status_Feed = "0x45c2b8C204568A03Dc7A2E32B71D67Fe97F908A9";
var ynETH_ETH_Exchange_Rate = "0xE9C5bedC9dfA8d758a8C2e81482c2e0623301d4D";
var rsETH_ETH_Exchange_Rate = "0x3a44916dc37Bb7D73419Fc6492d6d9Dfd8e6ddf7";
var RSETH_ETH = "0x37f85bbB5366282F9C25e0bdCdbbF8243E594E52";
var CRV_USD = "0x8658273E2f7bc06d3F8462703b8a733204312fF2";
var WBTC_USD = "0x61C432B58A43516899d8dF33A5F57edb8d57EB77";
var LINK_ETH = "0x78409c5b2dE2aC8ac76f45458FBaDD707e87B98a";
var DOGE_USD = "0x2667de5E58Ae152ce9c5EA6D1a8E051444294B82";
var STETH_USD = "0x439a2b573C8Ecd215990Fc25b4F547E89CF67b79";
var agETH_ETH_Exchange_Rate = "0xf42e954D384247cE971d65f09a34500551942d67";
var BTC_USD = "0xCaca6BFdeDA537236Ee406437D2F8a400026C589";
var wstETH_stETH_Exchange_Rate = "0xE61Da4C909F7d86797a0D06Db63c34f76c9bCBDC";
var RETH_ETH = "0x3fBB86e564fC1303625BA88EaE55740f3A649d36";
var AAVE_USD = "0x538E0fC727ce4604e25354D082890cdb5553d33B";
var USDT_USD = "0xf376A91Ae078927eb3686D6010a6f1482424954E";
var rswETH_ETH_Exchange_Rate = "0x97acBC804c5Cd994829Bdf89A2569575B159ebD4";
var USDC_USD = "0x43d12Fb3AfCAd5347fA764EeAB105478337b7200";
var WBTC_BTC = "0x7e73693088d88694146ab30f1dA5903a4489e992";
var ezETH_ETH = "0xd63624330Ed4e50dD65FeA45c08558F78a3d8abd";
var ynETHx_ETH_Exchange_Rate = "0x14473e2BD405dcb5550ccA1d36da40227A6810EC";
var CHAIN_ID = 534352;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_USD,
  AVAX_USD,
  BNB_USD,
  BTC_USD,
  CHAIN_ID,
  COMP_USD,
  CRV_USD,
  DAI_USD,
  DOGE_USD,
  ETH_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LINK_ETH,
  LINK_USD,
  RETH_ETH,
  RSETH_ETH,
  SCR_USD,
  SOL_USD,
  STETH_USD,
  STG_USD,
  STONE_ETH_Exchange_Rate,
  USDC_USD,
  USDT_USD,
  USDe_USD,
  WBTC_BTC,
  WBTC_USD,
  WSTETH_ETH,
  agETH_ETH_Exchange_Rate,
  ezETH_ETH,
  pufETH_ETH_Exchange_Rate,
  rsETH_ETH_Exchange_Rate,
  rswETH_ETH_Exchange_Rate,
  stBTC_BTC_Exchange_Rate,
  stBTC_Proof_of_Reserves,
  uniETH_ETH_Exchange_Rate,
  weETH_eETH_Exchange_Rate,
  wstETH_stETH_Exchange_Rate,
  ylstETH_stETH_exchange_rate,
  ynETH_ETH_Exchange_Rate,
  ynETHx_ETH_Exchange_Rate
});
