// @bgd-labs/aave-address-book/dist/ChainlinkBase.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/ChainlinkBase.ts
var ChainlinkBase_exports = {};
__export(ChainlinkBase_exports, {
  AAVE_USD: () => AAVE_USD,
  ADA_USD: () => ADA_USD,
  AERO_USD: () => AERO_USD,
  AMP_USD: () => AMP_USD,
  ANON_USD: () => ANON_USD,
  APT_USD: () => APT_USD,
  ARKB_Reserves: () => ARKB_Reserves,
  AUD_USD: () => AUD_USD,
  AVAIL_USD: () => AVAIL_USD,
  AVAX_USD: () => AVAX_USD,
  AXL_USD: () => AXL_USD,
  BNB_USD: () => BNB_USD,
  BRL_USD: () => BRL_USD,
  BTC_USD: () => BTC_USD,
  CAD_USD: () => CAD_USD,
  CBDOGE_USD: () => CBDOGE_USD,
  CBETH_ETH: () => CBETH_ETH,
  CBETH_USD: () => CBETH_USD,
  CBXRP_USD: () => CBXRP_USD,
  CETH_Reserves: () => CETH_Reserves,
  CHAIN_ID: () => CHAIN_ID,
  CHF_USD: () => CHF_USD,
  COMP_USD: () => COMP_USD,
  CTX_USD: () => CTX_USD,
  CUSDO_USD: () => CUSDO_USD,
  DAI_USD: () => DAI_USD,
  DEGEN_USD: () => DEGEN_USD,
  DOGE_USD: () => DOGE_USD,
  ETH_USD: () => ETH_USD,
  EURC_USD: () => EURC_USD,
  EUR_USD: () => EUR_USD,
  GBP_USD: () => GBP_USD,
  GHO_USD: () => GHO_USD,
  HOME_USD: () => HOME_USD,
  IDR_USD: () => IDR_USD,
  L2_Sequencer_Uptime_Status_Feed: () => L2_Sequencer_Uptime_Status_Feed,
  LBTC_BTC: () => LBTC_BTC,
  LBTC_USD: () => LBTC_USD,
  LINK_ETH: () => LINK_ETH,
  LINK_USD: () => LINK_USD,
  LSETH_ETH: () => LSETH_ETH,
  LTC_USD: () => LTC_USD,
  LsETH_ETH_Exchange_Rate: () => LsETH_ETH_Exchange_Rate,
  MAG7SSI_USD: () => MAG7SSI_USD,
  MATIC_USD: () => MATIC_USD,
  MAVIA_USD: () => MAVIA_USD,
  MELANIA_USD: () => MELANIA_USD,
  MEW_USD: () => MEW_USD,
  MLN_USD: () => MLN_USD,
  MOG_USD: () => MOG_USD,
  MORPHO_USD: () => MORPHO_USD,
  MXN_USD: () => MXN_USD,
  NGN_USD: () => NGN_USD,
  NZD_USD: () => NZD_USD,
  OGN_USD: () => OGN_USD,
  OP_USD: () => OP_USD,
  OUSDT_USD: () => OUSDT_USD,
  PCE_Price_Index_Level: () => PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate: () => PCE_Price_Index_Percent_Change_Annual_Rate,
  PEPE_USD: () => PEPE_USD,
  PHP_USD: () => PHP_USD,
  POL_USD: () => POL_USD,
  RDNT_USD: () => RDNT_USD,
  RETH_ETH: () => RETH_ETH,
  RSETH_ETH: () => RSETH_ETH,
  RSR_USD: () => RSR_USD,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate: () => Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level: () => Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate: () => Real_GDP_Percent_Change_Annual_Rate,
  SGD_USD: () => SGD_USD,
  SHIB_USD: () => SHIB_USD,
  SNX_USD: () => SNX_USD,
  SOL_USD: () => SOL_USD,
  STETH_ETH: () => STETH_ETH,
  STG_USD: () => STG_USD,
  SUI_USD: () => SUI_USD,
  SUPEROETHB_ETH: () => SUPEROETHB_ETH,
  SolvBTCBBN_SolvBTC_Exchange_Rate: () => SolvBTCBBN_SolvBTC_Exchange_Rate,
  TBTC_USD: () => TBTC_USD,
  TRUMP_USD: () => TRUMP_USD,
  TRY_USD: () => TRY_USD,
  Total_Market_Cap_USD: () => Total_Market_Cap_USD,
  USDC_USD: () => USDC_USD,
  USDO_Reserves: () => USDO_Reserves,
  USDPlus_USD: () => USDPlus_USD,
  USDS_USD: () => USDS_USD,
  USDT_USD: () => USDT_USD,
  USDX_USD: () => USDX_USD,
  USDe_USD: () => USDe_USD,
  USDz_USD: () => USDz_USD,
  USR_USD: () => USR_USD,
  VIRTUAL_USD: () => VIRTUAL_USD,
  VVV_USD: () => VVV_USD,
  WBTC_USD: () => WBTC_USD,
  WELL_USD: () => WELL_USD,
  WIF_USD: () => WIF_USD,
  WMTx_USD: () => WMTx_USD,
  WSTETH_ETH: () => WSTETH_ETH,
  XAU_USD: () => XAU_USD,
  XDC_USD: () => XDC_USD,
  XRP_USD: () => XRP_USD,
  YFI_USD: () => YFI_USD,
  ZAR_USD: () => ZAR_USD,
  ZBU_USD: () => ZBU_USD,
  ZRO_USD: () => ZRO_USD,
  cbBTC_Reserves: () => cbBTC_Reserves,
  cbBTC_USD: () => cbBTC_USD,
  cbETH_ETH_Exchange_Rate: () => cbETH_ETH_Exchange_Rate,
  dlcBTC_Proof_of_Reserves: () => dlcBTC_Proof_of_Reserves,
  ezETH_ETH: () => ezETH_ETH,
  ezETH_ETH_Exchange_Rate: () => ezETH_ETH_Exchange_Rate,
  iBTC_Proof_of_Reserves: () => iBTC_Proof_of_Reserves,
  inETH_ETH_Exchange_Rate: () => inETH_ETH_Exchange_Rate,
  instETH_ETH_Exchange_Rate: () => instETH_ETH_Exchange_Rate,
  inwstETH_wstETH_Exchange_Rate: () => inwstETH_wstETH_Exchange_Rate,
  mooBIFI_USD: () => mooBIFI_USD,
  pufETH_ETH: () => pufETH_ETH,
  pufETH_ETH_Exchange_Rate: () => pufETH_ETH_Exchange_Rate,
  rETH_ETH_Exchange_Rate: () => rETH_ETH_Exchange_Rate,
  rsETH_ETH_Exchange_Rate: () => rsETH_ETH_Exchange_Rate,
  rswETH_ETH_Exchange_Rate: () => rswETH_ETH_Exchange_Rate,
  sUSDS_USDS_Exchange_Rate: () => sUSDS_USDS_Exchange_Rate,
  sUSDe_USD: () => sUSDe_USD,
  sUSDe_USDe_Exchange_Rate: () => sUSDe_USDe_Exchange_Rate,
  sUSDz_USDz_Exchange_Rate: () => sUSDz_USDz_Exchange_Rate,
  sfrxETH_frxETH_Exchange_Rate: () => sfrxETH_frxETH_Exchange_Rate,
  solvBTC_BTC: () => solvBTC_BTC,
  swBTC_WBTC_Exchange_Rate: () => swBTC_WBTC_Exchange_Rate,
  syrupUSDC_USDC_Exchange_Rate: () => syrupUSDC_USDC_Exchange_Rate,
  tETH_wstETH_Exchange_Rate: () => tETH_wstETH_Exchange_Rate,
  ultraETHs_ETH_Exchange_Rate: () => ultraETHs_ETH_Exchange_Rate,
  uniBTC_BTC_Exchange_Rate: () => uniBTC_BTC_Exchange_Rate,
  vyUSD_USD_Exchange_Rate: () => vyUSD_USD_Exchange_Rate,
  wOETH_OETH_Exchange_Rate: () => wOETH_OETH_Exchange_Rate,
  weETH_ETH: () => weETH_ETH,
  weETH_eETH_Exchange_Rate: () => weETH_eETH_Exchange_Rate,
  wrsETH_ETH_Exchange_Rate: () => wrsETH_ETH_Exchange_Rate,
  wstETH_ETH_Exchange_Rate: () => wstETH_ETH_Exchange_Rate,
  wstETH_stETH_Exchange_Rate: () => wstETH_stETH_Exchange_Rate,
  wstUSR_stUSR_Exchange_Rate: () => wstUSR_stUSR_Exchange_Rate,
  xSolvBTC_NAV: () => xSolvBTC_NAV,
  yUSD_USD_Exchange_Rate: () => yUSD_USD_Exchange_Rate,
  ynETH_ETH_Exchange_Rate: () => ynETH_ETH_Exchange_Rate,
  ynETHx_ETH_Exchange_Rate: () => ynETHx_ETH_Exchange_Rate
});
module.exports = __toCommonJS(ChainlinkBase_exports);
var sUSDS_USDS_Exchange_Rate = "0x906B24a339b848369B24Dc9Ed368b947fB9693bf";
var ezETH_ETH = "0x960BDD1dFD20d7c98fa482D793C3dedD73A113a3";
var AXL_USD = "0x676C4C6C31D97A5581D3204C04A8125B350E2F9D";
var wstETH_stETH_Exchange_Rate = "0xB88BAc61a4Ca37C43a3725912B1f472c9A5bc061";
var sUSDz_USDz_Exchange_Rate = "0xD89c7fFB39C44b17EAecd8717a75A36c19C07582";
var wOETH_OETH_Exchange_Rate = "0xe96EB1EDa83d18cbac224233319FA5071464e1b9";
var CBETH_ETH = "0x806b4Ac04501c29769051e42783cF04dCE41440b";
var BRL_USD = "0x0b0E64c05083FdF9ED7C5D3d8262c4216eFc9394";
var TBTC_USD = "0x6D75BFB5A5885f841b132198C9f0bE8c872057BF";
var LBTC_USD = "0x9e07546c9Fe8868855CD04B26051a26D1599E270";
var USDC_USD = "0x7e860098F58bBFC8648a4311b374B1D669a2bc6B";
var RDNT_USD = "0xEf2E24ba6def99B5e0b71F6CDeaF294b02163094";
var USR_USD = "0x4a595E0a62E50A2E5eC95A70c8E612F9746af006";
var syrupUSDC_USDC_Exchange_Rate = "0x311D3A3faA1d5939c681E33C2CDAc041FF388EB2";
var ZAR_USD = "0x2ecc8A8B370fC6a217166b2782a35339bEBEe98B";
var CBXRP_USD = "0xEEe1a9D5A0C36d99972C057Cb959267e88Ab9160";
var SGD_USD = "0x81575495532fB311Efc5C993B612564274F0949b";
var tETH_wstETH_Exchange_Rate = "0x8004571d9f54dE016fc3D448e7AEe2d70947727A";
var inETH_ETH_Exchange_Rate = "0x83ac12dBb5Bd7Fa597ab2FFEc9F2F13DeDdFe163";
var cbETH_ETH_Exchange_Rate = "0x868a501e68F3D1E89CfC0D22F6b22E8dabce5F04";
var weETH_eETH_Exchange_Rate = "0x35e9D7001819Ea3B39Da906aE6b06A62cfe2c181";
var ynETH_ETH_Exchange_Rate = "0xb4482096e3cdE116C15fC0D700a73a58FEdeB8c0";
var DAI_USD = "0x591e79239a7d679378eC8c847e5038150364C78F";
var wrsETH_ETH_Exchange_Rate = "0xe8dD07CCf5BC4922424140E44Eb970F5950725ef";
var ezETH_ETH_Exchange_Rate = "0xC4300B7CF0646F0Fe4C5B2ACFCCC4dCA1346f5d8";
var XDC_USD = "0x237A94A589DD38DF7e50CeFDa0b8916a54d01ecC";
var OP_USD = "0x3E3A6bD129A63564FE7abde85FA67c3950569060";
var NGN_USD = "0xdfbb5Cbc88E382de007bfe6CE99C388176ED80aD";
var LINK_ETH = "0xc5E65227fe3385B88468F9A01600017cDC9F3A12";
var USDe_USD = "0x790181e93e9F4Eedb5b864860C12e4d2CffFe73B";
var PHP_USD = "0x0396000dc82bfAEe746A9Ac6dC69dAd3223Ca9c6";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Level = "0x65623109aA4561AD3cfF503542083548CeD7e085";
var USDS_USD = "0x2330aaE3bca5F05169d5f4597964D44522F62930";
var CBETH_USD = "0xd7818272B9e248357d13057AAb0B417aF31E817d";
var WMTx_USD = "0x311681f6E0b34670Fb03e066cc08C6D09149a44c";
var inwstETH_wstETH_Exchange_Rate = "0xb58c5C550Ba19c4CEeE071F8CeeB58f8770e6978";
var SUI_USD = "0x491a921c41d6a97C57426E0c0108a231cd6E5f60";
var XAU_USD = "0x5213eBB69743b85644dbB6E25cdF994aFBb8cF31";
var GBP_USD = "0xCceA6576904C118037695eB71195a5425E69Fa15";
var yUSD_USD_Exchange_Rate = "0xc1a849217F3BaB97F1a46b990e369D6705B4be96";
var MAG7SSI_USD = "0x2B1de6AD89847C11aF2ede14edB013AA79E94aC9";
var ARKB_Reserves = "0xB366E8Efb9661323ff477CedF70f55F897D6cFeA";
var cbBTC_Reserves = "0x0F8E057D1D7b282EF968D26E9cB432617dF52519";
var RETH_ETH = "0xf397bF97280B488cA19ee3093E81C0a77F02e9a5";
var wstETH_ETH_Exchange_Rate = "0xa669E5272E60f78299F4824495cE01a3923f4380";
var weETH_ETH = "0xFC1415403EbB0c693f9a7844b92aD2Ff24775C65";
var RSETH_ETH = "0xd7221b10FBBC1e1ba95Fd0B4D031C15f7F365296";
var EUR_USD = "0xc91D87E81faB8f93699ECf7Ee9B44D11e1D53F0F";
var wstUSR_stUSR_Exchange_Rate = "0x0594c1a01375c1151c2ca78BE4870836EbFA9846";
var cbBTC_USD = "0x07DA0E54543a844a80ABE69c8A12F22B3aA59f9D";
var PCE_Price_Index_Level = "0x18A3fcA54FaC5B05837205bA4b823fc56191F793";
var dlcBTC_Proof_of_Reserves = "0x30A76F4E688Cf52f4A06D7AAd987A7037f3Ae6f7";
var XRP_USD = "0x9f0C1dD78C4CBdF5b9cf923a549A201EdC676D34";
var OGN_USD = "0x91D7AEd72bF772A0DA30199B925aCB866ACD3D9e";
var rETH_ETH_Exchange_Rate = "0x1E6A29666288a310326B37d823Fe4Ea3937424D2";
var ETH_USD = "0x71041dddad3595F9CEd3DcCFBe3D1F4b0a16Bb70";
var IDR_USD = "0x05A6cF213EcC5501A11a08EBefA4A8a60313ef97";
var TRY_USD = "0x29413773e7CD4Dfd6Ad89a50887877b88a6C592C";
var sfrxETH_frxETH_Exchange_Rate = "0x1Eba1d6941088c8FCE2CbcaC80754C77871aD093";
var L2_Sequencer_Uptime_Status_Feed = "0xBCF85224fc0756B9Fa45aA7892530B47e10b6433";
var AVAIL_USD = "0x947eD6A3664fCE27a365b720286c01074d3782E0";
var COMP_USD = "0x9DDa783DE64A9d1A60c49ca761EbE528C35BA428";
var LTC_USD = "0x206a34e47093125fbf4C75b7c7E88b84c6A77a69";
var vyUSD_USD_Exchange_Rate = "0x99C098FA069B120dd81E56c0f2178093cc7a851f";
var sUSDe_USD = "0x79cf4a31B29D69191f0b6E97916eB93FEB81E533";
var AUD_USD = "0x46e51B8cA41d709928EdA9Ae43e42193E6CDf229";
var WIF_USD = "0x674940e1dBf7FD841b33156DA9A88afbD95AaFBa";
var STG_USD = "0x63Af8341b62E683B87bB540896bF283D96B4D385";
var Total_Market_Cap_USD = "0x962C0Df8Ca7f7C682B3872ccA31Ea9c8999ab23c";
var AAVE_USD = "0x3d6774EF702A10b20FCa8Ed40FC022f7E4938e07";
var LINK_USD = "0x17CAb8FE31E32f08326e5E27412894e49B0f9D65";
var xSolvBTC_NAV = "0x17738F7dacFc1De7d06f22cC52211EBf68744dBA";
var SNX_USD = "0xe3971Ed6F1A5903321479Ef3148B5950c0612075";
var VVV_USD = "0x8eC6a128a430f7A850165bcF18facc9520a9873F";
var POL_USD = "0x5E988c11a4f92155C30D9fb69Ed75597f712B113";
var AMP_USD = "0x1688e4B274a4CC9fD398EbA6Ae4dfb6528A9D2bc";
var STETH_ETH = "0xf586d0728a47229e747d824a939000Cf21dEF5A0";
var ultraETHs_ETH_Exchange_Rate = "0xbb9786e37D54251477EbC1325b04ACdCA18C2254";
var sUSDe_USDe_Exchange_Rate = "0xdEd37FC1400B8022968441356f771639ad1B23aA";
var PEPE_USD = "0xB48ac6409C0c3718b956089b0fFE295A10ACDdad";
var CAD_USD = "0xA840145F87572E82519d578b1F36340368a25D5d";
var MEW_USD = "0x9FB8b5A4b3FE655564f0c76616ae79DE90Cc7382";
var AVAX_USD = "0xE70f2D34Fd04046aaEC26a198A35dD8F2dF5cd92";
var MATIC_USD = "0x12129aAC52D6B0f0125677D4E1435633E61fD25f";
var BTC_USD = "0x64c911996D3c6aC71f9b455B1E8E7266BcbD848F";
var HOME_USD = "0x121934C415937863d64ef93436169444633EE0d8";
var instETH_ETH_Exchange_Rate = "0x9C6BF4884Ff0c7873652F7d5142FA3b9859a526D";
var CTX_USD = "0x90F3676B40F6dc2C1E074985D0544Bb8e1815B00";
var LBTC_BTC = "0x1E6c22AAA11F507af12034A5Dc4126A6A25DC8d2";
var pufETH_ETH = "0x9452Ca03474C6B704B4e102339B451D640f57f07";
var USDX_USD = "0xC342785EE44b9F8a40a564200CDF14dFCefd36C3";
var GHO_USD = "0x42868EFcee13C0E71af89c04fF7d96f5bec479b0";
var ADA_USD = "0x34cD971a092d5411bD69C10a5F0A7EEF72C69041";
var USDPlus_USD = "0xd5Ec94430eF4170D819E0996BC53ed40d31638d8";
var SHIB_USD = "0xC8D5D660bb585b68fa0263EeD7B4224a5FC99669";
var OUSDT_USD = "0x0E230b1077c663f8Fb5e68d84A8e3e33D97d7436";
var MAVIA_USD = "0x979447581b39caCA33EF0CA8208592393D16cc13";
var rswETH_ETH_Exchange_Rate = "0x97b770B0200CCe161907a9cbe0C6B177679f8F7C";
var VIRTUAL_USD = "0xEaf310161c9eF7c813A14f8FEF6Fb271434019F7";
var ZRO_USD = "0xdc31a4CCfCA039BeC6222e20BE7770E12581bfEB";
var ZBU_USD = "0x19c6501ee6FF5Faf36346031A92C46AF128807d3";
var WSTETH_ETH = "0x43a5C292A453A3bF3606fa856197f09D7B74251a";
var SUPEROETHB_ETH = "0x39C6E14CdE46D4FFD9F04Ff159e7ce8eC20E10B4";
var uniBTC_BTC_Exchange_Rate = "0xbC7c5023eE571e4D9C4890C90a16be05c1EEf410";
var APT_USD = "0x88a98431C25329AA422B21D147c1518b34dD36F4";
var SolvBTCBBN_SolvBTC_Exchange_Rate = "0x67283A47E470afbCcc4aC74ccC32401a81027691";
var MLN_USD = "0x122b5334A8b55861dBc6729c294451471FbF318D";
var AERO_USD = "0x4EC5970fC728C5f65ba413992CD5fF6FD70fcfF0";
var CETH_Reserves = "0x0b68ac37a1668DAaab1882543368E076C38C40e9";
var CHF_USD = "0x3A1d6444fb6a402470098E23DaD0B7E86E14252F";
var Real_GDP_Percent_Change_Annual_Rate = "0xe0eda54fC1362C0d7d0ff855E4fCEA79916Fe094";
var ynETHx_ETH_Exchange_Rate = "0x4e7dB2f9a28348AB48a968dd4217D565D1F15Ba4";
var Real_GDP_Level = "0x0df397aFE00085C138a99eFB39C498e08eB95aD1";
var PCE_Price_Index_Percent_Change_Annual_Rate = "0x2a18E2d46Cb067b69e0759dB39b16597fC42D962";
var LSETH_ETH = "0xeDC243c7E3c1A9dAf067C90641D2346d2694d2e5";
var USDz_USD = "0xe25969e2Fa633a0C027fAB8F30Fc9C6A90D60B48";
var USDO_Reserves = "0x5218Ebeb96bD2bAFe21F9b143f5672552629ba79";
var MORPHO_USD = "0xe95e258bb6615d47515Fc849f8542dA651f12bF6";
var USDT_USD = "0xf19d560eB8d2ADf07BD6D13ed03e1D11215721F9";
var WBTC_USD = "0xCCADC697c55bbB68dc5bCdf8d3CBe83CdD4E071E";
var SOL_USD = "0x975043adBb80fc32276CbF9Bbcfd4A601a12462D";
var pufETH_ETH_Exchange_Rate = "0x69a1d14a4e58e97EDE8337DE61eEB2e4a55886E0";
var TRUMP_USD = "0x7bAfa1Af54f17cC0775a1Cf813B9fF5dED2C51E5";
var mooBIFI_USD = "0x721F1B4dc604AEA0661Aa9982AB624e5756B31f2";
var DOGE_USD = "0x8422f3d3CAFf15Ca682939310d6A5e619AE08e57";
var ANON_USD = "0x674e028B95330E77F5cF89834254a6Bfa806f3a2";
var MELANIA_USD = "0xFAf372CaBc765b63f6fabD436c845D965eDA1CA5";
var iBTC_Proof_of_Reserves = "0x7FCED5198e43ec93Ef2179DFC70a8dcf494DcB80";
var solvBTC_BTC = "0xB4a1a7f260C9FF7fEd6A6fbb9fe5a9acFa725DBf";
var DEGEN_USD = "0xE62BcE5D7CB9d16AB8b4D622538bc0A50A5799c2";
var MXN_USD = "0x9e8Ee77c76d4fa41306056D1C3196AF5da1600bd";
var BNB_USD = "0x4b7836916781CAAfbb7Bd1E5FDd20ED544B453b1";
var rsETH_ETH_Exchange_Rate = "0x99DAf760d2CFB770cc17e883dF45454FE421616b";
var CBDOGE_USD = "0x95051De9Db5Cac61682f64505006f2991dEaB3c2";
var EURC_USD = "0xDAe398520e2B67cd3f27aeF9Cf14D93D927f8250";
var WELL_USD = "0xc15d9944dAefE2dB03e53bef8DDA25a56832C5fe";
var CUSDO_USD = "0xBbdA526258856c06f167399097e13D79E8f88ba8";
var YFI_USD = "0xD40e758b5eC80820B68DFC302fc5Ce1239083548";
var Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate = "0xe2b3688371130f333443428Cf03f27Ce0378F9dC";
var NZD_USD = "0x06bdFe07E71C476157FC025d3cCD4BBe08e83EF9";
var MOG_USD = "0x4aeb6D15769EaD32D0c5Be2940F40c7CFf53801d";
var swBTC_WBTC_Exchange_Rate = "0xBD867487712ADeC5A59b9Ae475Ee942f652B4C91";
var RSR_USD = "0xAa98aE504658766Dfe11F31c5D95a0bdcABDe0b1";
var LsETH_ETH_Exchange_Rate = "0x2621897C993fdE08873Ef58dA1453aEE49a70144";
var CHAIN_ID = 8453;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_USD,
  ADA_USD,
  AERO_USD,
  AMP_USD,
  ANON_USD,
  APT_USD,
  ARKB_Reserves,
  AUD_USD,
  AVAIL_USD,
  AVAX_USD,
  AXL_USD,
  BNB_USD,
  BRL_USD,
  BTC_USD,
  CAD_USD,
  CBDOGE_USD,
  CBETH_ETH,
  CBETH_USD,
  CBXRP_USD,
  CETH_Reserves,
  CHAIN_ID,
  CHF_USD,
  COMP_USD,
  CTX_USD,
  CUSDO_USD,
  DAI_USD,
  DEGEN_USD,
  DOGE_USD,
  ETH_USD,
  EURC_USD,
  EUR_USD,
  GBP_USD,
  GHO_USD,
  HOME_USD,
  IDR_USD,
  L2_Sequencer_Uptime_Status_Feed,
  LBTC_BTC,
  LBTC_USD,
  LINK_ETH,
  LINK_USD,
  LSETH_ETH,
  LTC_USD,
  LsETH_ETH_Exchange_Rate,
  MAG7SSI_USD,
  MATIC_USD,
  MAVIA_USD,
  MELANIA_USD,
  MEW_USD,
  MLN_USD,
  MOG_USD,
  MORPHO_USD,
  MXN_USD,
  NGN_USD,
  NZD_USD,
  OGN_USD,
  OP_USD,
  OUSDT_USD,
  PCE_Price_Index_Level,
  PCE_Price_Index_Percent_Change_Annual_Rate,
  PEPE_USD,
  PHP_USD,
  POL_USD,
  RDNT_USD,
  RETH_ETH,
  RSETH_ETH,
  RSR_USD,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Level,
  Real_Final_Sales_to_Private_Domestic_Purchasers_Percent_Change_Annual_Rate,
  Real_GDP_Level,
  Real_GDP_Percent_Change_Annual_Rate,
  SGD_USD,
  SHIB_USD,
  SNX_USD,
  SOL_USD,
  STETH_ETH,
  STG_USD,
  SUI_USD,
  SUPEROETHB_ETH,
  SolvBTCBBN_SolvBTC_Exchange_Rate,
  TBTC_USD,
  TRUMP_USD,
  TRY_USD,
  Total_Market_Cap_USD,
  USDC_USD,
  USDO_Reserves,
  USDPlus_USD,
  USDS_USD,
  USDT_USD,
  USDX_USD,
  USDe_USD,
  USDz_USD,
  USR_USD,
  VIRTUAL_USD,
  VVV_USD,
  WBTC_USD,
  WELL_USD,
  WIF_USD,
  WMTx_USD,
  WSTETH_ETH,
  XAU_USD,
  XDC_USD,
  XRP_USD,
  YFI_USD,
  ZAR_USD,
  ZBU_USD,
  ZRO_USD,
  cbBTC_Reserves,
  cbBTC_USD,
  cbETH_ETH_Exchange_Rate,
  dlcBTC_Proof_of_Reserves,
  ezETH_ETH,
  ezETH_ETH_Exchange_Rate,
  iBTC_Proof_of_Reserves,
  inETH_ETH_Exchange_Rate,
  instETH_ETH_Exchange_Rate,
  inwstETH_wstETH_Exchange_Rate,
  mooBIFI_USD,
  pufETH_ETH,
  pufETH_ETH_Exchange_Rate,
  rETH_ETH_Exchange_Rate,
  rsETH_ETH_Exchange_Rate,
  rswETH_ETH_Exchange_Rate,
  sUSDS_USDS_Exchange_Rate,
  sUSDe_USD,
  sUSDe_USDe_Exchange_Rate,
  sUSDz_USDz_Exchange_Rate,
  sfrxETH_frxETH_Exchange_Rate,
  solvBTC_BTC,
  swBTC_WBTC_Exchange_Rate,
  syrupUSDC_USDC_Exchange_Rate,
  tETH_wstETH_Exchange_Rate,
  ultraETHs_ETH_Exchange_Rate,
  uniBTC_BTC_Exchange_Rate,
  vyUSD_USD_Exchange_Rate,
  wOETH_OETH_Exchange_Rate,
  weETH_ETH,
  weETH_eETH_Exchange_Rate,
  wrsETH_ETH_Exchange_Rate,
  wstETH_ETH_Exchange_Rate,
  wstETH_stETH_Exchange_Rate,
  wstUSR_stUSR_Exchange_Rate,
  xSolvBTC_NAV,
  yUSD_USD_Exchange_Rate,
  ynETH_ETH_Exchange_Rate,
  ynETHx_ETH_Exchange_Rate
});
