// @bgd-labs/aave-address-book/dist/GhoArbitrum.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GhoArbitrum.ts
var GhoArbitrum_exports = {};
__export(GhoArbitrum_exports, {
  CHAIN_ID: () => CHAIN_ID,
  GHO_AAVE_CORE_STEWARD: () => GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD: () => GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD: () => GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL: () => GHO_CCIP_TOKEN_POOL,
  GHO_TOKEN: () => GHO_TOKEN,
  RISK_COUNCIL: () => RISK_COUNCIL
});
module.exports = __toCommonJS(GhoArbitrum_exports);
var GHO_TOKEN = "0x7dfF72693f6A4149b17e7C6314655f6A9F7c8B33";
var GHO_CCIP_TOKEN_POOL = "0xB94Ab28c6869466a46a42abA834ca2B3cECCA5eB";
var GHO_AAVE_CORE_STEWARD = "0xd2D586f849620ef042FE3aF52eAa10e9b78bf7De";
var GHO_BUCKET_STEWARD = "0xa9afaE6A53E90f9E4CE0717162DF5Bc3d9aBe7B2";
var GHO_CCIP_STEWARD = "0xCd5ab470AaC5c13e1063ee700503f3346b7C90Db";
var RISK_COUNCIL = "0x8513e6F37dBc52De87b166980Fa3F50639694B60";
var CHAIN_ID = 42161;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL,
  GHO_TOKEN,
  RISK_COUNCIL
});
