// @bgd-labs/aave-address-book/dist/GhoBase.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GhoBase.ts
var GhoBase_exports = {};
__export(GhoBase_exports, {
  CHAIN_ID: () => CHAIN_ID,
  GHO_AAVE_CORE_STEWARD: () => GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD: () => GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD: () => GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL: () => GHO_CCIP_TOKEN_POOL,
  GHO_TOKEN: () => GHO_TOKEN,
  RISK_COUNCIL: () => RISK_COUNCIL
});
module.exports = __toCommonJS(GhoBase_exports);
var GHO_TOKEN = "0x6Bb7a212910682DCFdbd5BCBb3e28FB4E8da10Ee";
var GHO_CCIP_TOKEN_POOL = "0x98217A06721Ebf727f2C8d9aD7718ec28b7aAe34";
var GHO_AAVE_CORE_STEWARD = "0xC5BcC58BE6172769ca1a78B8A45752E3C5059c39";
var GHO_BUCKET_STEWARD = "0x3c47237479e7569653eF9beC4a7Cd2ee3F78b396";
var GHO_CCIP_STEWARD = "0xB94Ab28c6869466a46a42abA834ca2B3cECCA5eB";
var RISK_COUNCIL = "0x8513e6F37dBc52De87b166980Fa3F50639694B60";
var CHAIN_ID = 8453;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL,
  GHO_TOKEN,
  RISK_COUNCIL
});
