// @bgd-labs/aave-address-book/dist/GhoEthereum.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GhoEthereum.ts
var GhoEthereum_exports = {};
__export(GhoEthereum_exports, {
  CHAIN_ID: () => CHAIN_ID,
  GHO_AAVE_CORE_STEWARD: () => GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD: () => GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD: () => GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL: () => GHO_CCIP_TOKEN_POOL,
  GHO_FLASHMINTER_FACILITATOR: () => GHO_FLASHMINTER_FACILITATOR,
  GHO_GSM_STEWARD: () => GHO_GSM_STEWARD,
  GHO_LIQUIDITY_COMMITTEE: () => GHO_LIQUIDITY_COMMITTEE,
  GHO_TOKEN: () => GHO_TOKEN,
  GSM_REGISTRY: () => GSM_REGISTRY,
  GSM_USDC: () => GSM_USDC,
  GSM_USDC_FEE_STRATEGY: () => GSM_USDC_FEE_STRATEGY,
  GSM_USDC_ORACLE_SWAP_FREEZER: () => GSM_USDC_ORACLE_SWAP_FREEZER,
  GSM_USDC_PRICE_STRATEGY: () => GSM_USDC_PRICE_STRATEGY,
  GSM_USDT: () => GSM_USDT,
  GSM_USDT_FEE_STRATEGY: () => GSM_USDT_FEE_STRATEGY,
  GSM_USDT_ORACLE_SWAP_FREEZER: () => GSM_USDT_ORACLE_SWAP_FREEZER,
  GSM_USDT_PRICE_STRATEGY: () => GSM_USDT_PRICE_STRATEGY,
  RISK_COUNCIL: () => RISK_COUNCIL
});
module.exports = __toCommonJS(GhoEthereum_exports);
var GHO_TOKEN = "0x40D16FC0246aD3160Ccc09B8D0D3A2cD28aE6C2f";
var GHO_CCIP_TOKEN_POOL = "0x06179f7C1be40863405f374E7f5F8806c728660A";
var GHO_FLASHMINTER_FACILITATOR = "0xb639D208Bcf0589D54FaC24E655C79EC529762B8";
var GHO_LIQUIDITY_COMMITTEE = "0xA1c93D2687f7014Aaf588c764E3Ce80aF016229b";
var GHO_GSM_STEWARD = "0xD1E856a947CdF56b4f000ee29d34F5808E0A6848";
var GSM_REGISTRY = "0x167527DB01325408696326e3580cd8e55D99Dc1A";
var GSM_USDC = "0xFeeb6FE430B7523fEF2a38327241eE7153779535";
var GSM_USDT = "0x535b2f7C20B9C83d70e519cf9991578eF9816B7B";
var GSM_USDC_FEE_STRATEGY = "0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5";
var GSM_USDT_FEE_STRATEGY = "0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5";
var GSM_USDC_PRICE_STRATEGY = "0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a";
var GSM_USDT_PRICE_STRATEGY = "0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c";
var GSM_USDC_ORACLE_SWAP_FREEZER = "0xef6beCa8D9543eC007bceA835aF768B58F730C1f";
var GSM_USDT_ORACLE_SWAP_FREEZER = "0x71381e6718b37C12155CB961Ca3D374A8BfFa0e5";
var GHO_AAVE_CORE_STEWARD = "0x98217A06721Ebf727f2C8d9aD7718ec28b7aAe34";
var GHO_BUCKET_STEWARD = "0x46Aa1063e5265b43663E81329333B47c517A5409";
var GHO_CCIP_STEWARD = "0xC5BcC58BE6172769ca1a78B8A45752E3C5059c39";
var RISK_COUNCIL = "0x8513e6F37dBc52De87b166980Fa3F50639694B60";
var CHAIN_ID = 1;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL,
  GHO_FLASHMINTER_FACILITATOR,
  GHO_GSM_STEWARD,
  GHO_LIQUIDITY_COMMITTEE,
  GHO_TOKEN,
  GSM_REGISTRY,
  GSM_USDC,
  GSM_USDC_FEE_STRATEGY,
  GSM_USDC_ORACLE_SWAP_FREEZER,
  GSM_USDC_PRICE_STRATEGY,
  GSM_USDT,
  GSM_USDT_FEE_STRATEGY,
  GSM_USDT_ORACLE_SWAP_FREEZER,
  GSM_USDT_PRICE_STRATEGY,
  RISK_COUNCIL
});
