// @bgd-labs/aave-address-book/dist/GhoGnosis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GhoGnosis.ts
var GhoGnosis_exports = {};
__export(GhoGnosis_exports, {
  CHAIN_ID: () => CHAIN_ID,
  GHO_AAVE_CORE_STEWARD: () => GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD: () => GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD: () => GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL: () => GHO_CCIP_TOKEN_POOL,
  GHO_ORACLE: () => GHO_ORACLE,
  GHO_TOKEN: () => GHO_TOKEN
});
module.exports = __toCommonJS(GhoGnosis_exports);
var GHO_TOKEN = "0xfc421aD3C883Bf9E7C4f42dE845C4e4405799e73";
var GHO_CCIP_TOKEN_POOL = "0xDe6539018B095353A40753Dc54C91C68c9487D4E";
var GHO_ORACLE = "0x360d8aa8F6b09B7BC57aF34db2Eb84dD87bf4d12";
var GHO_AAVE_CORE_STEWARD = "0x6e637e1E48025E51315d50ab96d5b3be1971A715";
var GHO_BUCKET_STEWARD = "0x6Bb7a212910682DCFdbd5BCBb3e28FB4E8da10Ee";
var GHO_CCIP_STEWARD = "0x06179f7C1be40863405f374E7f5F8806c728660A";
var CHAIN_ID = 100;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  GHO_AAVE_CORE_STEWARD,
  GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL,
  GHO_ORACLE,
  GHO_TOKEN
});
