// @bgd-labs/aave-address-book/dist/GhoInk.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GhoInk.ts
var GhoInk_exports = {};
__export(GhoInk_exports, {
  CHAIN_ID: () => CHAIN_ID,
  GHO_BUCKET_STEWARD: () => GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD: () => GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL: () => GHO_CCIP_TOKEN_POOL,
  GHO_ORACLE: () => GHO_ORACLE,
  GHO_TOKEN: () => GHO_TOKEN
});
module.exports = __toCommonJS(GhoInk_exports);
var GHO_TOKEN = "0xfc421aD3C883Bf9E7C4f42dE845C4e4405799e73";
var GHO_CCIP_TOKEN_POOL = "0xDe6539018B095353A40753Dc54C91C68c9487D4E";
var GHO_ORACLE = "0x20fd5f3FCac8883a3A0A2bBcD658A2d2c6EFa6B6";
var GHO_BUCKET_STEWARD = "0xA5Ba213867E175A182a5dd6A9193C6158738105A";
var GHO_CCIP_STEWARD = "0x2Ce400703dAcc37b7edFA99D228b8E70a4d3831B";
var CHAIN_ID = 57073;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  GHO_BUCKET_STEWARD,
  GHO_CCIP_STEWARD,
  GHO_CCIP_TOKEN_POOL,
  GHO_ORACLE,
  GHO_TOKEN
});
