// @bgd-labs/aave-address-book/dist/GovernanceV3Polygon.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Polygon.ts
var GovernanceV3Polygon_exports = {};
__export(GovernanceV3Polygon_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE: () => DATA_WAREHOUSE,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER,
  VM_DATA_HELPER: () => VM_DATA_HELPER,
  VOTING_MACHINE: () => VOTING_MACHINE,
  VOTING_STRATEGY: () => VOTING_STRATEGY
});
module.exports = __toCommonJS(GovernanceV3Polygon_exports);
var CROSS_CHAIN_CONTROLLER = "0xF6B99959F0b5e79E1CC7062E12aF632CEb18eF0d";
var CL_EMERGENCY_ORACLE = "0xDAFA1989A504c48Ee20a582f2891eeB25E2fA23F";
var VOTING_MACHINE = "0x44c8b753229006A8047A05b90379A7e92185E97C";
var PAYLOADS_CONTROLLER = "0x401B5D0294E23637c18fcc38b1Bca814CDa2637C";
var PC_DATA_HELPER = "0xE3B770Dc4ae3f8bECaB3Ed12dE692c741603e16A";
var VM_DATA_HELPER = "0x77976B51569896523EE215962Ee91ff236Fa50E8";
var GRANULAR_GUARDIAN = "0x0D2CccD3dD420dC6DE2f24DB44aA22fADE290a02";
var GOVERNANCE_GUARDIAN = "0x1A0581dd5C7C3DA4Ba1CDa7e0BcA7286afc4973b";
var EXECUTOR_LVL_1 = "0xDf7d0e6454DB638881302729F5ba99936EaAB233";
var VOTING_STRATEGY = "0xd755F1CA351E989391a5bacDeC8F780dFD76f5d2";
var DATA_WAREHOUSE = "0xE3D355a9E58D59Ad11F318deAED528f79A9C1e5a";
var CHAIN_ID = 137;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER,
  VM_DATA_HELPER,
  VOTING_MACHINE,
  VOTING_STRATEGY
});
