// @bgd-labs/aave-address-book/dist/GovernanceV3Celo.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Celo.ts
var GovernanceV3Celo_exports = {};
__export(GovernanceV3Celo_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Celo_exports);
var CROSS_CHAIN_CONTROLLER = "0x50F4dAA86F3c747ce15C3C38bD0383200B61d6Dd";
var CL_EMERGENCY_ORACLE = "0x91b21900E91CD302EBeD05E45D8f270ddAED944d";
var PAYLOADS_CONTROLLER = "0xE48E10834C04E394A04BF22a565D063D40b9FA42";
var PC_DATA_HELPER = "0x8657Cd5a0957e8C5BE15c69C67078b5d730D720a";
var GRANULAR_GUARDIAN = "0xbE815420A63A413BB8D508d8022C0FF150Ea7C39";
var GOVERNANCE_GUARDIAN = "0x056E4C4E80D1D14a637ccbD0412CDAAEc5B51F4E";
var EXECUTOR_LVL_1 = "0x1dF462e2712496373A347f8ad10802a5E95f053D";
var CHAIN_ID = 42220;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
