// @bgd-labs/aave-address-book/dist/GovernanceV3Avalanche.mjs
// src/ts/GovernanceV3Avalanche.ts
var CROSS_CHAIN_CONTROLLER = "0x27FC7D54C893dA63C0AE6d57e1B2B13A70690928";
var CL_EMERGENCY_ORACLE = "0x41185495Bc8297a65DC46f94001DC7233775EbEe";
var VOTING_MACHINE = "0x4D1863d22D0ED8579f8999388BCC833CB057C2d6";
var PAYLOADS_CONTROLLER = "0x1140CB7CAfAcC745771C2Ea31e7B5C653c5d0B80";
var PC_DATA_HELPER = "0xE3B770Dc4ae3f8bECaB3Ed12dE692c741603e16A";
var VM_DATA_HELPER = "0x77976B51569896523EE215962Ee91ff236Fa50E8";
var GRANULAR_GUARDIAN = "0xc1162BCb2E5E3ca4725512008c7522dF8C8B7B65";
var GOVERNANCE_GUARDIAN = "0x360c0a69Ed2912351227a0b745f890CB2eBDbcFe";
var EXECUTOR_LVL_1 = "0x3C06dce358add17aAf230f2234bCCC4afd50d090";
var VOTING_STRATEGY = "0x6771B74568181f6042C9e5CFF98f62e7Ce40482e";
var DATA_WAREHOUSE = "0x450820e342B731b542caA5328d06F26528914dDd";
var CHAIN_ID = 43114;
export {
  CHAIN_ID,
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  DATA_WAREHOUSE,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER,
  VM_DATA_HELPER,
  VOTING_MACHINE,
  VOTING_STRATEGY
};
