// @bgd-labs/aave-address-book/dist/GovernanceV3Ink.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Ink.ts
var GovernanceV3Ink_exports = {};
__export(GovernanceV3Ink_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Ink_exports);
var CROSS_CHAIN_CONTROLLER = "0x990B75fD1a2345D905a385dBC6e17BEe0Cb2f505";
var PAYLOADS_CONTROLLER = "0x44D73D7C4b2f98F426Bf8B5e87628d9eE38ef0Cf";
var PC_DATA_HELPER = "0xd0929668178973d5994D5654929aCB3d6c2b9949";
var GRANULAR_GUARDIAN = "0xa2bDB2335Faf1940c99654c592B1a80618d79Fc9";
var GOVERNANCE_GUARDIAN = "0x1bBcC6F0BB563067Ca45450023a13E34fa963Fa9";
var EXECUTOR_LVL_1 = "0x47aAdaAE1F05C978E6aBb7568d11B7F6e0FC4d6A";
var CHAIN_ID = 57073;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
