// @bgd-labs/aave-address-book/dist/GovernanceV3Plasma.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Plasma.ts
var GovernanceV3Plasma_exports = {};
__export(GovernanceV3Plasma_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Plasma_exports);
var CROSS_CHAIN_CONTROLLER = "0x643441742f73e270e565619be6DE5f4D55E08cd6";
var PAYLOADS_CONTROLLER = "0xe76EB348E65eF163d85ce282125FF5a7F5712A1d";
var PC_DATA_HELPER = "0xA806DA549FcB2B4912a7dFFE4c1aA7A1ed0Bd5C9";
var GRANULAR_GUARDIAN = "0x60665b4F4FF7073C5fed2656852dCa271DfE2684";
var GOVERNANCE_GUARDIAN = "0x19CE4363FEA478Aa04B9EA2937cc5A2cbcD44be6";
var CL_EMERGENCY_ORACLE = "0xF61FE74Ec1cFbd9Ee8Bd27592D2EDEe0E2aA85Cf";
var EXECUTOR_LVL_1 = "0x47aAdaAE1F05C978E6aBb7568d11B7F6e0FC4d6A";
var CHAIN_ID = 9745;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
