// @bgd-labs/aave-address-book/dist/GovernanceV3Linea.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Linea.ts
var GovernanceV3Linea_exports = {};
__export(GovernanceV3Linea_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Linea_exports);
var CROSS_CHAIN_CONTROLLER = "0x0D3f821e9741C8a8Bcac231162320251Db0cdf52";
var PAYLOADS_CONTROLLER = "0x3BcE23a1363728091bc57A58a226CF2940C2e074";
var PC_DATA_HELPER = "0x6d4F341d8Bb3Dc5ABe822Aa940F1884508C13f99";
var GRANULAR_GUARDIAN = "0xc1cd6faF6e9138b4e6C21d438f9ebF2bd6F6cA16";
var GOVERNANCE_GUARDIAN = "0x056E4C4E80D1D14a637ccbD0412CDAAEc5B51F4E";
var EXECUTOR_LVL_1 = "0x8c2d95FE7aeB57b86961F3abB296A54f0ADb7F88";
var CHAIN_ID = 59144;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
