// @bgd-labs/aave-address-book/dist/GovernanceV3ZkSync.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3ZkSync.ts
var GovernanceV3ZkSync_exports = {};
__export(GovernanceV3ZkSync_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3ZkSync_exports);
var CROSS_CHAIN_CONTROLLER = "0x800813f4714BC7A0a95310e3fB9e4f18872CA92C";
var PAYLOADS_CONTROLLER = "0x2E79349c3F5e4751E87b966812C9E65E805996F1";
var PC_DATA_HELPER = "0xe28A3235DCF1Acb8397B546bd588bAAFD7081505";
var GRANULAR_GUARDIAN = "0xe0e23196D42b54F262a3DE952e6B34B197D1A228";
var GOVERNANCE_GUARDIAN = "0x4257bf0746D783f0D962913d7d8AFA408B62547E";
var EXECUTOR_LVL_1 = "0x04cE39789e11a49595cD0ECEf6f4Bd54ABF4d020";
var CHAIN_ID = 324;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
