// @bgd-labs/aave-address-book/dist/GovernanceV3Sonic.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Sonic.ts
var GovernanceV3Sonic_exports = {};
__export(GovernanceV3Sonic_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CL_EMERGENCY_ORACLE: () => CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Sonic_exports);
var CROSS_CHAIN_CONTROLLER = "0x58e003a3C6f2Aeed6a2a6Bc77B504566523cb15c";
var PAYLOADS_CONTROLLER = "0x0846C28Dd54DEA4Fd7Fb31bcc5EB81673D68c695";
var PC_DATA_HELPER = "0x6fD45D32375d5aDB8D76275A3932c740F03a8718";
var GRANULAR_GUARDIAN = "0x10078c1D8E46dd1ed5D8df2C42d5ABb969b11566";
var GOVERNANCE_GUARDIAN = "0x63C4422D6cc849549daeb600B7EcE52bD18fAd7f";
var CL_EMERGENCY_ORACLE = "0xECB564e91f620fBFb59d0C4A41d7f10aDb0D1934";
var EXECUTOR_LVL_1 = "0x7b62461a3570c6AC8a9f8330421576e417B71EE7";
var CHAIN_ID = 146;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CL_EMERGENCY_ORACLE,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
