// @bgd-labs/aave-address-book/dist/GovernanceV3Bob.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Bob.ts
var GovernanceV3Bob_exports = {};
__export(GovernanceV3Bob_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Bob_exports);
var CROSS_CHAIN_CONTROLLER = "0xf630C8A7bC033FD20fcc45d8B43bFe92dE73154F";
var PAYLOADS_CONTROLLER = "0x17fa87007bfF1dC7e6b3a36ED936E6355e37237C";
var PC_DATA_HELPER = "0x9fE056F44510F970d724adA16903ba5D75CC4742";
var GRANULAR_GUARDIAN = "0xb2C672931Bd1Da226e29997Ec8cEB60Fb1DA3959";
var GOVERNANCE_GUARDIAN = "0x19CE4363FEA478Aa04B9EA2937cc5A2cbcD44be6";
var EXECUTOR_LVL_1 = "0x90800d1F54384523723eD3962c7Cd59d7866c83d";
var CHAIN_ID = 60808;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
