// @bgd-labs/aave-address-book/dist/GovernanceV3Mantle.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/GovernanceV3Mantle.ts
var GovernanceV3Mantle_exports = {};
__export(GovernanceV3Mantle_exports, {
  CHAIN_ID: () => CHAIN_ID,
  CROSS_CHAIN_CONTROLLER: () => CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1: () => EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN: () => GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN: () => GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER: () => PAYLOADS_CONTROLLER,
  PC_DATA_HELPER: () => PC_DATA_HELPER
});
module.exports = __toCommonJS(GovernanceV3Mantle_exports);
var CROSS_CHAIN_CONTROLLER = "0x1283C5015B1Fb5616FA3aCb0C18e6879a02869cB";
var PAYLOADS_CONTROLLER = "0xF089f77173A3009A98c45f49D547BF714A7B1e01";
var PC_DATA_HELPER = "0x5e06b10B3b9c3E1c0996D2544A35B9839Be02922";
var GRANULAR_GUARDIAN = "0xb26670d2800DBB9cfCe2f2660FfDcA48C799c86d";
var GOVERNANCE_GUARDIAN = "0x14816fC7f443A9C834d30eeA64daD20C4f56fBCD";
var EXECUTOR_LVL_1 = "0x70884634D0098782592111A2A6B8d223be31CB7b";
var CHAIN_ID = 5e3;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CHAIN_ID,
  CROSS_CHAIN_CONTROLLER,
  EXECUTOR_LVL_1,
  GOVERNANCE_GUARDIAN,
  GRANULAR_GUARDIAN,
  PAYLOADS_CONTROLLER,
  PC_DATA_HELPER
});
