// @bgd-labs/aave-address-book/dist/MiscOptimism.mjs
// src/ts/MiscOptimism.ts
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xD3cF979e676265e4f6379749DECe4708B9A22476";
var CREATE_3_FACTORY = "0x3b56998Ec06477704622ca8e2eA1b4db134cec32";
var AAVE_CL_ROBOT_OPERATOR = "0x55Cf9583D7D30DC4936bAee1f747591dBECe5df7";
var PROTOCOL_GUARDIAN = "0x56C1a4b54921DEA9A344967a8693C7E661D72968";
var AAVE_MERKLE_DISTRIBUTOR = "0x1685D81212580DD4cDA287616C2f6F4794927e18";
var wstETH_stETH_AGGREGATOR = "0xe59EBa0D492cA53C6f46015EEa00517F2707dc77";
var rETH_ETH_AGGREGATOR = "0x22F3727be377781d1579B7C9222382b21c9d1a8f";
var LEGACY_BRIDGE_EXECUTOR = "0x7d9103572bE58FfE99dc390E8246f02dcAe6f611";
var AAVE_OPT_ETH_BRIDGE = "0xc3250A20F8a7BbDd23adE87737EE46A45Fe5543E";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var ALC_SAFE = "0xA1c93D2687f7014Aaf588c764E3Ce80aF016229b";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var MERIT_AHAB_SAFE = "0xdeadD8aB03075b7FBA81864202a2f59EE25B312b";
var APE_SAFE = "0xAA43203167317DeeF8288095C44b84a686918d2E";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var MERIT_INCENTIVE_SAFE = "0xAA870e4B82deaDa3727235f34183Ec9B728714C8";
var AHAB_SAFE = "0xAA2461f0f0A3dE5fEAF3273eAe16DEF861cf594e";
var CHAIN_ID = 10;
export {
  AAVE_CL_ROBOT_OPERATOR,
  AAVE_MERKLE_DISTRIBUTOR,
  AAVE_OPT_ETH_BRIDGE,
  AFC_SAFE,
  AHAB_SAFE,
  ALC_SAFE,
  APE_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE,
  MERIT_AHAB_SAFE,
  MERIT_INCENTIVE_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  rETH_ETH_AGGREGATOR,
  wstETH_stETH_AGGREGATOR
};
