// @bgd-labs/aave-address-book/dist/MiscBase.mjs
// src/ts/MiscBase.ts
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xc85b1E333aecc99340b2320493Fe2d22b8734795";
var CREATE_3_FACTORY = "0x3b56998Ec06477704622ca8e2eA1b4db134cec32";
var AAVE_CL_ROBOT_OPERATOR = "0x88db99eeBb390a2a4DcAC2E1DDb09c07E911C5C3";
var PROTOCOL_GUARDIAN = "0x56C1a4b54921DEA9A344967a8693C7E661D72968";
var wstETH_stETH_AGGREGATOR = "0xB88BAc61a4Ca37C43a3725912B1f472c9A5bc061";
var cbETH_ETH_AGGREGATOR = "0x868a501e68F3D1E89CfC0D22F6b22E8dabce5F04";
var LEGACY_BRIDGE_EXECUTOR = "0xA9F30e6ED4098e9439B2ac8aEA2d3fc26BcEbb45";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var ALC_SAFE = "0xA1c93D2687f7014Aaf588c764E3Ce80aF016229b";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var APE_SAFE = "0xAA43203167317DeeF8288095C44b84a686918d2E";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var ALC_INCENTIVE_SAFE = "0xAAB6f926DCDaE536F54ce58478Dbc1a0d0f98871";
var MERIT_INCENTIVE_SAFE = "0xAA870e4B82deaDa3727235f34183Ec9B728714C8";
var AHAB_SAFE = "0xAA2461f0f0A3dE5fEAF3273eAe16DEF861cf594e";
var CHAIN_ID = 8453;
export {
  AAVE_CL_ROBOT_OPERATOR,
  AFC_SAFE,
  AHAB_SAFE,
  ALC_INCENTIVE_SAFE,
  ALC_SAFE,
  APE_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE,
  MERIT_INCENTIVE_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  cbETH_ETH_AGGREGATOR,
  wstETH_stETH_AGGREGATOR
};
