// @bgd-labs/aave-address-book/dist/MiscLinea.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/MiscLinea.ts
var MiscLinea_exports = {};
__export(MiscLinea_exports, {
  AFC_SAFE: () => AFC_SAFE,
  CHAIN_ID: () => CHAIN_ID,
  MASIV_SAFE: () => MASIV_SAFE,
  PROTOCOL_GUARDIAN: () => PROTOCOL_GUARDIAN,
  PROXY_ADMIN: () => PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY: () => TRANSPARENT_PROXY_FACTORY
});
module.exports = __toCommonJS(MiscLinea_exports);
var TRANSPARENT_PROXY_FACTORY = "0xEf62Fc23727c17b612081114b1919d139a7b1B83";
var PROXY_ADMIN = "0x160E35e28fEE90F3656420584e0a990276219b5A";
var PROTOCOL_GUARDIAN = "0x0BF186764D8333a938f35e5dD124a7b9b9dccDF9";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var CHAIN_ID = 59144;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AFC_SAFE,
  CHAIN_ID,
  MASIV_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY
});
