// @bgd-labs/aave-address-book/dist/MiscCelo.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/MiscCelo.ts
var MiscCelo_exports = {};
__export(MiscCelo_exports, {
  AFC_SAFE: () => AFC_SAFE,
  CHAIN_ID: () => CHAIN_ID,
  MASIV_SAFE: () => MASIV_SAFE,
  PROTOCOL_GUARDIAN: () => PROTOCOL_GUARDIAN,
  PROXY_ADMIN: () => PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY: () => TRANSPARENT_PROXY_FACTORY
});
module.exports = __toCommonJS(MiscCelo_exports);
var TRANSPARENT_PROXY_FACTORY = "0x1Be33D186a081eFDf15310e4bdEd81E828C47f9b";
var PROXY_ADMIN = "0x54BDcc37c4143f944A3EE51C892a6cBDF305E7a0";
var PROTOCOL_GUARDIAN = "0x88E7aB6ee481Cf92e548c0e1169F824F99142c85";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var CHAIN_ID = 42220;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AFC_SAFE,
  CHAIN_ID,
  MASIV_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY
});
