// @bgd-labs/aave-address-book/dist/MiscPolygon.mjs
// src/ts/MiscPolygon.ts
var AAVE_POL_ETH_BRIDGE = "0x1C2BA5b8ab8e795fF44387ba6d251fa65AD20b36";
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xD3cF979e676265e4f6379749DECe4708B9A22476";
var CREATE_3_FACTORY = "0x3b56998Ec06477704622ca8e2eA1b4db134cec32";
var AAVE_CL_ROBOT_OPERATOR = "0xB4C212f5cD17E200019b07e6B1fDf124d35DBCf5";
var PROTOCOL_GUARDIAN = "0xCb45E82419baeBCC9bA8b1e5c7858e48A3B26Ea6";
var AAVE_MERKLE_DISTRIBUTOR = "0x7A9ff54A6eE4a21223036890bB8c4ea2D62c686b";
var wstETH_stETH_AGGREGATOR = "0x3Ea1eC855fBda8bA0396975eC260AD2e9B2Bc01c";
var MaticX_RATE_PROVIDER = "0xeE652bbF72689AA59F0B8F981c9c90e2A8Af8d8f";
var stMATIC_RATE_PROVIDER = "0xdEd6C522d803E35f65318a9a4d7333a22d582199";
var LEGACY_BRIDGE_EXECUTOR = "0xdc9A35B16DB4e126cFeDC41322b3a36454B1F772";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var CHAIN_ID = 137;
export {
  AAVE_CL_ROBOT_OPERATOR,
  AAVE_MERKLE_DISTRIBUTOR,
  AAVE_POL_ETH_BRIDGE,
  AFC_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE,
  MaticX_RATE_PROVIDER,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  stMATIC_RATE_PROVIDER,
  wstETH_stETH_AGGREGATOR
};
