// @bgd-labs/aave-address-book/dist/MiscGnosis.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/MiscGnosis.ts
var MiscGnosis_exports = {};
__export(MiscGnosis_exports, {
  AFC_SAFE: () => AFC_SAFE,
  AHAB_SAFE: () => AHAB_SAFE,
  ALC_INCENTIVE_SAFE: () => ALC_INCENTIVE_SAFE,
  APE_SAFE: () => APE_SAFE,
  APE_VOTING_SAFE: () => APE_VOTING_SAFE,
  CHAIN_ID: () => CHAIN_ID,
  CREATE_3_FACTORY: () => CREATE_3_FACTORY,
  MASIV_SAFE: () => MASIV_SAFE,
  MERIT_AHAB_SAFE: () => MERIT_AHAB_SAFE,
  MERIT_INCENTIVE_SAFE: () => MERIT_INCENTIVE_SAFE,
  PROTOCOL_GUARDIAN: () => PROTOCOL_GUARDIAN,
  PROXY_ADMIN: () => PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY: () => TRANSPARENT_PROXY_FACTORY,
  wstETH_stETH_AGGREGATOR: () => wstETH_stETH_AGGREGATOR
});
module.exports = __toCommonJS(MiscGnosis_exports);
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xe892E40C92c2E4D281Be59b2E6300F271d824E75";
var CREATE_3_FACTORY = "0x2cC41488f4494c5e11cB6cbA82199442B9f495B6";
var PROTOCOL_GUARDIAN = "0xCb45E82419baeBCC9bA8b1e5c7858e48A3B26Ea6";
var wstETH_stETH_AGGREGATOR = "0x0064AC007fF665CF8D0D3Af5E0AD1c26a3f853eA";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var MERIT_AHAB_SAFE = "0xdeadD8aB03075b7FBA81864202a2f59EE25B312b";
var APE_SAFE = "0xAA43203167317DeeF8288095C44b84a686918d2E";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var ALC_INCENTIVE_SAFE = "0xAAB6f926DCDaE536F54ce58478Dbc1a0d0f98871";
var MERIT_INCENTIVE_SAFE = "0xAA870e4B82deaDa3727235f34183Ec9B728714C8";
var AHAB_SAFE = "0xAA2461f0f0A3dE5fEAF3273eAe16DEF861cf594e";
var CHAIN_ID = 100;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AFC_SAFE,
  AHAB_SAFE,
  ALC_INCENTIVE_SAFE,
  APE_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  MASIV_SAFE,
  MERIT_AHAB_SAFE,
  MERIT_INCENTIVE_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  wstETH_stETH_AGGREGATOR
});
