// @bgd-labs/aave-address-book/dist/MiscArbitrum.mjs
// src/ts/MiscArbitrum.ts
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xD3cF979e676265e4f6379749DECe4708B9A22476";
var CREATE_3_FACTORY = "0x3b56998Ec06477704622ca8e2eA1b4db134cec32";
var AAVE_CL_ROBOT_OPERATOR = "0xaa944aD95e51CB83C1f35FAEEDfC7d2c31B0BB4d";
var PROTOCOL_GUARDIAN = "0xCb45E82419baeBCC9bA8b1e5c7858e48A3B26Ea6";
var wstETH_stETH_AGGREGATOR = "0xB1552C5e96B312d0Bf8b554186F846C40614a540";
var rETH_ETH_AGGREGATOR = "0xF3272CAfe65b190e76caAF483db13424a3e23dD2";
var LEGACY_BRIDGE_EXECUTOR = "0x7d9103572bE58FfE99dc390E8246f02dcAe6f611";
var AAVE_ARB_ETH_BRIDGE = "0x0335ffa9af5CE05590d6C9A75B645470e07744a9";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var ALC_SAFE = "0xA1c93D2687f7014Aaf588c764E3Ce80aF016229b";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var MERIT_AHAB_SAFE = "0xdeadD8aB03075b7FBA81864202a2f59EE25B312b";
var APE_SAFE = "0xAA43203167317DeeF8288095C44b84a686918d2E";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var ALC_INCENTIVE_SAFE = "0xAAB6f926DCDaE536F54ce58478Dbc1a0d0f98871";
var MERIT_INCENTIVE_SAFE = "0xAA870e4B82deaDa3727235f34183Ec9B728714C8";
var AHAB_SAFE = "0xAA2461f0f0A3dE5fEAF3273eAe16DEF861cf594e";
var CHAIN_ID = 42161;
export {
  AAVE_ARB_ETH_BRIDGE,
  AAVE_CL_ROBOT_OPERATOR,
  AFC_SAFE,
  AHAB_SAFE,
  ALC_INCENTIVE_SAFE,
  ALC_SAFE,
  APE_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE,
  MERIT_AHAB_SAFE,
  MERIT_INCENTIVE_SAFE,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  rETH_ETH_AGGREGATOR,
  wstETH_stETH_AGGREGATOR
};
