// @bgd-labs/aave-address-book/dist/MiscPolygon.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ts/MiscPolygon.ts
var MiscPolygon_exports = {};
__export(MiscPolygon_exports, {
  AAVE_CL_ROBOT_OPERATOR: () => AAVE_CL_ROBOT_OPERATOR,
  AAVE_MERKLE_DISTRIBUTOR: () => AAVE_MERKLE_DISTRIBUTOR,
  AAVE_POL_ETH_BRIDGE: () => AAVE_POL_ETH_BRIDGE,
  AFC_SAFE: () => AFC_SAFE,
  APE_VOTING_SAFE: () => APE_VOTING_SAFE,
  CHAIN_ID: () => CHAIN_ID,
  CREATE_3_FACTORY: () => CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR: () => LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE: () => MASIV_SAFE,
  MaticX_RATE_PROVIDER: () => MaticX_RATE_PROVIDER,
  PROTOCOL_GUARDIAN: () => PROTOCOL_GUARDIAN,
  PROXY_ADMIN: () => PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY: () => TRANSPARENT_PROXY_FACTORY,
  stMATIC_RATE_PROVIDER: () => stMATIC_RATE_PROVIDER,
  wstETH_stETH_AGGREGATOR: () => wstETH_stETH_AGGREGATOR
});
module.exports = __toCommonJS(MiscPolygon_exports);
var AAVE_POL_ETH_BRIDGE = "0x1C2BA5b8ab8e795fF44387ba6d251fa65AD20b36";
var TRANSPARENT_PROXY_FACTORY = "0xEB0682d148e874553008730f0686ea89db7DA412";
var PROXY_ADMIN = "0xD3cF979e676265e4f6379749DECe4708B9A22476";
var CREATE_3_FACTORY = "0x3b56998Ec06477704622ca8e2eA1b4db134cec32";
var AAVE_CL_ROBOT_OPERATOR = "0xB4C212f5cD17E200019b07e6B1fDf124d35DBCf5";
var PROTOCOL_GUARDIAN = "0xCb45E82419baeBCC9bA8b1e5c7858e48A3B26Ea6";
var AAVE_MERKLE_DISTRIBUTOR = "0x7A9ff54A6eE4a21223036890bB8c4ea2D62c686b";
var wstETH_stETH_AGGREGATOR = "0x3Ea1eC855fBda8bA0396975eC260AD2e9B2Bc01c";
var MaticX_RATE_PROVIDER = "0xeE652bbF72689AA59F0B8F981c9c90e2A8Af8d8f";
var stMATIC_RATE_PROVIDER = "0xdEd6C522d803E35f65318a9a4d7333a22d582199";
var LEGACY_BRIDGE_EXECUTOR = "0xdc9A35B16DB4e126cFeDC41322b3a36454B1F772";
var AFC_SAFE = "0x22740deBa78d5a0c24C58C740e3715ec29de1bFa";
var MASIV_SAFE = "0xdef1FA4CEfe67365ba046a7C630D6B885298E210";
var APE_VOTING_SAFE = "0xa9e777D56C0Ad861f6a03967E080e767ad8D39b6";
var CHAIN_ID = 137;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AAVE_CL_ROBOT_OPERATOR,
  AAVE_MERKLE_DISTRIBUTOR,
  AAVE_POL_ETH_BRIDGE,
  AFC_SAFE,
  APE_VOTING_SAFE,
  CHAIN_ID,
  CREATE_3_FACTORY,
  LEGACY_BRIDGE_EXECUTOR,
  MASIV_SAFE,
  MaticX_RATE_PROVIDER,
  PROTOCOL_GUARDIAN,
  PROXY_ADMIN,
  TRANSPARENT_PROXY_FACTORY,
  stMATIC_RATE_PROVIDER,
  wstETH_stETH_AGGREGATOR
});
