// @alchemy/aa-accounts/dist/cjs/plugindefs/multi-owner/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiOwnerPluginGenConfig = void 0;
const aa_core_1 = require("@alchemy/aa-core");
const viem_1 = require("viem");
const abi_js_1 = require("./abi.js");
exports.MultiOwnerPluginGenConfig = {
    name: "MultiOwnerPlugin",
    abi: abi_js_1.MultiOwnerPluginAbi,
    addresses: {
        [aa_core_1.sepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.baseSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.polygon.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.polygonAmoy.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.polygonMumbai.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.optimism.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.optimismSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.arbitrum.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.arbitrumSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.base.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.baseSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.fraxtal.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.fraxtalSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.zora.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.zoraSepolia.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
        [aa_core_1.mainnet.id]: "0xcE0000007B008F50d762D155002600004cD6c647",
    },
    chain: aa_core_1.sepolia,
    installConfig: {
        initAbiParams: (0, viem_1.parseAbiParameters)("address[]"),
    },
};
