// @alchemy/aa-accounts/dist/cjs/src/msca/plugins/multisig/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.multisigPluginActions = void 0;
const getThreshold_js_1 = require("./actions/getThreshold.js");
const isOwnerOf_js_1 = require("./actions/isOwnerOf.js");
const proposeUserOperation_js_1 = require("./actions/proposeUserOperation.js");
const readOwners_js_1 = require("./actions/readOwners.js");
const signMultisigUserOperation_js_1 = require("./actions/signMultisigUserOperation.js");
const plugin_js_1 = require("./plugin.js");
const multisigPluginActions = (client) => ({
    ...(0, plugin_js_1.multisigPluginActions)(client),
    readOwners: (args) => (0, readOwners_js_1.readOwners)(client, args),
    isOwnerOf: (args) => (0, isOwnerOf_js_1.isOwnerOf)(client, args),
    getThreshold: (args) => (0, getThreshold_js_1.getThreshold)(client, args),
    proposeUserOperation: (args) => (0, proposeUserOperation_js_1.proposeUserOperation)(client, args),
    signMultisigUserOperation: (params) => (0, signMultisigUserOperation_js_1.signMultisigUserOperation)(client, params),
});
exports.multisigPluginActions = multisigPluginActions;
