// @alchemy/aa-accounts/dist/cjs/src/msca/plugins/multisig/signer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.multisigSignMethods = void 0;
const viem_1 = require("viem");
const plugin_js_1 = require("./plugin.js");
const multisigSignMethods = ({ client, accountAddress, signer, threshold, pluginAddress = plugin_js_1.MultisigPlugin.meta.addresses[client.chain.id], }) => {
    const signWith712Wrapper = async (msg) => {
        const [, name, version, chainId, verifyingContract, salt] = await client.readContract({
            abi: plugin_js_1.MultisigPluginAbi,
            address: pluginAddress,
            functionName: "eip712Domain",
            account: accountAddress,
        });
        return signer().signTypedData({
            domain: {
                chainId: Number(chainId),
                name,
                salt,
                verifyingContract,
                version,
            },
            types: {
                AlchemyMultisigMessage: [{ name: "message", type: "bytes" }],
            },
            message: {
                message: msg,
            },
            primaryType: "AlchemyMultisigMessage",
        });
    };
    return {
        getDummySignature: async () => {
            const [, thresholdRead] = await client.readContract({
                abi: plugin_js_1.MultisigPluginAbi,
                address: pluginAddress,
                functionName: "ownershipInfoOf",
                args: [accountAddress],
            });
            const actualThreshold = thresholdRead === 0n ? threshold : thresholdRead;
            return ("0x" +
                "FF".repeat(32 * 3) +
                "fffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa3c" +
                "fffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1c".repeat(Number(actualThreshold) - 1));
        },
        signUserOperationHash: (uoHash) => {
            return signer().signMessage({ raw: uoHash });
        },
        signMessage({ message }) {
            return signWith712Wrapper((0, viem_1.hashMessage)(message));
        },
        signTypedData: (typedDataDefinition) => {
            return signWith712Wrapper((0, viem_1.hashTypedData)(typedDataDefinition));
        },
    };
};
exports.multisigSignMethods = multisigSignMethods;
